---
title: "AIM2_update(01_13_25)"
output: html_document
date: "2025-01-13"
---

```{r}
library(dplyr)
library(tximport)
library(DESeq2)
library(ggplot2)
library(sva)
library(pheatmap) # for hierarchical clustering
library(purrr) # for "set_names()"
library(tibble)
library(ggVennDiagram)
library(UpSetR)
library(GenomicFeatures)
library(devtools) # for making the Org.db
library(AnnotationForge)
library(robustbase) # needed for rrcov
library(rrcov) # for statistically testing for outliers in PCA
library(stringr) # for str_detect to filter metadata
library(xlsx)
```

## Import my data (tximport)
```{r }
sample_table = read.csv("/Users/school/Projects/CR/samples.txt", header = T, stringsAsFactors = T)
sample_table$P_SAV <- as.character(sample_table$P_SAV)
sample_table$Age <- factor(sample_table$Age, levels = c("M","A") )
sample_table$TRT <- factor(sample_table$TRT, levels = c("U", "T"))
names(sample_table)[names(sample_table) == "TRT"] <- "LFI"
row.names(sample_table)=sample_table$Sample_ID

paste0('/Users/school/Projects/CR/quants/', 
       pull(sample_table, RNAseq_ID), '/quant.sf')
sample_files = paste0('/Users/school/Projects/CR/quants/',
                      pull(sample_table, RNAseq_ID), '/quant.sf')
names(sample_files) = pull(sample_table, Sample_ID)
all(file.exists(sample_files))
data.frame(thefiles = sample_files, doihave = file.exists(sample_files))

txdb_aplysia <- makeTxDbFromGFF("/Users/school/Documents/UAB/UAB009/genome/AplCal3.0_GFF.gff",
                        format = "gff",
                        organism = "Aplysia californica")
k_aplysia <- keys(txdb_aplysia, keytype = "TXNAME")
tx2gene_aplysia <- AnnotationDbi::select(txdb_aplysia, k_aplysia, "GENEID", "TXNAME")
tx2gene_aplysia$TXNAME <- gsub("\\.[0-9]","",tx2gene_aplysia$TXNAME) 

count_data_All = tximport(files = sample_files, importer=read.delim,
         type = "salmon",
         tx2gene = tx2gene_aplysia,
         ignoreTxVersion = T)

```

### dealing with ouliers and DUDs
```{r}
# I know from previous work on these samples there is one DUD and 3 outliers. The following code chunk will deal these and end in the removal of these samples for downstream analysis
# We will start by locating the outliers. This involves making a model with all samples and poting a PCA with 99.9% CI and identifying samples that fall outside these intervals.
# the DUD is sample ID 840 which can be identified using the metadata dataframe "sample_table" This sample is an Aged, AL, Trained (A,T,CR) sample 
dds <- DESeqDataSetFromTximport(count_data_All,
                                colData = sample_table,
                                design = ~ Age + LFI + Diet)
colData(dds)

vsd <- vst(dds)
pc12 <- plotPCA(vsd, intgroup = c("Age", "LFI", "Diet"), returnData = T) #PC1=25, PC2=17
ggplot(pc12, aes(x=PC1, y=PC2, color=Age)) +
  geom_point(aes(fill=Age), 
       colour="black",pch=21, size=5)  +
  xlab(paste0("PC1: 25% varience")) +
  ylab(paste0("PC2: 17% variance")) +
  coord_fixed() +
  #scale_color_manual(labels = c("Younger","Aged"),
  #                   values = c("red","blue")) +
  #ggtitle("PCA with VST data") +
  stat_ellipse(level = 0.999) +
  theme_bw() +
  guides(fill=F) +
  theme(axis.text.x=element_text(size=18), 
        axis.title.x=element_text(size=18),
        axis.text.y=element_text(size=18), 
        axis.title.y=element_text(size=18),
        legend.title = element_text(size=16),
        legend.key.size = unit(0.6, 'cm'),
        legend.position = c(0.91, 0.91)) +
  scale_x_continuous(limits = c(-40, 40)) +
  scale_y_continuous(limits = c(-40, 40))

# Looking at the above PCA we find that there are 4 ouliers. Outliers are defined as any sample that is outside its corresponding ellipse 
# 3 Younger (red) samples fall outside their ellipse
# 1 Aged (blue) sample falls outside its ellipse
# Now lets find out what these samples are...
## To find the Younger (red) samples that are ouliers I will filter the pc12 data for points that are below -15 on PC2
filter(pc12, pc12$PC2 < -15)
## The above filter yielded 3 samples: 805 (M,U,CR), 809 (M,U,CR), and 815 (M,U,CR)

## To find the Aged (blue) outlier I will filter the pc12 data for points that are below -11 on PC2
filter(pc12, pc12$PC2 < -10)
## The above filter yielded 1 sample: 857 (A,T,CR)


# So now I will delete these samples from my metedata file and use the new metadat_minusOutliers to import my quants and so forth
metadata <- sample_table
metadata_minusOutliers <- metadata[c(1:3,5:7,9:11,13:25,27:31,33:48),]
paste0('/Users/school/Projects/CR/quants/', 
       pull(metadata_minusOutliers, RNAseq_ID), '/quant.sf')
sample_files_minusOutliers = paste0('/Users/school/Projects/CR/quants/',
                      pull(metadata_minusOutliers, RNAseq_ID), '/quant.sf')
names(sample_files_minusOutliers) = pull(metadata_minusOutliers, Sample_ID)
all(file.exists(sample_files_minusOutliers))
data.frame(thefiles = metadata_minusOutliers, doihave = file.exists(sample_files_minusOutliers))

count_data_minusOutliers = tximport(files = sample_files_minusOutliers, importer=read.delim,
         type = "salmon",
         tx2gene = tx2gene_aplysia,
         ignoreTxVersion = T)
```
### PCA
```{r}
dds_minusOutliers <- DESeqDataSetFromTximport(count_data_minusOutliers,
                                colData = metadata_minusOutliers,
                                design = ~ Age + LFI + Diet)
colData(dds)
dds_minusOutliers$group <- factor(paste0(dds_minusOutliers$Age, dds_minusOutliers$LFI, dds_minusOutliers$Diet))
design(dds_minusOutliers) <- ~group
colData(dds_minusOutliers)



vsd_minusOutliers <- vst(dds_minusOutliers)
pc12_minusOutliers <- plotPCA(vsd_minusOutliers, intgroup = c("Age", "LFI", "Diet", "group"), returnData = T) #PC1=29, PC2=09
ggplot(pc12_minusOutliers, aes(x=PC1, y=PC2, color=LFI)) +
  geom_point(aes(fill=LFI), 
       colour="black",pch=21, size=5)  +
  xlab(paste0("PC1: 29% varience")) +
  ylab(paste0("PC2: 09% variance")) +
  coord_fixed() +
  #scale_color_manual(labels = c("Younger","Aged"),
  #                   values = c("red","blue")) +
  #ggtitle("PCA with VST data") +
  stat_ellipse(level = 0.999) +
  theme_bw() +
  guides(fill=F) +
  theme(axis.text.x=element_text(size=18), 
        axis.title.x=element_text(size=18),
        axis.text.y=element_text(size=18), 
        axis.title.y=element_text(size=18),
        legend.title = element_text(size=16),
        legend.key.size = unit(0.6, 'cm'),
        legend.position = c(0.91, 0.91)) +
  scale_x_continuous(limits = c(-40, 40)) +
  scale_y_continuous(limits = c(-40, 40))
```
### DESeq2
```{r}
aplysia_dds_minusOutliers <- DESeq(dds_minusOutliers)

# AUAL vs MUAL
summary(results(aplysia_dds_minusOutliers, contrast = c("group", "AUAL", "MUAL"), alpha = 0.05, lfcThreshold = 0))
#AUAL_vs_MUALresSig <- subset(results(aplysia_dds_minusOutliers, contrast = c("group", "AUAL", "MUAL"), lfcThreshold = 0), padj < 0.05)
```




#_____________
# Run with all Samples
```{r}
## as a comparison I will run this will all samples (ignoring any outlieres or DUDs)
dds <- DESeqDataSetFromTximport(count_data_All,
                                colData = sample_table,
                                design = ~ Age + LFI + Diet)
colData(dds)
dds$group <- factor(paste0(dds$Age, dds$LFI, dds$Diet))
design(dds) <- ~group
colData(dds)

## counts pre-filtering
dim(dds) # starting with 21,034 transcripts in 48 samples
smallestGroupSize <- 6 # there are 6 samples in each group
keep <- rowSums(counts(dds) >= 1) >= smallestGroupSize
dds2 <- dds[keep,]
dim(dds2) # 11,536 transcripts in 48 samples
```

### PCA
```{r}
vsd <- vst(dds2)
pc12 <- plotPCA(vsd, intgroup = c("Age","LFI","Diet","group"), returnData = T) #PC1=25, PC2=17
ggplot(pc12, aes(x=PC1, y=PC2, color=Age)) +
  geom_point(aes(fill=Age), 
       colour="black",pch=21, size=5)  +
  xlab(paste0("PC1: 25% varience")) +
  ylab(paste0("PC2: 17% variance")) +
  coord_fixed() +
  #scale_color_manual(labels = c("Younger","Aged"),
  #                   values = c("red","blue")) +
  #ggtitle("PCA with VST data") +
  stat_ellipse(level = 0.999) +
  theme_bw() +
  guides(fill=F) +
  theme(axis.text.x=element_text(size=18), 
        axis.title.x=element_text(size=18),
        axis.text.y=element_text(size=18), 
        axis.title.y=element_text(size=18),
        legend.title = element_text(size=16),
        legend.key.size = unit(0.6, 'cm'),
        legend.position = c(0.91, 0.91)) +
  scale_x_continuous(limits = c(-40, 40)) +
  scale_y_continuous(limits = c(-40, 40))
```

### pcaGrid testing for outliers
```{r}
colnames(vsd)
vsd_colnames <- colnames(vsd)
test <- data.frame(vsd$group)
rowData(vsd)
DESeq2:::vst

PcaGrid(vsd)
# I kept getting this error when trying to run PcaGrid on my varrience stabilization transformation data (vsd)
##Error in h(simpleError(msg, call)) :
###error in evaluating the argument 'x' in selecting a method for function 'as.matrix': no method for coercing this S4 class to a vector
#-----------------------\
# I did some digging and found butcus online. I figured that the input must be a data frame. The manual says: x = a numeric matrix (or data frame) which provides the data for the principal components analysis. 
# So there is a data frame there. For the life of me I could not figure out how to make the vsd (an S4 object) into a data frame
# I just started looking at the different things in the vsd by using "vsd$" the clicking "tab" button and seeing what popped up
###### That did not help as all that returned was the metadata input. 
# I then ran vsd by itself to get a look at it had something titled rownames which are the genes but more interestingly it had rowData names
# I decided to try looking at these by typing the item I wanted to look at followed by vsd in parentheses like --> rownames(vsd)
# I tried that with rowData and BAM! I think I got what I was looking for. I am going to input this as my data for PcaGrid and see what we get

PcaGrid(rowData(vsd)) # this give output of standard deviations
summary(PcaGrid(rowData(vsd))) # this gives importance of components
plot(PcaGrid(rowData(vsd))) ## Ok, we have a problem! This is plotting the genes and determining if they are outliers
#### I don't want to compare the genes! I want to compare the samples!

biplot(PcaGrid(rowData(vsd)))

plot(PcaGrid(testy3))

colData(vsd$Sample_ID)


# OK, so this is a long one. I viewed the DESeq2 function for plotPCA and used it to get a data frame with PC1 and PC2 values to use as inputs in the PcaGrid
# DESeq2:::plotPCA.DESeqTransform
# 
# test <- function (object, intgroup = "condition", ntop = 500, returnData = FALSE) 
# {
#     rv <- rowVars(assay(object))
#     select <- order(rv, decreasing = TRUE)[seq_len(min(ntop, 
#         length(rv)))]
#     pca <- prcomp(t(assay(object)[select, ]))
#     percentVar <- pca$sdev^2/sum(pca$sdev^2)
#     if (!all(intgroup %in% names(colData(vsd)))) {
#         stop("the argument 'intgroup' should specify columns of colData(dds)")
#     }
#     intgroup.df <- as.data.frame(colData(object)[, intgroup, 
#         drop = FALSE])
#     group <- if (length(intgroup) > 1) {
#         factor(apply(intgroup.df, 1, paste, collapse = ":"))
#     }
#     else {
#         colData(object)[[intgroup]]
#     }
#     d <- data.frame(PC1 = pca$x[, 1], PC2 = pca$x[, 2], group = group, 
#         intgroup.df, name = colnames(object))
#     if (returnData) {
#         attr(d, "percentVar") <- percentVar[1:2]
#         return(d)
#     }
# }

### below is how I condenced what was above (above is also not the full function)
makeDataFrame_PC1andPC2_fromVSTdata <- function (object)
{
 rv <- rowVars(assay(object))
    select <- order(rv, decreasing = TRUE)[seq_len(min(500, 
        length(rv)))]
    pca <- prcomp(t(assay(object)[select, ]))
    testy3 <- data.frame(PC1 = pca$x[, 1], PC2 = pca$x[, 2])
}

data4PcaGrid <- makeDataFrame_PC1andPC2_fromVSTdata(vsd)

plot(PcaGrid(data4PcaGrid))
## now when I plot it I get the samples and not the genes. Looks like 805, 809, and 815 are definitely outliers
###### This is consistent with what I decided before.
###### There are many other samples that fall above the line, I do not know if these are outliers or just happen-stance (more reading to do)
rv <- rowVars(assay(vsd))
    select <- order(rv, decreasing = TRUE)[seq_len(min(500, 
        length(rv)))]
    pca <- prcomp(t(assay(vsd)[select, ]))
biplot(PcaGrid(pca$x)) # this looks at both "scores"(data points projected onto principal componenets) and "loadings"(variable contributions to those components)
plot(PcaGrid(pca$x)) # Now this one is using all the PCA values (not just PC1 and PC2) to look for outliers and here only oone sample falls otside the line. Sample 809! 
##### maybe try looking at the data with only this one out

```

## How to use PcaGrid correctly please read below. Above was the first time I used it and I thing I did it wrong. Below I did some troublesooting and at the bottom of this code chunk is how I will use thos tool moving forward
### Test for outliers in PCA
```{r}
vsd <- vst(dds2)
# I will now create a function that yields the PC values (weight) of each sample for each PC using a compressed version of the 'plotPCA' function from DESeq2 (the first 3 lines of this code will do)
DESeq2:::plotPCA.DESeqTransform # this allows me to view the function
prcomp = function (object)
{ rv <- rowVars(assay(object))
    select <- order(rv, decreasing = TRUE)[seq_len(min(500, 
        length(rv)))]
    pca <- prcomp(t(assay(object)[select, ]))
}
environment(prcomp) <- asNamespace('DESeq2')
pca <- prcomp(vsd) # this performed a classical PCA
# we can graph this with 'PcaGrid' from 'rrcov' package to look for outliers
pca$x
# we can also perform a robust PCA using 'PcaCov' function from 'rrcov' package
rpca <- PcaCov(test3)
rpca$loadings
plot(rpca)
test %>% View()
hbk %>% View()
typeof(hbk)
test2 <- as.data.frame(test)
test2 %>% View()
as.list(test2)
test2.2 <- na.omit(test2)

select <- rownames(pca$rotation)
test3 <- assay(vsd)[select,]
# now use 'P' from 'rrcov' package to run a robust PCA
plot(PcaGrid(pca$x))

test <- assay(vsd)
plot(PcaGrid(t(test), k=12)) # this may be it
plot(PcaGrid(test, k=12))
t <- data(hbk)
pca <- PcaCov(hbk)
pca
d <- PcaClassic(hbk, k=2)
print(pca, print.x=T)
PcaCov(~., data = hbk)
plot(pca)
plot(d)
pca2 <- PcaCov(hbk, k=2) # PCA diagnosic plot (outlier map)
plot(pca2$loadings)
hbk %>% View()

pca3 <- PcaCov(test3)
plot(pca3$loadings)

pca4 <- PcaGrid(t(test3))
plot(pca4) ### This makes the most sence!!!!!!
## So we get the top 500 variable genes from DESeq2 plotPCA updated functin that I make. Then assay the vsd and select for those genes. Now we have an assayed vsd with 500 genes. We run PcaGrid on the transposed vsd assay and save as an object we then plot the object and get 2 samples as outliers 809 and 805, this makies the most scense! May need to set the k but this is what we wil follow. See below.


## vst the data and get the top 500 genes
vsd <- vst(dds2)
# I will now create a function that yields the PC values (weight) of each sample for each PC using a compressed version of the 'plotPCA' function from DESeq2 (the first 3 lines of this code will do)
DESeq2:::plotPCA.DESeqTransform # this allows me to view the function
prcomp = function (object)
{ rv <- rowVars(assay(object))
    select <- order(rv, decreasing = TRUE)[seq_len(min(500, 
        length(rv)))]
    pca <- prcomp(t(assay(object)[select, ]))
}
environment(prcomp) <- asNamespace('DESeq2')
pca <- prcomp(vsd)

# make an object of just the top 500 gene names
select <- rownames(pca$rotation)
test3 <- assay(vsd)[select,] # assay the vsd and keep only the top 500 genes

pca4 <- PcaGrid(t(test3)) # run PcaGrid on those top 500 genes
plot(pca4) ### This makes the most sence!!!!!!


## To be clear "Assaying a variance stabilized data set" means extracting the transformed values from a data set that has undergone a variance-stabilizing transformation, essentially retrieving the numerical values of the data after the transformation has been applied, allowing for further analysis where more even variance across different data points is desired. ## This came from Google AI
```

### DESeq2
```{r}
aplysia_dds2 <- DESeq(dds2)

## Age differences
# AUAL vs MUAL
summary(results(aplysia_dds2, contrast = c("group", "AUAL", "MUAL"), alpha = 0.05, lfcThreshold = 0))
AUAL_vs_MUALresSig <- subset(results(aplysia_dds2, contrast = c("group", "AUAL", "MUAL"), lfcThreshold = 0), padj < 0.05)
#############################################################################
## LFI in Mature
# MTAL vs MUAL
summary(results(aplysia_dds2, contrast = c("group", "MTAL", "MUAL"), alpha = 0.05, lfcThreshold = 0))
MTAL_vs_MUALresSig <- subset(results(aplysia_dds2, contrast = c("group", "MTAL", "MUAL"), lfcThreshold = 0), padj < 0.05)
#############################################################################
## LFI in Aged
# ATAL vs AUAL
summary(results(aplysia_dds2, contrast = c("group", "ATAL", "AUAL"), alpha = 0.05, lfcThreshold = 0))
ATAL_vs_AUALresSig <- subset(results(aplysia_dds2, contrast = c("group", "ATAL", "AUAL"), lfcThreshold = 0), padj < 0.05)
#############################################################################
## Age under CR
# AUCR vs MUCR
summary(results(aplysia_dds2, contrast = c("group", "AUCR", "MUCR"), alpha = 0.05, lfcThreshold = 0))
AUCR_vs_MUCRresSig <- subset(results(aplysia_dds2, contrast = c("group", "AUCR", "MUCR"), lfcThreshold = 0), padj < 0.05)
#############################################################################
## LFI in Mature CR
# MTCR vs MUCR
summary(results(aplysia_dds2, contrast = c("group", "MTCR", "MUCR"), alpha = 0.05, lfcThreshold = 0))
MTCR_vs_MUCRresSig <- subset(results(aplysia_dds2, contrast = c("group", "MTCR", "MUCR"), lfcThreshold = 0), padj < 0.05)
#############################################################################
## LFI in Aged CR
# ATCR vs AUCR
summary(results(aplysia_dds2, contrast = c("group", "ATCR", "AUCR"), alpha = 0.05, lfcThreshold = 0))
ATCR_vs_AUCRresSig <- subset(results(aplysia_dds2, contrast = c("group", "ATCR", "AUCR"), lfcThreshold = 0), padj < 0.05)



#########################################################################
## I will plot ALL DEGs
# first make a dataframe with all the DEGS
AUAL_vs_MUAL_DEGl2fc <- AUAL_vs_MUALresSig[2] %>% as.data.frame() %>%
  rownames_to_column() %>% set_names(c("Gene", "AUAL_vs_MUAL_l2fc"))
MTAL_vs_MUALDEGl2fc <- MTAL_vs_MUALresSig[2] %>% as.data.frame() %>%
  rownames_to_column() %>% set_names(c("Gene", "MTAL_vs_MUAL_l2fc"))
ATAL_vs_AUALDEGl2fc <- ATAL_vs_AUALresSig[2] %>% as.data.frame() %>%
  rownames_to_column() %>% set_names(c("Gene", "ATAL_vs_AUAL_l2fc"))
AUCR_vs_MUCRDEGl2fc <- AUCR_vs_MUCRresSig[2] %>% as.data.frame() %>%
  rownames_to_column() %>% set_names(c("Gene", "AUCR_vs_MUCR_l2fc"))
MTCR_vs_MUCRDEGl2fc <- MTCR_vs_MUCRresSig[2] %>% as.data.frame() %>%
  rownames_to_column() %>% set_names(c("Gene", "MTCR_vs_MUCR_l2fc"))
ATCR_vs_AUCRDEGl2fc <- ATCR_vs_AUCRresSig[2] %>% as.data.frame() %>%
  rownames_to_column() %>% set_names(c("Gene", "ATCR_vs_AUCR_l2fc"))

listInputs <- list("AUAL_vs_MUAL" = AUAL_vs_MUAL_DEGl2fc$Gene,
                   "MTAL_vs_MUAL" = MTAL_vs_MUALDEGl2fc$Gene,
                   "ATAL_vs_AUAL" = ATAL_vs_AUALDEGl2fc$Gene,
                   "AUCR_vs_MUCR" = AUCR_vs_MUCRDEGl2fc$Gene,
                   "MTCR_vs_MUCR" = MTCR_vs_MUCRDEGl2fc$Gene,
                   "ATCR_vs_AUCR" = ATCR_vs_AUCRDEGl2fc$Gene)
upset(fromList(listInputs), sets=c("AUAL_vs_MUAL", "MTAL_vs_MUAL", "ATAL_vs_AUAL", "AUCR_vs_MUCR", "MTCR_vs_MUCR", "ATCR_vs_AUCR"), order.by = "degree")
```

### Heirachical clustering heatmap
```{r}
# let's make a dataframe with all DEGs together from all comparisons
All_DEGs <- full_join(AUAL_vs_MUAL_DEGl2fc, MTAL_vs_MUALDEGl2fc, by="Gene")
All_DEGs <- full_join(All_DEGs, ATAL_vs_AUALDEGl2fc, by="Gene")
All_DEGs <- full_join(All_DEGs, AUCR_vs_MUCRDEGl2fc, by="Gene")
All_DEGs <- full_join(All_DEGs, MTCR_vs_MUCRDEGl2fc, by="Gene")
All_DEGs <- full_join(All_DEGs, ATCR_vs_AUCRDEGl2fc, by="Gene")
rownames(All_DEGs) <- All_DEGs$Gene
dim(All_DEGs) #94 DEGs, 6 comparisons and 1 column of Genes

rlog_out <- rlog(dds2, blind=F)
df <- as.data.frame(colData(dds2)[,c("Age","Diet","LFI")])
mat1 <- assay(rlog_out)[rownames(All_DEGs),]
colnames(mat1)
mat1.scaled <- t(apply(mat1, 1, scale))
colnames(mat1.scaled) <- colnames(mat1)
dim(mat1.scaled) #94 genes in 48 samples

pheatmap(mat1.scaled, cluster_rows = T, cluster_cols = T, annotation_col = df, show_rownames = F) #this everything with clusters

matrix.col.order_Diet <- c("810","816","828","830","836","837", #AL_U_M
                      "801","806","808","817","821","835", #AL_T_M
                      "863","865","874","877","878","881", #AL_U_A
                      "839","840","841","847","849","871", #AL_T_A
                      "805","809","815","823","824","829", #CR_U_M
                      "802","804","807","811","813","826", #CR_T_M
                      "872","873","875","876","879","880", #CR_U_A
                      "852","855","857","858","860","866") #CR_T_A
mat1.scaled.orderedByDiet <- mat1.scaled[,matrix.col.order_Diet]
pheatmap(mat1.scaled.orderedByDiet, cluster_rows = T, cluster_cols = F, annotation_col = df, show_rownames = F, show_colnames = F, cutree_rows = 5)

matrix.col.order_Age <- c("810","816","828","830","836","837", #AL_U_M
                      "801","806","808","817","821","835", #AL_T_M
                      "805","809","815","823","824","829", #CR_U_M
                      "802","804","807","811","813","826", #CR_T_M
                      "863","865","874","877","878","881", #AL_U_A
                      "839","840","841","847","849","871", #AL_T_A
                      "872","873","875","876","879","880", #CR_U_A
                      "852","855","857","858","860","866") #CR_T_A
mat1.scaled.orderedByAge <- mat1.scaled[,matrix.col.order_Age]
pheatmap(mat1.scaled.orderedByAge, cluster_rows = T, cluster_cols = F, annotation_col = df, show_rownames = F, show_colnames = F, cutree_rows = 5)

matrix.col.order_LFI <- c("810","816","828","830","836","837", #AL_U_M
                      "805","809","815","823","824","829", #CR_U_M
                      "863","865","874","877","878","881", #AL_U_A
                      "872","873","875","876","879","880", #CR_U_A
                      "801","806","808","817","821","835", #AL_T_M
                      "802","804","807","811","813","826", #CR_T_M
                      "839","840","841","847","849","871", #AL_T_A
                      "852","855","857","858","860","866") #CR_T_A
mat1.scaled.orderedByLFI <- mat1.scaled[,matrix.col.order_LFI]
pheatmap(mat1.scaled.orderedByLFI, cluster_rows = T, cluster_cols = F, annotation_col = df, show_rownames = F, show_colnames = T, cutree_rows = 9)
```

### making an Org.db
```{r}
# I want to make an Org.db for this species to use in GO
#devtools::install_version("dbplyr", version = "2.3.4")
library(dbplyr)
packageVersion("dbplyr")
# after you check to make sure the correct dbplyr version is installed you need to restart R to use it

## first will need to run the below code so that you do not encounter a connection problem with Ensemble then the AnnotationForge should work well
httr::set_config(httr::config(ssl_verifypeer = FALSE))
## Species: Plodia interpunctella
makeOrgPackageFromNCBI(version = "0.1",
                       author = "EC Randolph <erandolp@uab.edu>",
                       maintainer = "EC Randolph <erandolp@uab.edu>",
                       outputDir = "/Users/school/Documents/UAB/UAB009/Org.dbs/",
                       tax_id = "6500",
                       genus = "Aplysia",
                       species = "californica",
                       rebuildCache = F)
```
#_______________
# DUD exclusion comparisons
```{r}
## Explination in what is below

## The below section will be running the analyisis multiple ways through DEG detection and comparing. In this section we will remove the DUD sample and we will take out the outliers from the multiple ways used to detect them
## There will be 3 versions. 
### V1 will be elimiating the samples that were considered outliers based on visually where they fell in the first PCA plot using all samples and colored to Age. If samples fell outside the 99% CI elipsie for thier respective Age that sample was considered an outlier and removed
### V2 outliers are those samples thata are labled on the PcaGrid that fell above the line when using data from PC1 and PC2 as input
### V3 outliers will be the one sample that fell outside the line when using PcaGrid when all PCs were used as input data
## we will see how outlier elimination affects the amount of DEGs and then decide which to use I will make an additional comparsion by including a surogate variable analysis to each to see how that affects anything.

```

####### Import my data (tximport)
```{r }
## I will first make an object of my metadata and the eliminate samples from it before inporting the count data.
## This is the easiest way I have found to do this
sample_table = read.csv("/Users/school/Projects/CR/samples.txt", header = T, stringsAsFactors = T)
sample_table$P_SAV <- as.character(sample_table$P_SAV)
sample_table$Age <- factor(sample_table$Age, levels = c("M","A") )
sample_table$TRT <- factor(sample_table$TRT, levels = c("U", "T"))
names(sample_table)[names(sample_table) == "TRT"] <- "LFI"
row.names(sample_table)=sample_table$Sample_ID
```

### V1
```{r }
# This with samples that fell outside the 99% CI for Age
# These samples are 805, 809, 815 Mature, CR, Untrained (M,U,CR) -and- 857 Aged, CR, Trained (A,T,CR)
# and DUD sample 840 Aged, AL, Trained (A,T,CR)

# So now I will delete these samples from my metadata file and use the new metadata fine to import my quants and so forth
## "metadata_V1[!(metadata_V1$Sample_ID %in% c("809","810")),]" # this works in removing rows of outliers! Use this code!
metadata_V1 <- sample_table[!(sample_table$Sample_ID %in% c("805","809","815","857","840")),]

paste0('/Users/school/Projects/CR/quants/', 
       pull(metadata_V1, RNAseq_ID), '/quant.sf')
sample_files_V1 = paste0('/Users/school/Projects/CR/quants/',
                      pull(metadata_V1, RNAseq_ID), '/quant.sf')
names(sample_files_V1) = pull(metadata_V1, Sample_ID)
all(file.exists(sample_files_V1))
data.frame(thefiles = metadata_V1, doihave = file.exists(sample_files_V1))

# making a tx2gene for gene level quantification
txdb_aplysia <- makeTxDbFromGFF("/Users/school/Documents/UAB/UAB009/genome/AplCal3.0_GFF.gff",
                        format = "gff",
                        organism = "Aplysia californica")
k_aplysia <- keys(txdb_aplysia, keytype = "TXNAME")
tx2gene_aplysia <- AnnotationDbi::select(txdb_aplysia, k_aplysia, "GENEID", "TXNAME")
tx2gene_aplysia$TXNAME <- gsub("\\.[0-9]","",tx2gene_aplysia$TXNAME) 

count_data_V1 = tximport(files = sample_files_V1, importer=read.delim,
         type = "salmon",
         tx2gene = tx2gene_aplysia,
         ignoreTxVersion = T)

```

##### Model dds
```{r}
## as a comparison I will run this will all samples (ignoring any outlieres or DUDs)
dds_V1 <- DESeqDataSetFromTximport(count_data_V1,
                                colData = metadata_V1,
                                design = ~ Age + LFI + Diet)
colData(dds_V1)
dds_V1$group <- factor(paste0(dds_V1$Age, dds_V1$LFI, dds_V1$Diet))
design(dds_V1) <- ~group
colData(dds_V1)

## counts pre-filtering
dim(dds_V1) # starting with 21,034 transcripts in 43 samples
smallestGroupSize_V1 <- 3 # there are 3 samples in the smallest group
keep_V1 <- rowSums(counts(dds_V1) >= 1) >= smallestGroupSize_V1
dds2_V1 <- dds_V1[keep_V1,]
dim(dds2_V1) # 12,823 transcripts in 43 samples
```

##### PCA
```{r}
vsd_V1 <- vst(dds2_V1, blind = F)
pc12_V1 <- plotPCA(vsd_V1, intgroup = c("Age","LFI","Diet","group"), returnData = T) #PC1=31, PC2=09
ggplot(pc12_V1, aes(x=PC1, y=PC2, color=Age)) +
  geom_point(aes(fill=Age), 
       colour="black",pch=21, size=5)  +
  xlab(paste0("PC1: 31% varience")) +
  ylab(paste0("PC2: 9% variance")) +
  coord_fixed() +
  #scale_color_manual(labels = c("Younger","Aged"),
  #                   values = c("red","blue")) +
  #ggtitle("PCA with VST data") +
  stat_ellipse(level = 0.999) +
  theme_bw() +
  guides(fill=F) +
  theme(axis.text.x=element_text(size=18), 
        axis.title.x=element_text(size=18),
        axis.text.y=element_text(size=18), 
        axis.title.y=element_text(size=18),
        legend.title = element_text(size=16),
        legend.key.size = unit(0.6, 'cm'),
        legend.position = c(0.91, 0.91)) +
  scale_x_continuous(limits = c(-40, 40)) +
  scale_y_continuous(limits = c(-40, 40))
```

##### DESeq2
```{r}
dds2_V1.1 <- estimateSizeFactors(dds2_V1)
ddseq2_V1 <- DESeq(dds2_V1.1)

## Age differences
# AUAL vs MUAL
summary(results(ddseq2_V1, contrast = c("group", "AUAL", "MUAL"), alpha = 0.05, lfcThreshold = 0))
AUAL_vs_MUALresSig_V1 <- subset(results(ddseq2_V1, contrast = c("group", "AUAL", "MUAL"), lfcThreshold = 0), padj < 0.05)
#############################################################################
## LFI in Mature
# MTAL vs MUAL
summary(results(ddseq2_V1, contrast = c("group", "MTAL", "MUAL"), alpha = 0.05, lfcThreshold = 0))
MTAL_vs_MUALresSig_V1 <- subset(results(ddseq2_V1, contrast = c("group", "MTAL", "MUAL"), lfcThreshold = 0), padj < 0.05)
#############################################################################
## LFI in Aged
# ATAL vs AUAL
summary(results(ddseq2_V1, contrast = c("group", "ATAL", "AUAL"), alpha = 0.05, lfcThreshold = 0))
ATAL_vs_AUALresSig_V1 <- subset(results(ddseq2_V1, contrast = c("group", "ATAL", "AUAL"), lfcThreshold = 0), padj < 0.05)
#############################################################################
## Age under CR
# AUCR vs MUCR
summary(results(ddseq2_V1, contrast = c("group", "AUCR", "MUCR"), alpha = 0.05, lfcThreshold = 0))
AUCR_vs_MUCRresSig_V1 <- subset(results(ddseq2_V1, contrast = c("group", "AUCR", "MUCR"), lfcThreshold = 0), padj < 0.05)
#############################################################################
## LFI in Mature CR
# MTCR vs MUCR
summary(results(ddseq2_V1, contrast = c("group", "MTCR", "MUCR"), alpha = 0.05, lfcThreshold = 0))
MTCR_vs_MUCRresSig_V1 <- subset(results(ddseq2_V1, contrast = c("group", "MTCR", "MUCR"), lfcThreshold = 0), padj < 0.05)
#############################################################################
## LFI in Aged CR
# ATCR vs AUCR
summary(results(ddseq2_V1, contrast = c("group", "ATCR", "AUCR"), alpha = 0.05, lfcThreshold = 0))
ATCR_vs_AUCRresSig_V1 <- subset(results(ddseq2_V1, contrast = c("group", "ATCR", "AUCR"), lfcThreshold = 0), padj < 0.05)



#########################################################################
## I will plot ALL DEGs
# first make a dataframe with all the DEGS
AUAL_vs_MUAL_DEGl2fc_V1 <- AUAL_vs_MUALresSig_V1[2] %>% as.data.frame() %>%
  rownames_to_column() %>% set_names(c("Gene", "AUAL_vs_MUAL_l2fc_V1"))
MTAL_vs_MUALDEGl2fc_V1 <- MTAL_vs_MUALresSig_V1[2] %>% as.data.frame() %>%
  rownames_to_column() %>% set_names(c("Gene", "MTAL_vs_MUAL_l2fc_V1"))
ATAL_vs_AUALDEGl2fc_V1 <- ATAL_vs_AUALresSig_V1[2] %>% as.data.frame() %>%
  rownames_to_column() %>% set_names(c("Gene", "ATAL_vs_AUAL_l2fc_V1"))
AUCR_vs_MUCRDEGl2fc_V1 <- AUCR_vs_MUCRresSig_V1[2] %>% as.data.frame() %>%
  rownames_to_column() %>% set_names(c("Gene", "AUCR_vs_MUCR_l2fc_V1"))
MTCR_vs_MUCRDEGl2fc_V1 <- MTCR_vs_MUCRresSig_V1[2] %>% as.data.frame() %>%
  rownames_to_column() %>% set_names(c("Gene", "MTCR_vs_MUCR_l2fc_V1"))
ATCR_vs_AUCRDEGl2fc_V1 <- ATCR_vs_AUCRresSig_V1[2] %>% as.data.frame() %>%
  rownames_to_column() %>% set_names(c("Gene", "ATCR_vs_AUCR_l2fc_V1"))

listInputs_V1 <- list("AUAL_vs_MUAL_V1" = AUAL_vs_MUAL_DEGl2fc_V1$Gene,
                   "MTAL_vs_MUAL_V1" = MTAL_vs_MUALDEGl2fc_V1$Gene,
                   "ATAL_vs_AUAL_V1" = ATAL_vs_AUALDEGl2fc_V1$Gene,
                   "AUCR_vs_MUCR_V1" = AUCR_vs_MUCRDEGl2fc_V1$Gene,
                   "MTCR_vs_MUCR_V1" = MTCR_vs_MUCRDEGl2fc_V1$Gene,
                   "ATCR_vs_AUCR_V1" = ATCR_vs_AUCRDEGl2fc_V1$Gene)
upset(fromList(listInputs_V1), sets=c("AUAL_vs_MUAL_V1", "MTAL_vs_MUAL_V1", "ATAL_vs_AUAL_V1", "AUCR_vs_MUCR_V1", "MTCR_vs_MUCR_V1", "ATCR_vs_AUCR_V1"), order.by = "degree")
```

##### Heirachical clustering heatmap
```{r}
# let's make a dataframe with all DEGs together from all comparisons
All_DEGs_V1 <- full_join(AUAL_vs_MUAL_DEGl2fc_V1, MTAL_vs_MUALDEGl2fc_V1, by="Gene")
All_DEGs_V1 <- full_join(All_DEGs_V1, ATAL_vs_AUALDEGl2fc_V1, by="Gene")
All_DEGs_V1 <- full_join(All_DEGs_V1, AUCR_vs_MUCRDEGl2fc_V1, by="Gene")
All_DEGs_V1 <- full_join(All_DEGs_V1, MTCR_vs_MUCRDEGl2fc_V1, by="Gene")
All_DEGs_V1 <- full_join(All_DEGs_V1, ATCR_vs_AUCRDEGl2fc_V1, by="Gene")
rownames(All_DEGs_V1) <- All_DEGs_V1$Gene
dim(All_DEGs_V1) #34 DEGs, 6 comparisons and 1 column of Genes

rlog_out_V1 <- rlog(dds2_V1.1, blind=F)
df_V1 <- as.data.frame(colData(dds2_V1.1)[,c("Age","Diet","LFI")])
mat1_V1 <- assay(rlog_out_V1)[rownames(All_DEGs_V1),]
colnames(mat1_V1)
mat1.scaled_V1 <- t(apply(mat1_V1, 1, scale))
colnames(mat1.scaled_V1) <- colnames(mat1_V1)
dim(mat1.scaled_V1) #34 genes in 43 samples

pheatmap(mat1.scaled_V1, cluster_rows = T, cluster_cols = T, annotation_col = df_V1, show_rownames = F) #this everything with clusters

matrix.col.order_Diet_V1 <- c("810","816","828","830","836","837", #AL_U_M
                      "801","806","808","817","821","835", #AL_T_M
                      "863","865","874","877","878","881", #AL_U_A
                      "839","841","847","849","871", #AL_T_A
                      "823","824","829", #CR_U_M
                      "802","804","807","811","813","826", #CR_T_M
                      "872","873","875","876","879","880", #CR_U_A
                      "852","855","858","860","866") #CR_T_A
mat1.scaled.orderedByDiet_V1 <- mat1.scaled_V1[,matrix.col.order_Diet_V1]
pheatmap(mat1.scaled.orderedByDiet_V1, cluster_rows = T, cluster_cols = F, annotation_col = df_V1, show_rownames = F, show_colnames = F, cutree_rows = 7)

matrix.col.order_Age_V1 <- c("810","816","828","830","836","837", #AL_U_M
                      "801","806","808","817","821","835", #AL_T_M
                      "823","824","829", #CR_U_M
                      "802","804","807","811","813","826", #CR_T_M
                      "863","865","874","877","878","881", #AL_U_A
                      "839","841","847","849","871", #AL_T_A
                      "872","873","875","876","879","880", #CR_U_A
                      "852","855","858","860","866") #CR_T_A
mat1.scaled.orderedByAge_V1 <- mat1.scaled_V1[,matrix.col.order_Age_V1]
pheatmap(mat1.scaled.orderedByAge_V1, cluster_rows = T, cluster_cols = F, annotation_col = df_V1, show_rownames = F, show_colnames = F, cutree_rows = 7)

matrix.col.order_LFI_V1 <- c("810","816","828","830","836","837", #AL_U_M
                      "823","824","829", #CR_U_M
                      "863","865","874","877","878","881", #AL_U_A
                      "872","873","875","876","879","880", #CR_U_A
                      "801","806","808","817","821","835", #AL_T_M
                      "802","804","807","811","813","826", #CR_T_M
                      "839","841","847","849","871", #AL_T_A
                      "852","855","858","860","866") #CR_T_A
mat1.scaled.orderedByLFI_V1 <- mat1.scaled_V1[,matrix.col.order_LFI_V1]
pheatmap(mat1.scaled.orderedByLFI_V1, cluster_rows = T, cluster_cols = F, annotation_col = df_V1, show_rownames = F, show_colnames = T, cutree_rows = 7)

## All the heatmaps above did not cluster the samples but I chose where the samples fell
## Here is one that clusters the samples
pheatmap(mat1.scaled.orderedByLFI_V1, cluster_rows = T, cluster_cols = T, annotation_col = df_V1, show_rownames = F, show_colnames = T, cutree_rows = 7)

```

### V1_w/SVA
##### SVA analysis
```{r}
mod_V1 <- model.matrix(~ Age + LFI + Diet, data=metadata_V1)
mod0_V1 <- model.matrix(~1, data=metadata_V1)

dat_V1 <- counts(dds2_V1.1, normalized=T)
num.sv(dat_V1, mod_V1, method = "be") #this tells me how many surrogate variables there are 
## this said there were zero 0 surrogate variables 
## Still going to try this with 2

#za2 <- za[rowSums(za >= 7) >= 4,] #had to filter the data because too many zeros messed up the operation

svobl_V1 <- svaseq(dat_V1, mod_V1, mod0_V1, n.sv = 2)
metadata_V1$sv1=svobl_V1$sv[,1]
metadata_V1$sv2=svobl_V1$sv[,2]

## now that I have located surrogate variables all I have to do is incorporate them into my DESeq2 model matrix - I do not need to eliminate them from my raw data. This has the possibility of over correcting. 


#Quick check at variance
#plot(svobl$sv[,],pch=19,col="blue")
#plot(as.matrix(za2[,33]),pch=19,col="blue") #this has nothing to do with sav, this is visualizing the TPMs of genes in samples within the matrix. You can call each sample by selecing the column number and putting it after the comma
# ggplot(data=sample_table_minusOutliers, aes( x=Age, y=sv2)) +
#    geom_point()
```

##### Model dds
```{r}
## as a comparison I will run this will all samples (ignoring any outlieres or DUDs)
dds_V1.2 <- DESeqDataSetFromTximport(count_data_V1,
                                colData = metadata_V1,
                                design = ~ sv1 + sv2 + Age + LFI + Diet)
colData(dds_V1.2)
dds_V1.2$group <- factor(paste0(dds_V1.2$Age, dds_V1.2$LFI, dds_V1.2$Diet))
design(dds_V1.2) <- ~group
colData(dds_V1.2)

## counts pre-filtering
dim(dds_V1.2) # starting with 21,034 transcripts in 43 samples
smallestGroupSize_V1.2 <- 3 # there are 3 samples in the smallest group
keep_V1.2 <- rowSums(counts(dds_V1.2) >= 1) >= smallestGroupSize_V1.2
dds2_V1.2 <- dds_V1.2[keep_V1.2,]
dim(dds2_V1.2) # 12,823 transcripts in 43 samples
```

##### PCA
```{r}
vsd_V1.2 <- vst(dds2_V1.2)
pc12_V1.2 <- plotPCA(vsd_V1.2, intgroup = c("Age","LFI","Diet","group"), returnData = T) #PC1=30, PC2=09
ggplot(pc12_V1.2, aes(x=PC1, y=PC2, color=Age)) +
  geom_point(aes(fill=Age), 
       colour="black",pch=21, size=5)  +
  xlab(paste0("PC1: 30% varience")) +
  ylab(paste0("PC2: 9% variance")) +
  coord_fixed() +
  #scale_color_manual(labels = c("Younger","Aged"),
  #                   values = c("red","blue")) +
  #ggtitle("PCA with VST data") +
  stat_ellipse(level = 0.999) +
  theme_bw() +
  guides(fill=F) +
  theme(axis.text.x=element_text(size=18), 
        axis.title.x=element_text(size=18),
        axis.text.y=element_text(size=18), 
        axis.title.y=element_text(size=18),
        legend.title = element_text(size=16),
        legend.key.size = unit(0.6, 'cm'),
        legend.position = c(0.91, 0.91)) +
  scale_x_continuous(limits = c(-40, 40)) +
  scale_y_continuous(limits = c(-40, 40))
```

##### DESeq2
```{r}
dds2_V1.3 <- estimateSizeFactors(dds2_V1.2)
ddseq2_V1.2 <- DESeq(dds2_V1.3)

## Age differences
# AUAL vs MUAL
summary(results(ddseq2_V1.2, contrast = c("group", "AUAL", "MUAL"), alpha = 0.05, lfcThreshold = 0))
AUAL_vs_MUALresSig_V1.2 <- subset(results(ddseq2_V1.2, contrast = c("group", "AUAL", "MUAL"), lfcThreshold = 0), padj < 0.05)
#############################################################################
## LFI in Mature
# MTAL vs MUAL
summary(results(ddseq2_V1.2, contrast = c("group", "MTAL", "MUAL"), alpha = 0.05, lfcThreshold = 0))
MTAL_vs_MUALresSig_V1.2 <- subset(results(ddseq2_V1.2, contrast = c("group", "MTAL", "MUAL"), lfcThreshold = 0), padj < 0.05)
#############################################################################
## LFI in Aged
# ATAL vs AUAL
summary(results(ddseq2_V1.2, contrast = c("group", "ATAL", "AUAL"), alpha = 0.05, lfcThreshold = 0))
ATAL_vs_AUALresSig_V1.2 <- subset(results(ddseq2_V1.2, contrast = c("group", "ATAL", "AUAL"), lfcThreshold = 0), padj < 0.05)
#############################################################################
## Age under CR
# AUCR vs MUCR
summary(results(ddseq2_V1.2, contrast = c("group", "AUCR", "MUCR"), alpha = 0.05, lfcThreshold = 0))
AUCR_vs_MUCRresSig_V1.2 <- subset(results(ddseq2_V1.2, contrast = c("group", "AUCR", "MUCR"), lfcThreshold = 0), padj < 0.05)
#############################################################################
## LFI in Mature CR
# MTCR vs MUCR
summary(results(ddseq2_V1.2, contrast = c("group", "MTCR", "MUCR"), alpha = 0.05, lfcThreshold = 0))
MTCR_vs_MUCRresSig_V1.2 <- subset(results(ddseq2_V1.2, contrast = c("group", "MTCR", "MUCR"), lfcThreshold = 0), padj < 0.05)
#############################################################################
## LFI in Aged CR
# ATCR vs AUCR
summary(results(ddseq2_V1.2, contrast = c("group", "ATCR", "AUCR"), alpha = 0.05, lfcThreshold = 0))
ATCR_vs_AUCRresSig_V1.2 <- subset(results(ddseq2_V1.2, contrast = c("group", "ATCR", "AUCR"), lfcThreshold = 0), padj < 0.05)



#########################################################################
## I will plot ALL DEGs
# first make a dataframe with all the DEGS
AUAL_vs_MUAL_DEGl2fc_V1.2 <- AUAL_vs_MUALresSig_V1.2[2] %>% as.data.frame() %>%
  rownames_to_column() %>% set_names(c("Gene", "AUAL_vs_MUAL_l2fc_V1.2"))
MTAL_vs_MUALDEGl2fc_V1.2 <- MTAL_vs_MUALresSig_V1.2[2] %>% as.data.frame() %>%
  rownames_to_column() %>% set_names(c("Gene", "MTAL_vs_MUAL_l2fc_V1.2"))
ATAL_vs_AUALDEGl2fc_V1.2 <- ATAL_vs_AUALresSig_V1.2[2] %>% as.data.frame() %>%
  rownames_to_column() %>% set_names(c("Gene", "ATAL_vs_AUAL_l2fc_V1.2"))
AUCR_vs_MUCRDEGl2fc_V1.2 <- AUCR_vs_MUCRresSig_V1.2[2] %>% as.data.frame() %>%
  rownames_to_column() %>% set_names(c("Gene", "AUCR_vs_MUCR_l2fc_V1.2"))
MTCR_vs_MUCRDEGl2fc_V1.2 <- MTCR_vs_MUCRresSig_V1.2[2] %>% as.data.frame() %>%
  rownames_to_column() %>% set_names(c("Gene", "MTCR_vs_MUCR_l2fc_V1.2"))
ATCR_vs_AUCRDEGl2fc_V1.2 <- ATCR_vs_AUCRresSig_V1.2[2] %>% as.data.frame() %>%
  rownames_to_column() %>% set_names(c("Gene", "ATCR_vs_AUCR_l2fc_V1.2"))

listInputs_V1.2 <- list("AUAL_vs_MUAL_V1.2" = AUAL_vs_MUAL_DEGl2fc_V1.2$Gene,
                   "MTAL_vs_MUAL_V1.2" = MTAL_vs_MUALDEGl2fc_V1.2$Gene,
                   "ATAL_vs_AUAL_V1.2" = ATAL_vs_AUALDEGl2fc_V1.2$Gene,
                   "AUCR_vs_MUCR_V1.2" = AUCR_vs_MUCRDEGl2fc_V1.2$Gene,
                   "MTCR_vs_MUCR_V1.2" = MTCR_vs_MUCRDEGl2fc_V1.2$Gene,
                   "ATCR_vs_AUCR_V1.2" = ATCR_vs_AUCRDEGl2fc_V1.2$Gene)
upset(fromList(listInputs_V1.2), sets=c("AUAL_vs_MUAL_V1.2", "MTAL_vs_MUAL_V1.2", "ATAL_vs_AUAL_V1.2", "AUCR_vs_MUCR_V1.2", "MTCR_vs_MUCR_V1.2", "ATCR_vs_AUCR_V1.2"), order.by = "degree")
```

##### Heirachical clustering heatmap
```{r}
# let's make a dataframe with all DEGs together from all comparisons
All_DEGs_V1.2 <- full_join(AUAL_vs_MUAL_DEGl2fc_V1.2, MTAL_vs_MUALDEGl2fc_V1.2, by="Gene")
All_DEGs_V1.2 <- full_join(All_DEGs_V1.2, ATAL_vs_AUALDEGl2fc_V1.2, by="Gene")
All_DEGs_V1.2 <- full_join(All_DEGs_V1.2, AUCR_vs_MUCRDEGl2fc_V1.2, by="Gene")
All_DEGs_V1.2 <- full_join(All_DEGs_V1.2, MTCR_vs_MUCRDEGl2fc_V1.2, by="Gene")
All_DEGs_V1.2 <- full_join(All_DEGs_V1.2, ATCR_vs_AUCRDEGl2fc_V1.2, by="Gene")
rownames(All_DEGs_V1.2) <- All_DEGs_V1.2$Gene
dim(All_DEGs_V1.2) #34 DEGs, 6 comparisons and 1 column of Genes

rlog_out_V1.2 <- rlog(dds2_V1.3, blind=F)
df_V1.2 <- as.data.frame(colData(dds2_V1.3)[,c("Age","Diet","LFI")])
mat1_V1.2 <- assay(rlog_out_V1.2)[rownames(All_DEGs_V1.2),]
colnames(mat1_V1.2)
mat1.scaled_V1.2 <- t(apply(mat1_V1.2, 1, scale))
colnames(mat1.scaled_V1.2) <- colnames(mat1_V1.2)
dim(mat1.scaled_V1.2) #34 genes in 43 samples

pheatmap(mat1.scaled_V1.2, cluster_rows = T, cluster_cols = T, annotation_col = df_V1.2, show_rownames = F) #this everything with clusters

matrix.col.order_Diet_V1.2 <- c("810","816","828","830","836","837", #AL_U_M
                      "801","806","808","817","821","835", #AL_T_M
                      "863","865","874","877","878","881", #AL_U_A
                      "839","841","847","849","871", #AL_T_A
                      "823","824","829", #CR_U_M
                      "802","804","807","811","813","826", #CR_T_M
                      "872","873","875","876","879","880", #CR_U_A
                      "852","855","858","860","866") #CR_T_A
mat1.scaled.orderedByDiet_V1.2 <- mat1.scaled_V1.2[,matrix.col.order_Diet_V1.2]
pheatmap(mat1.scaled.orderedByDiet_V1.2, cluster_rows = T, cluster_cols = F, annotation_col = df_V1.2, show_rownames = F, show_colnames = F, cutree_rows = 5)

matrix.col.order_Age_V1.2 <- c("810","816","828","830","836","837", #AL_U_M
                      "801","806","808","817","821","835", #AL_T_M
                      "823","824","829", #CR_U_M
                      "802","804","807","811","813","826", #CR_T_M
                      "863","865","874","877","878","881", #AL_U_A
                      "839","841","847","849","871", #AL_T_A
                      "872","873","875","876","879","880", #CR_U_A
                      "852","855","858","860","866") #CR_T_A
mat1.scaled.orderedByAge_V1.2 <- mat1.scaled_V1.2[,matrix.col.order_Age_V1.2]
pheatmap(mat1.scaled.orderedByAge_V1.2, cluster_rows = T, cluster_cols = F, annotation_col = df_V1.2, show_rownames = F, show_colnames = F, cutree_rows = 5)

matrix.col.order_LFI_V1.2 <- c("810","816","828","830","836","837", #AL_U_M
                      "823","824","829", #CR_U_M
                      "863","865","874","877","878","881", #AL_U_A
                      "872","873","875","876","879","880", #CR_U_A
                      "801","806","808","817","821","835", #AL_T_M
                      "802","804","807","811","813","826", #CR_T_M
                      "839","841","847","849","871", #AL_T_A
                      "852","855","858","860","866") #CR_T_A
mat1.scaled.orderedByLFI_V1.2 <- mat1.scaled_V1.2[,matrix.col.order_LFI_V1.2]
pheatmap(mat1.scaled.orderedByLFI_V1.2, cluster_rows = T, cluster_cols = F, annotation_col = df_V1.2, show_rownames = F, show_colnames = T, cutree_rows = 9)

## All the heatmaps above did not cluster the samples but I chose where the samples fell
## Here is one that clusters the samples
pheatmap(mat1.scaled.orderedByLFI_V1.2, cluster_rows = T, cluster_cols = T, annotation_col = df_V1.2, show_rownames = F, show_colnames = T, cutree_rows = 9)

```

### V2
```{r }
# This with samples that fell outside the 99% CI for Age
# These samples are 805, 809, 815 Mature, CR, Untrained (M,U,CR)
# and DUD sample 840 Aged, AL, Trained (A,T,CR)

# So now I will delete these samples from my metadata file and use the new metadata fine to import my quants and so forth
metadata_V2 <- sample_table[!(sample_table$Sample_ID %in% c("805","809","815","840")),]

paste0('/Users/school/Projects/CR/quants/', 
       pull(metadata_V2, RNAseq_ID), '/quant.sf')
sample_files_V2 = paste0('/Users/school/Projects/CR/quants/',
                      pull(metadata_V2, RNAseq_ID), '/quant.sf')
names(sample_files_V2) = pull(metadata_V2, Sample_ID)
all(file.exists(sample_files_V2))
data.frame(thefiles = metadata_V2, doihave = file.exists(sample_files_V2))



count_data_V2 = tximport(files = sample_files_V2, importer=read.delim,
         type = "salmon",
         tx2gene = tx2gene_aplysia,
         ignoreTxVersion = T)

```

##### Model dds
```{r}
## as a comparison I will run this will all samples (ignoring any outlieres or DUDs)
dds_V2 <- DESeqDataSetFromTximport(count_data_V2,
                                colData = metadata_V2,
                                design = ~ Age + LFI + Diet)
colData(dds_V2)
dds_V2$group <- factor(paste0(dds_V2$Age, dds_V2$LFI, dds_V2$Diet))
design(dds_V2) <- ~group
colData(dds_V2)

## counts pre-filtering
dim(dds_V2) # starting with 21,034 transcripts in 44 samples
smallestGroupSize_V2 <- 3 # there are 3 samples in the smallest group
keep_V2 <- rowSums(counts(dds_V2) >= 1) >= smallestGroupSize_V2
dds2_V2 <- dds_V2[keep_V2,]
dim(dds2_V2) # 12,855 transcripts in 44 samples
```

##### PCA
```{r}
vsd_V2 <- vst(dds2_V2, blind = F)
pc12_V2 <- plotPCA(vsd_V2, intgroup = c("Age","LFI","Diet","group"), returnData = T) #PC1=29, PC2=09
ggplot(pc12_V2, aes(x=PC1, y=PC2, color=LFI)) +
  geom_point(aes(fill=LFI), 
       colour="black",pch=21, size=5)  +
  xlab(paste0("PC1: 29% varience")) +
  ylab(paste0("PC2: 9% variance")) +
  coord_fixed() +
  #scale_color_manual(labels = c("Younger","Aged"),
  #                   values = c("red","blue")) +
  #ggtitle("PCA with VST data") +
  stat_ellipse(level = 0.999) +
  theme_bw() +
  guides(fill=F) +
  theme(axis.text.x=element_text(size=18), 
        axis.title.x=element_text(size=18),
        axis.text.y=element_text(size=18), 
        axis.title.y=element_text(size=18),
        legend.title = element_text(size=16),
        legend.key.size = unit(0.6, 'cm'),
        legend.position = c(0.91, 0.91)) +
  scale_x_continuous(limits = c(-40, 40)) +
  scale_y_continuous(limits = c(-40, 40))
```

##### DESeq2
```{r}
dds2_V2.1 <- estimateSizeFactors(dds2_V2)
ddseq2_V2 <- DESeq(dds2_V2.1)

## Age differences
# AUAL vs MUAL
summary(results(ddseq2_V2, contrast = c("group", "AUAL", "MUAL"), alpha = 0.05, lfcThreshold = 0))
AUAL_vs_MUALresSig_V2 <- subset(results(ddseq2_V2, contrast = c("group", "AUAL", "MUAL"), lfcThreshold = 0), padj < 0.05)
#############################################################################
## LFI in Mature
# MTAL vs MUAL
summary(results(ddseq2_V2, contrast = c("group", "MTAL", "MUAL"), alpha = 0.05, lfcThreshold = 0))
MTAL_vs_MUALresSig_V2 <- subset(results(ddseq2_V2, contrast = c("group", "MTAL", "MUAL"), lfcThreshold = 0), padj < 0.05)
#############################################################################
## LFI in Aged
# ATAL vs AUAL
summary(results(ddseq2_V2, contrast = c("group", "ATAL", "AUAL"), alpha = 0.05, lfcThreshold = 0))
ATAL_vs_AUALresSig_V2 <- subset(results(ddseq2_V2, contrast = c("group", "ATAL", "AUAL"), lfcThreshold = 0), padj < 0.05)
#############################################################################
## Age under CR
# AUCR vs MUCR
summary(results(ddseq2_V2, contrast = c("group", "AUCR", "MUCR"), alpha = 0.05, lfcThreshold = 0))
AUCR_vs_MUCRresSig_V2 <- subset(results(ddseq2_V2, contrast = c("group", "AUCR", "MUCR"), lfcThreshold = 0), padj < 0.05)
#############################################################################
## LFI in Mature CR
# MTCR vs MUCR
summary(results(ddseq2_V2, contrast = c("group", "MTCR", "MUCR"), alpha = 0.05, lfcThreshold = 0))
MTCR_vs_MUCRresSig_V2 <- subset(results(ddseq2_V2, contrast = c("group", "MTCR", "MUCR"), lfcThreshold = 0), padj < 0.05)
#############################################################################
## LFI in Aged CR
# ATCR vs AUCR
summary(results(ddseq2_V2, contrast = c("group", "ATCR", "AUCR"), alpha = 0.05, lfcThreshold = 0))
ATCR_vs_AUCRresSig_V2 <- subset(results(ddseq2_V2, contrast = c("group", "ATCR", "AUCR"), lfcThreshold = 0), padj < 0.05)



#########################################################################
## I will plot ALL DEGs
# first make a dataframe with all the DEGS
AUAL_vs_MUAL_DEGl2fc_V2 <- AUAL_vs_MUALresSig_V2[2] %>% as.data.frame() %>%
  rownames_to_column() %>% set_names(c("Gene", "AUAL_vs_MUAL_l2fc_V2"))
MTAL_vs_MUALDEGl2fc_V2 <- MTAL_vs_MUALresSig_V2[2] %>% as.data.frame() %>%
  rownames_to_column() %>% set_names(c("Gene", "MTAL_vs_MUAL_l2fc_V2"))
ATAL_vs_AUALDEGl2fc_V2 <- ATAL_vs_AUALresSig_V2[2] %>% as.data.frame() %>%
  rownames_to_column() %>% set_names(c("Gene", "ATAL_vs_AUAL_l2fc_V2"))
AUCR_vs_MUCRDEGl2fc_V2 <- AUCR_vs_MUCRresSig_V2[2] %>% as.data.frame() %>%
  rownames_to_column() %>% set_names(c("Gene", "AUCR_vs_MUCR_l2fc_V2"))
MTCR_vs_MUCRDEGl2fc_V2 <- MTCR_vs_MUCRresSig_V2[2] %>% as.data.frame() %>%
  rownames_to_column() %>% set_names(c("Gene", "MTCR_vs_MUCR_l2fc_V2"))
ATCR_vs_AUCRDEGl2fc_V2 <- ATCR_vs_AUCRresSig_V2[2] %>% as.data.frame() %>%
  rownames_to_column() %>% set_names(c("Gene", "ATCR_vs_AUCR_l2fc_V2"))

listInputs_V2 <- list("AUAL_vs_MUAL_V2" = AUAL_vs_MUAL_DEGl2fc_V2$Gene,
                   "MTAL_vs_MUAL_V2" = MTAL_vs_MUALDEGl2fc_V2$Gene,
                   "ATAL_vs_AUAL_V2" = ATAL_vs_AUALDEGl2fc_V2$Gene,
                   "AUCR_vs_MUCR_V2" = AUCR_vs_MUCRDEGl2fc_V2$Gene,
                   "MTCR_vs_MUCR_V2" = MTCR_vs_MUCRDEGl2fc_V2$Gene,
                   "ATCR_vs_AUCR_V2" = ATCR_vs_AUCRDEGl2fc_V2$Gene)
upset(fromList(listInputs_V2), sets=c("AUAL_vs_MUAL_V2", "MTAL_vs_MUAL_V2", "ATAL_vs_AUAL_V2", "AUCR_vs_MUCR_V2", "MTCR_vs_MUCR_V2", "ATCR_vs_AUCR_V2"), order.by = "degree")
```

##### Heirachical clustering heatmap
```{r}
# let's make a dataframe with all DEGs together from all comparisons
All_DEGs_V2 <- full_join(AUAL_vs_MUAL_DEGl2fc_V2, MTAL_vs_MUALDEGl2fc_V2, by="Gene")
All_DEGs_V2 <- full_join(All_DEGs_V2, ATAL_vs_AUALDEGl2fc_V2, by="Gene")
All_DEGs_V2 <- full_join(All_DEGs_V2, AUCR_vs_MUCRDEGl2fc_V2, by="Gene")
All_DEGs_V2 <- full_join(All_DEGs_V2, MTCR_vs_MUCRDEGl2fc_V2, by="Gene")
All_DEGs_V2 <- full_join(All_DEGs_V2, ATCR_vs_AUCRDEGl2fc_V2, by="Gene")
rownames(All_DEGs_V2) <- All_DEGs_V2$Gene
dim(All_DEGs_V2) # 43 DEGs, 6 comparisons and 1 column of Genes

rlog_out_V2 <- rlog(dds2_V2.1, blind=F)
df_V2 <- as.data.frame(colData(dds2_V2.1)[,c("Age","Diet","LFI")])
mat1_V2 <- assay(rlog_out_V2)[rownames(All_DEGs_V2),]
colnames(mat1_V2)
mat1.scaled_V2 <- t(apply(mat1_V2, 1, scale))
colnames(mat1.scaled_V2) <- colnames(mat1_V2)
dim(mat1.scaled_V2) #34 genes in 43 samples

pheatmap(mat1.scaled_V2, cluster_rows = T, cluster_cols = T, annotation_col = df_V2, show_rownames = F) #this everything with clusters

matrix.col.order_Diet_V2 <- c("810","816","828","830","836","837", #AL_U_M
                      "801","806","808","817","821","835", #AL_T_M
                      "863","865","874","877","878","881", #AL_U_A
                      "839","841","847","849","871", #AL_T_A
                      "823","824","829", #CR_U_M
                      "802","804","807","811","813","826", #CR_T_M
                      "872","873","875","876","879","880", #CR_U_A
                      "852","855","857","858","860","866") #CR_T_A
mat1.scaled.orderedByDiet_V2 <- mat1.scaled_V2[,matrix.col.order_Diet_V2]
pheatmap(mat1.scaled.orderedByDiet_V2, cluster_rows = T, cluster_cols = F, annotation_col = df_V2, show_rownames = F, show_colnames = F, cutree_rows = 7)

matrix.col.order_Age_V2 <- c("810","816","828","830","836","837", #AL_U_M
                      "801","806","808","817","821","835", #AL_T_M
                      "823","824","829", #CR_U_M
                      "802","804","807","811","813","826", #CR_T_M
                      "863","865","874","877","878","881", #AL_U_A
                      "839","841","847","849","871", #AL_T_A
                      "872","873","875","876","879","880", #CR_U_A
                      "852","855","857","858","860","866") #CR_T_A
mat1.scaled.orderedByAge_V2 <- mat1.scaled_V2[,matrix.col.order_Age_V2]
pheatmap(mat1.scaled.orderedByAge_V2, cluster_rows = T, cluster_cols = F, annotation_col = df_V2, show_rownames = F, show_colnames = F, cutree_rows = 7)

matrix.col.order_LFI_V2 <- c("810","816","828","830","836","837", #AL_U_M
                      "823","824","829", #CR_U_M
                      "863","865","874","877","878","881", #AL_U_A
                      "872","873","875","876","879","880", #CR_U_A
                      "801","806","808","817","821","835", #AL_T_M
                      "802","804","807","811","813","826", #CR_T_M
                      "839","841","847","849","871", #AL_T_A
                      "852","855","857","858","860","866") #CR_T_A
mat1.scaled.orderedByLFI_V2 <- mat1.scaled_V2[,matrix.col.order_LFI_V2]
pheatmap(mat1.scaled.orderedByLFI_V2, cluster_rows = T, cluster_cols = F, annotation_col = df_V2, show_rownames = F, show_colnames = T, cutree_rows = 7)

## All the heatmaps above did not cluster the samples but I chose where the samples fell
## Here is one that clusters the samples
pheatmap(mat1.scaled.orderedByLFI_V2, cluster_rows = T, cluster_cols = T, annotation_col = df_V2, show_rownames = F, show_colnames = T, cutree_rows = 7)

```

### V2_w/SVA
##### SVA analysis
```{r}
mod_V2 <- model.matrix(~ Age + LFI + Diet, data=metadata_V2)
mod0_V2 <- model.matrix(~1, data=metadata_V2)

dat_V2 <- counts(dds2_V2.1, normalized=T)
num.sv(dat_V2, mod_V2, method = "be") #this tells me how many surrogate variables there are 
## this said there were zero 0 surrogate variables 
## Still going to try this with 2

#za2 <- za[rowSums(za >= 7) >= 4,] #had to filter the data because too many zeros messed up the operation

svobl_V2 <- svaseq(dat_V2, mod_V2, mod0_V2, n.sv = 2)
metadata_V2$sV2=svobl_V2$sv[,1]
metadata_V2$sv2=svobl_V2$sv[,2]

## now that I have located surrogate variables all I have to do is incorporate them into my DESeq2 model matrix - I do not need to eliminate them from my raw data. This has the possibility of over correcting. 


#Quick check at variance
#plot(svobl$sv[,],pch=19,col="blue")
#plot(as.matrix(za2[,33]),pch=19,col="blue") #this has nothing to do with sav, this is visualizing the TPMs of genes in samples within the matrix. You can call each sample by selecing the column number and putting it after the comma
# ggplot(data=sample_table_minusOutliers, aes( x=Age, y=sv2)) +
#    geom_point()
```

##### Model dds
```{r}
## as a comparison I will run this will all samples (ignoring any outlieres or DUDs)
dds_V2.2 <- DESeqDataSetFromTximport(count_data_V2,
                                colData = metadata_V2,
                                design = ~ sV2 + sv2 + Age + LFI + Diet)
colData(dds_V2.2)
dds_V2.2$group <- factor(paste0(dds_V2.2$Age, dds_V2.2$LFI, dds_V2.2$Diet))
design(dds_V2.2) <- ~group
colData(dds_V2.2)

## counts pre-filtering
dim(dds_V2.2) # starting with 21,034 transcripts in 44 samples
smallestGroupSize_V2.2 <- 3 # there are 3 samples in the smallest group
keep_V2.2 <- rowSums(counts(dds_V2.2) >= 1) >= smallestGroupSize_V2.2
dds2_V2.2 <- dds_V2.2[keep_V2.2,]
dim(dds2_V2.2) # 12,855 transcripts in 44 samples
```

##### PCA
```{r}
vsd_V2.2 <- vst(dds2_V2.2)
pc12_V2.2 <- plotPCA(vsd_V2.2, intgroup = c("Age","LFI","Diet","group"), returnData = T) #PC1=28, PC2=09
ggplot(pc12_V2.2, aes(x=PC1, y=PC2, color=Age)) +
  geom_point(aes(fill=Age), 
       colour="black",pch=21, size=5)  +
  xlab(paste0("PC1: 28% varience")) +
  ylab(paste0("PC2: 9% variance")) +
  coord_fixed() +
  #scale_color_manual(labels = c("Younger","Aged"),
  #                   values = c("red","blue")) +
  #ggtitle("PCA with VST data") +
  stat_ellipse(level = 0.999) +
  theme_bw() +
  guides(fill=F) +
  theme(axis.text.x=element_text(size=18), 
        axis.title.x=element_text(size=18),
        axis.text.y=element_text(size=18), 
        axis.title.y=element_text(size=18),
        legend.title = element_text(size=16),
        legend.key.size = unit(0.6, 'cm'),
        legend.position = c(0.91, 0.91)) +
  scale_x_continuous(limits = c(-40, 40)) +
  scale_y_continuous(limits = c(-40, 40))
```

##### DESeq2
```{r}
dds2_V2.3 <- estimateSizeFactors(dds2_V2.2)
ddseq2_V2.2 <- DESeq(dds2_V2.3)

## Age differences
# AUAL vs MUAL
summary(results(ddseq2_V2.2, contrast = c("group", "AUAL", "MUAL"), alpha = 0.05, lfcThreshold = 0))
AUAL_vs_MUALresSig_V2.2 <- subset(results(ddseq2_V2.2, contrast = c("group", "AUAL", "MUAL"), lfcThreshold = 0), padj < 0.05)
#############################################################################
## LFI in Mature
# MTAL vs MUAL
summary(results(ddseq2_V2.2, contrast = c("group", "MTAL", "MUAL"), alpha = 0.05, lfcThreshold = 0))
MTAL_vs_MUALresSig_V2.2 <- subset(results(ddseq2_V2.2, contrast = c("group", "MTAL", "MUAL"), lfcThreshold = 0), padj < 0.05)
#############################################################################
## LFI in Aged
# ATAL vs AUAL
summary(results(ddseq2_V2.2, contrast = c("group", "ATAL", "AUAL"), alpha = 0.05, lfcThreshold = 0))
ATAL_vs_AUALresSig_V2.2 <- subset(results(ddseq2_V2.2, contrast = c("group", "ATAL", "AUAL"), lfcThreshold = 0), padj < 0.05)
#############################################################################
## Age under CR
# AUCR vs MUCR
summary(results(ddseq2_V2.2, contrast = c("group", "AUCR", "MUCR"), alpha = 0.05, lfcThreshold = 0))
AUCR_vs_MUCRresSig_V2.2 <- subset(results(ddseq2_V2.2, contrast = c("group", "AUCR", "MUCR"), lfcThreshold = 0), padj < 0.05)
#############################################################################
## LFI in Mature CR
# MTCR vs MUCR
summary(results(ddseq2_V2.2, contrast = c("group", "MTCR", "MUCR"), alpha = 0.05, lfcThreshold = 0))
MTCR_vs_MUCRresSig_V2.2 <- subset(results(ddseq2_V2.2, contrast = c("group", "MTCR", "MUCR"), lfcThreshold = 0), padj < 0.05)
#############################################################################
## LFI in Aged CR
# ATCR vs AUCR
summary(results(ddseq2_V2.2, contrast = c("group", "ATCR", "AUCR"), alpha = 0.05, lfcThreshold = 0))
ATCR_vs_AUCRresSig_V2.2 <- subset(results(ddseq2_V2.2, contrast = c("group", "ATCR", "AUCR"), lfcThreshold = 0), padj < 0.05)



#########################################################################
## I will plot ALL DEGs
# first make a dataframe with all the DEGS
AUAL_vs_MUAL_DEGl2fc_V2.2 <- AUAL_vs_MUALresSig_V2.2[2] %>% as.data.frame() %>%
  rownames_to_column() %>% set_names(c("Gene", "AUAL_vs_MUAL_l2fc_V2.2"))
MTAL_vs_MUALDEGl2fc_V2.2 <- MTAL_vs_MUALresSig_V2.2[2] %>% as.data.frame() %>%
  rownames_to_column() %>% set_names(c("Gene", "MTAL_vs_MUAL_l2fc_V2.2"))
ATAL_vs_AUALDEGl2fc_V2.2 <- ATAL_vs_AUALresSig_V2.2[2] %>% as.data.frame() %>%
  rownames_to_column() %>% set_names(c("Gene", "ATAL_vs_AUAL_l2fc_V2.2"))
AUCR_vs_MUCRDEGl2fc_V2.2 <- AUCR_vs_MUCRresSig_V2.2[2] %>% as.data.frame() %>%
  rownames_to_column() %>% set_names(c("Gene", "AUCR_vs_MUCR_l2fc_V2.2"))
MTCR_vs_MUCRDEGl2fc_V2.2 <- MTCR_vs_MUCRresSig_V2.2[2] %>% as.data.frame() %>%
  rownames_to_column() %>% set_names(c("Gene", "MTCR_vs_MUCR_l2fc_V2.2"))
ATCR_vs_AUCRDEGl2fc_V2.2 <- ATCR_vs_AUCRresSig_V2.2[2] %>% as.data.frame() %>%
  rownames_to_column() %>% set_names(c("Gene", "ATCR_vs_AUCR_l2fc_V2.2"))

listInputs_V2.2 <- list("AUAL_vs_MUAL_V2.2" = AUAL_vs_MUAL_DEGl2fc_V2.2$Gene,
                   "MTAL_vs_MUAL_V2.2" = MTAL_vs_MUALDEGl2fc_V2.2$Gene,
                   "ATAL_vs_AUAL_V2.2" = ATAL_vs_AUALDEGl2fc_V2.2$Gene,
                   "AUCR_vs_MUCR_V2.2" = AUCR_vs_MUCRDEGl2fc_V2.2$Gene,
                   "MTCR_vs_MUCR_V2.2" = MTCR_vs_MUCRDEGl2fc_V2.2$Gene,
                   "ATCR_vs_AUCR_V2.2" = ATCR_vs_AUCRDEGl2fc_V2.2$Gene)
upset(fromList(listInputs_V2.2), sets=c("AUAL_vs_MUAL_V2.2", "MTAL_vs_MUAL_V2.2", "ATAL_vs_AUAL_V2.2", "AUCR_vs_MUCR_V2.2", "MTCR_vs_MUCR_V2.2", "ATCR_vs_AUCR_V2.2"), order.by = "degree")
```

##### Heirachical clustering heatmap
```{r}
# let's make a dataframe with all DEGs together from all comparisons
All_DEGs_V2.2 <- full_join(AUAL_vs_MUAL_DEGl2fc_V2.2, MTAL_vs_MUALDEGl2fc_V2.2, by="Gene")
All_DEGs_V2.2 <- full_join(All_DEGs_V2.2, ATAL_vs_AUALDEGl2fc_V2.2, by="Gene")
All_DEGs_V2.2 <- full_join(All_DEGs_V2.2, AUCR_vs_MUCRDEGl2fc_V2.2, by="Gene")
All_DEGs_V2.2 <- full_join(All_DEGs_V2.2, MTCR_vs_MUCRDEGl2fc_V2.2, by="Gene")
All_DEGs_V2.2 <- full_join(All_DEGs_V2.2, ATCR_vs_AUCRDEGl2fc_V2.2, by="Gene")
rownames(All_DEGs_V2.2) <- All_DEGs_V2.2$Gene
dim(All_DEGs_V2.2) #43 DEGs, 6 comparisons and 1 column of Genes

rlog_out_V2.2 <- rlog(dds2_V2.3, blind=F)
df_V2.2 <- as.data.frame(colData(dds2_V2.3)[,c("Age","Diet","LFI")])
mat1_V2.2 <- assay(rlog_out_V2.2)[rownames(All_DEGs_V2.2),]
colnames(mat1_V2.2)
mat1.scaled_V2.2 <- t(apply(mat1_V2.2, 1, scale))
colnames(mat1.scaled_V2.2) <- colnames(mat1_V2.2)
dim(mat1.scaled_V2.2) #43 genes in 44 samples

pheatmap(mat1.scaled_V2.2, cluster_rows = T, cluster_cols = T, annotation_col = df_V2.2, show_rownames = F) #this everything with clusters

matrix.col.order_Diet_V2.2 <- c("810","816","828","830","836","837", #AL_U_M
                      "801","806","808","817","821","835", #AL_T_M
                      "863","865","874","877","878","881", #AL_U_A
                      "839","841","847","849","871", #AL_T_A
                      "823","824","829", #CR_U_M
                      "802","804","807","811","813","826", #CR_T_M
                      "872","873","875","876","879","880", #CR_U_A
                      "852","855","857","858","860","866") #CR_T_A
mat1.scaled.orderedByDiet_V2.2 <- mat1.scaled_V2.2[,matrix.col.order_Diet_V2.2]
pheatmap(mat1.scaled.orderedByDiet_V2.2, cluster_rows = T, cluster_cols = F, annotation_col = df_V2.2, show_rownames = F, show_colnames = F, cutree_rows = 5)

matrix.col.order_Age_V2.2 <- c("810","816","828","830","836","837", #AL_U_M
                      "801","806","808","817","821","835", #AL_T_M
                      "823","824","829", #CR_U_M
                      "802","804","807","811","813","826", #CR_T_M
                      "863","865","874","877","878","881", #AL_U_A
                      "839","841","847","849","871", #AL_T_A
                      "872","873","875","876","879","880", #CR_U_A
                      "852","855","857","858","860","866") #CR_T_A
mat1.scaled.orderedByAge_V2.2 <- mat1.scaled_V2.2[,matrix.col.order_Age_V2.2]
pheatmap(mat1.scaled.orderedByAge_V2.2, cluster_rows = T, cluster_cols = F, annotation_col = df_V2.2, show_rownames = F, show_colnames = F, cutree_rows = 5)

matrix.col.order_LFI_V2.2 <- c("810","816","828","830","836","837", #AL_U_M
                      "823","824","829", #CR_U_M
                      "863","865","874","877","878","881", #AL_U_A
                      "872","873","875","876","879","880", #CR_U_A
                      "801","806","808","817","821","835", #AL_T_M
                      "802","804","807","811","813","826", #CR_T_M
                      "839","841","847","849","871", #AL_T_A
                      "852","855","857","858","860","866") #CR_T_A
mat1.scaled.orderedByLFI_V2.2 <- mat1.scaled_V2.2[,matrix.col.order_LFI_V2.2]
pheatmap(mat1.scaled.orderedByLFI_V2.2, cluster_rows = T, cluster_cols = F, annotation_col = df_V2.2, show_rownames = F, show_colnames = T, cutree_rows = 9)

## All the heatmaps above did not cluster the samples but I chose where the samples fell
## Here is one that clusters the samples
pheatmap(mat1.scaled.orderedByLFI_V2.2, cluster_rows = T, cluster_cols = T, annotation_col = df_V2.2, show_rownames = F, show_colnames = T, cutree_rows = 9)

```






## V3
```{r }
# This with samples that fell outside the 99% CI for Age
# These samples is 809 Mature, CR, Untrained (M,U,CR)
# and DUD sample 840 Aged, AL, Trained (A,T,CR)

# So now I will delete these samples from my metadata file and use the new metadata fine to import my quants and so forth

metadata_V3 <- sample_table[!(sample_table$Sample_ID %in% c("809","840")),]

paste0('/Users/school/Projects/CR/quants/', 
       pull(metadata_V3, RNAseq_ID), '/quant.sf')
sample_files_V3 = paste0('/Users/school/Projects/CR/quants/',
                      pull(metadata_V3, RNAseq_ID), '/quant.sf')
names(sample_files_V3) = pull(metadata_V3, Sample_ID)
all(file.exists(sample_files_V3))
data.frame(thefiles = metadata_V3, doihave = file.exists(sample_files_V3))


count_data_V3 = tximport(files = sample_files_V3, importer=read.delim,
         type = "salmon",
         tx2gene = tx2gene_aplysia,
         ignoreTxVersion = T)

```


##### Model dds
```{r}
## as a comparison I will run this will all samples (ignoring any outlieres or DUDs)
dds_V3 <- DESeqDataSetFromTximport(count_data_V3,
                                colData = metadata_V3,
                                design = ~ Age + LFI + Diet)
colData(dds_V3)
dds_V3$group <- factor(paste0(dds_V3$Age, dds_V3$LFI, dds_V3$Diet))
design(dds_V3) <- ~group
colData(dds_V3)

## counts pre-filtering
dim(dds_V3) # starting with 21,034 transcripts in 46 samples
smallestGroupSize_V3 <- 5 # there are 5 samples in the smallest group
keep_V3 <- rowSums(counts(dds_V3) >= 1) >= smallestGroupSize_V3
dds2_V3 <- dds_V3[keep_V3,]
dim(dds2_V3) # 11,852 transcripts in 46 samples
```

##### PCA
```{r}
vsd_V3 <- vst(dds2_V3, blind = F)
pc12_V3 <- plotPCA(vsd_V3, intgroup = c("Age","LFI","Diet","group"), returnData = T) #PC1=27, PC2=13
ggplot(pc12_V3, aes(x=PC1, y=PC2, color=LFI)) +
  geom_point(aes(fill=LFI), 
       colour="black",pch=21, size=5)  +
  xlab(paste0("PC1: 27% varience")) +
  ylab(paste0("PC2: 13% variance")) +
  coord_fixed() +
  #scale_color_manual(labels = c("Younger","Aged"),
  #                   values = c("red","blue")) +
  #ggtitle("PCA with VST data") +
  stat_ellipse(level = 0.999) +
  theme_bw() +
  guides(fill=F) +
  theme(axis.text.x=element_text(size=18), 
        axis.title.x=element_text(size=18),
        axis.text.y=element_text(size=18), 
        axis.title.y=element_text(size=18),
        legend.title = element_text(size=16),
        legend.key.size = unit(0.6, 'cm'),
        legend.position = c(0.91, 0.91)) +
  scale_x_continuous(limits = c(-40, 40)) +
  scale_y_continuous(limits = c(-40, 40))
```

##### DESeq2
```{r}
dds2_V3.1 <- estimateSizeFactors(dds2_V3)
ddseq2_V3 <- DESeq(dds2_V3.1)

## Age differences
# AUAL vs MUAL
summary(results(ddseq2_V3, contrast = c("group", "AUAL", "MUAL"), alpha = 0.05, lfcThreshold = 0))
AUAL_vs_MUALresSig_V3 <- subset(results(ddseq2_V3, contrast = c("group", "AUAL", "MUAL"), lfcThreshold = 0), padj < 0.05)
#############################################################################
## LFI in Mature
# MTAL vs MUAL
summary(results(ddseq2_V3, contrast = c("group", "MTAL", "MUAL"), alpha = 0.05, lfcThreshold = 0))
MTAL_vs_MUALresSig_V3 <- subset(results(ddseq2_V3, contrast = c("group", "MTAL", "MUAL"), lfcThreshold = 0), padj < 0.05)
#############################################################################
## LFI in Aged
# ATAL vs AUAL
summary(results(ddseq2_V3, contrast = c("group", "ATAL", "AUAL"), alpha = 0.05, lfcThreshold = 0))
ATAL_vs_AUALresSig_V3 <- subset(results(ddseq2_V3, contrast = c("group", "ATAL", "AUAL"), lfcThreshold = 0), padj < 0.05)
#############################################################################
## Age under CR
# AUCR vs MUCR
summary(results(ddseq2_V3, contrast = c("group", "AUCR", "MUCR"), alpha = 0.05, lfcThreshold = 0))
AUCR_vs_MUCRresSig_V3 <- subset(results(ddseq2_V3, contrast = c("group", "AUCR", "MUCR"), lfcThreshold = 0), padj < 0.05)
#############################################################################
## LFI in Mature CR
# MTCR vs MUCR
summary(results(ddseq2_V3, contrast = c("group", "MTCR", "MUCR"), alpha = 0.05, lfcThreshold = 0))
MTCR_vs_MUCRresSig_V3 <- subset(results(ddseq2_V3, contrast = c("group", "MTCR", "MUCR"), lfcThreshold = 0), padj < 0.05)
#############################################################################
## LFI in Aged CR
# ATCR vs AUCR
summary(results(ddseq2_V3, contrast = c("group", "ATCR", "AUCR"), alpha = 0.05, lfcThreshold = 0))
ATCR_vs_AUCRresSig_V3 <- subset(results(ddseq2_V3, contrast = c("group", "ATCR", "AUCR"), lfcThreshold = 0), padj < 0.05)



#########################################################################
## I will plot ALL DEGs
# first make a dataframe with all the DEGS
AUAL_vs_MUAL_DEGl2fc_V3 <- AUAL_vs_MUALresSig_V3[2] %>% as.data.frame() %>%
  rownames_to_column() %>% set_names(c("Gene", "AUAL_vs_MUAL_l2fc_V3"))
MTAL_vs_MUALDEGl2fc_V3 <- MTAL_vs_MUALresSig_V3[2] %>% as.data.frame() %>%
  rownames_to_column() %>% set_names(c("Gene", "MTAL_vs_MUAL_l2fc_V3"))
ATAL_vs_AUALDEGl2fc_V3 <- ATAL_vs_AUALresSig_V3[2] %>% as.data.frame() %>%
  rownames_to_column() %>% set_names(c("Gene", "ATAL_vs_AUAL_l2fc_V3"))
AUCR_vs_MUCRDEGl2fc_V3 <- AUCR_vs_MUCRresSig_V3[2] %>% as.data.frame() %>%
  rownames_to_column() %>% set_names(c("Gene", "AUCR_vs_MUCR_l2fc_V3"))
MTCR_vs_MUCRDEGl2fc_V3 <- MTCR_vs_MUCRresSig_V3[2] %>% as.data.frame() %>%
  rownames_to_column() %>% set_names(c("Gene", "MTCR_vs_MUCR_l2fc_V3"))
ATCR_vs_AUCRDEGl2fc_V3 <- ATCR_vs_AUCRresSig_V3[2] %>% as.data.frame() %>%
  rownames_to_column() %>% set_names(c("Gene", "ATCR_vs_AUCR_l2fc_V3"))

listInputs_V3 <- list("AUAL_vs_MUAL_V3" = AUAL_vs_MUAL_DEGl2fc_V3$Gene,
                   "MTAL_vs_MUAL_V3" = MTAL_vs_MUALDEGl2fc_V3$Gene,
                   "ATAL_vs_AUAL_V3" = ATAL_vs_AUALDEGl2fc_V3$Gene,
                   "AUCR_vs_MUCR_V3" = AUCR_vs_MUCRDEGl2fc_V3$Gene,
                   "MTCR_vs_MUCR_V3" = MTCR_vs_MUCRDEGl2fc_V3$Gene,
                   "ATCR_vs_AUCR_V3" = ATCR_vs_AUCRDEGl2fc_V3$Gene)
upset(fromList(listInputs_V3), sets=c("AUAL_vs_MUAL_V3", "MTAL_vs_MUAL_V3", "ATAL_vs_AUAL_V3", "AUCR_vs_MUCR_V3", "MTCR_vs_MUCR_V3", "ATCR_vs_AUCR_V3"), order.by = "degree")
```

##### Heirachical clustering heatmap
```{r}
# let's make a dataframe with all DEGs together from all comparisons
All_DEGs_V3 <- full_join(AUAL_vs_MUAL_DEGl2fc_V3, MTAL_vs_MUALDEGl2fc_V3, by="Gene")
All_DEGs_V3 <- full_join(All_DEGs_V3, ATAL_vs_AUALDEGl2fc_V3, by="Gene")
All_DEGs_V3 <- full_join(All_DEGs_V3, AUCR_vs_MUCRDEGl2fc_V3, by="Gene")
All_DEGs_V3 <- full_join(All_DEGs_V3, MTCR_vs_MUCRDEGl2fc_V3, by="Gene")
All_DEGs_V3 <- full_join(All_DEGs_V3, ATCR_vs_AUCRDEGl2fc_V3, by="Gene")
rownames(All_DEGs_V3) <- All_DEGs_V3$Gene
dim(All_DEGs_V3) #88 DEGs, 6 comparisons and 1 column of Genes

rlog_out_V3 <- rlog(dds2_V3.1, blind=F)
df_V3 <- as.data.frame(colData(dds2_V3.1)[,c("Age","Diet","LFI")])
mat1_V3 <- assay(rlog_out_V3)[rownames(All_DEGs_V3),]
colnames(mat1_V3)
mat1.scaled_V3 <- t(apply(mat1_V3, 1, scale))
colnames(mat1.scaled_V3) <- colnames(mat1_V3)
dim(mat1.scaled_V3) #88 genes in 46 samples

pheatmap(mat1.scaled_V3, cluster_rows = T, cluster_cols = T, annotation_col = df_V3, show_rownames = F) #this everything with clusters

matrix.col.order_Diet_V3 <- c("810","816","828","830","836","837", #AL_U_M
                      "801","806","808","817","821","835", #AL_T_M
                      "863","865","874","877","878","881", #AL_U_A
                      "839","841","847","849","871", #AL_T_A
                      "805","815","823","824","829", #CR_U_M
                      "802","804","807","811","813","826", #CR_T_M
                      "872","873","875","876","879","880", #CR_U_A
                      "852","855","857","858","860","866") #CR_T_A
mat1.scaled.orderedByDiet_V3 <- mat1.scaled_V3[,matrix.col.order_Diet_V3]
pheatmap(mat1.scaled.orderedByDiet_V3, cluster_rows = T, cluster_cols = F, annotation_col = df_V3, show_rownames = F, show_colnames = F, cutree_rows = 7)

matrix.col.order_Age_V3 <- c("810","816","828","830","836","837", #AL_U_M
                      "801","806","808","817","821","835", #AL_T_M
                      "805","815","823","824","829", #CR_U_M
                      "802","804","807","811","813","826", #CR_T_M
                      "863","865","874","877","878","881", #AL_U_A
                      "839","841","847","849","871", #AL_T_A
                      "872","873","875","876","879","880", #CR_U_A
                      "852","855","857","858","860","866") #CR_T_A
mat1.scaled.orderedByAge_V3 <- mat1.scaled_V3[,matrix.col.order_Age_V3]
pheatmap(mat1.scaled.orderedByAge_V3, cluster_rows = T, cluster_cols = F, annotation_col = df_V3, show_rownames = F, show_colnames = F, cutree_rows = 7)

matrix.col.order_LFI_V3 <- c("810","816","828","830","836","837", #AL_U_M
                      "805","815","823","824","829", #CR_U_M
                      "863","865","874","877","878","881", #AL_U_A
                      "872","873","875","876","879","880", #CR_U_A
                      "801","806","808","817","821","835", #AL_T_M
                      "802","804","807","811","813","826", #CR_T_M
                      "839","841","847","849","871", #AL_T_A
                      "852","855","857","858","860","866") #CR_T_A
mat1.scaled.orderedByLFI_V3 <- mat1.scaled_V3[,matrix.col.order_LFI_V3]
pheatmap(mat1.scaled.orderedByLFI_V3, cluster_rows = T, cluster_cols = F, annotation_col = df_V3, show_rownames = F, show_colnames = T, cutree_rows = 7)

## All the heatmaps above did not cluster the samples but I chose where the samples fell
## Here is one that clusters the samples
pheatmap(mat1.scaled.orderedByLFI_V3, cluster_rows = T, cluster_cols = T, annotation_col = df_V3, show_rownames = F, show_colnames = T, cutree_rows = 7)

```

### V3_w/SVA
##### SVA analysis
```{r}
mod_V3 <- model.matrix(~ Age + LFI + Diet, data=metadata_V3)
mod0_V3 <- model.matrix(~1, data=metadata_V3)

dat_V3 <- counts(dds2_V3.1, normalized=T)
num.sv(dat_V3, mod_V3, method = "be") #this tells me how many surrogate variables there are 
## this said there were zero 0 surrogate variables 
## Still going to try this with 2

#za2 <- za[rowSums(za >= 7) >= 4,] #had to filter the data because too many zeros messed up the operation

svobl_V3 <- svaseq(dat_V3, mod_V3, mod0_V3, n.sv = 2)
metadata_V3$sV3=svobl_V3$sv[,1]
metadata_V3$sv2=svobl_V3$sv[,2]

## now that I have located surrogate variables all I have to do is incorporate them into my DESeq2 model matrix - I do not need to eliminate them from my raw data. This has the possibility of over correcting. 


#Quick check at variance
#plot(svobl$sv[,],pch=19,col="blue")
#plot(as.matrix(za2[,33]),pch=19,col="blue") #this has nothing to do with sav, this is visualizing the TPMs of genes in samples within the matrix. You can call each sample by selecing the column number and putting it after the comma
# ggplot(data=sample_table_minusOutliers, aes( x=Age, y=sv2)) +
#    geom_point()
```

##### Model dds
```{r}
## as a comparison I will run this will all samples (ignoring any outlieres or DUDs)
dds_V3.2 <- DESeqDataSetFromTximport(count_data_V3,
                                colData = metadata_V3,
                                design = ~ sV3 + sv2 + Age + LFI + Diet)
colData(dds_V3.2)
dds_V3.2$group <- factor(paste0(dds_V3.2$Age, dds_V3.2$LFI, dds_V3.2$Diet))
design(dds_V3.2) <- ~group
colData(dds_V3.2)

## counts pre-filtering
dim(dds_V3.2) # starting with 21,034 transcripts in 46 samples
smallestGroupSize_V3.2 <- 5 # there are 5 samples in the smallest group
keep_V3.2 <- rowSums(counts(dds_V3.2) >= 1) >= smallestGroupSize_V3.2
dds2_V3.2 <- dds_V3.2[keep_V3.2,]
dim(dds2_V3.2) # 11,852 transcripts in 46 samples
```

##### PCA
```{r}
vsd_V3.2 <- vst(dds2_V3.2)
pc12_V3.2 <- plotPCA(vsd_V3.2, intgroup = c("Age","LFI","Diet","group"), returnData = T) #PC1=26, PC2=13
ggplot(pc12_V3.2, aes(x=PC1, y=PC2, color=LFI)) +
  geom_point(aes(fill=LFI), 
       colour="black",pch=21, size=5)  +
  xlab(paste0("PC1: 26% varience")) +
  ylab(paste0("PC2: 13% variance")) +
  coord_fixed() +
  #scale_color_manual(labels = c("Younger","Aged"),
  #                   values = c("red","blue")) +
  #ggtitle("PCA with VST data") +
  stat_ellipse(level = 0.999) +
  theme_bw() +
  guides(fill=F) +
  theme(axis.text.x=element_text(size=18), 
        axis.title.x=element_text(size=18),
        axis.text.y=element_text(size=18), 
        axis.title.y=element_text(size=18),
        legend.title = element_text(size=16),
        legend.key.size = unit(0.6, 'cm'),
        legend.position = c(0.91, 0.91)) +
  scale_x_continuous(limits = c(-40, 40)) +
  scale_y_continuous(limits = c(-40, 40))
```

##### DESeq2
```{r}
dds2_V3.3 <- estimateSizeFactors(dds2_V3.2)
ddseq2_V3.2 <- DESeq(dds2_V3.3)

## Age differences
# AUAL vs MUAL
summary(results(ddseq2_V3.2, contrast = c("group", "AUAL", "MUAL"), alpha = 0.05, lfcThreshold = 0))
AUAL_vs_MUALresSig_V3.2 <- subset(results(ddseq2_V3.2, contrast = c("group", "AUAL", "MUAL"), lfcThreshold = 0), padj < 0.05)
#############################################################################
## LFI in Mature
# MTAL vs MUAL
summary(results(ddseq2_V3.2, contrast = c("group", "MTAL", "MUAL"), alpha = 0.05, lfcThreshold = 0))
MTAL_vs_MUALresSig_V3.2 <- subset(results(ddseq2_V3.2, contrast = c("group", "MTAL", "MUAL"), lfcThreshold = 0), padj < 0.05)
#############################################################################
## LFI in Aged
# ATAL vs AUAL
summary(results(ddseq2_V3.2, contrast = c("group", "ATAL", "AUAL"), alpha = 0.05, lfcThreshold = 0))
ATAL_vs_AUALresSig_V3.2 <- subset(results(ddseq2_V3.2, contrast = c("group", "ATAL", "AUAL"), lfcThreshold = 0), padj < 0.05)
#############################################################################
## Age under CR
# AUCR vs MUCR
summary(results(ddseq2_V3.2, contrast = c("group", "AUCR", "MUCR"), alpha = 0.05, lfcThreshold = 0))
AUCR_vs_MUCRresSig_V3.2 <- subset(results(ddseq2_V3.2, contrast = c("group", "AUCR", "MUCR"), lfcThreshold = 0), padj < 0.05)
#############################################################################
## LFI in Mature CR
# MTCR vs MUCR
summary(results(ddseq2_V3.2, contrast = c("group", "MTCR", "MUCR"), alpha = 0.05, lfcThreshold = 0))
MTCR_vs_MUCRresSig_V3.2 <- subset(results(ddseq2_V3.2, contrast = c("group", "MTCR", "MUCR"), lfcThreshold = 0), padj < 0.05)
#############################################################################
## LFI in Aged CR
# ATCR vs AUCR
summary(results(ddseq2_V3.2, contrast = c("group", "ATCR", "AUCR"), alpha = 0.05, lfcThreshold = 0))
ATCR_vs_AUCRresSig_V3.2 <- subset(results(ddseq2_V3.2, contrast = c("group", "ATCR", "AUCR"), lfcThreshold = 0), padj < 0.05)



#########################################################################
## I will plot ALL DEGs
# first make a dataframe with all the DEGS
AUAL_vs_MUAL_DEGl2fc_V3.2 <- AUAL_vs_MUALresSig_V3.2[2] %>% as.data.frame() %>%
  rownames_to_column() %>% set_names(c("Gene", "AUAL_vs_MUAL_l2fc_V3.2"))
MTAL_vs_MUALDEGl2fc_V3.2 <- MTAL_vs_MUALresSig_V3.2[2] %>% as.data.frame() %>%
  rownames_to_column() %>% set_names(c("Gene", "MTAL_vs_MUAL_l2fc_V3.2"))
ATAL_vs_AUALDEGl2fc_V3.2 <- ATAL_vs_AUALresSig_V3.2[2] %>% as.data.frame() %>%
  rownames_to_column() %>% set_names(c("Gene", "ATAL_vs_AUAL_l2fc_V3.2"))
AUCR_vs_MUCRDEGl2fc_V3.2 <- AUCR_vs_MUCRresSig_V3.2[2] %>% as.data.frame() %>%
  rownames_to_column() %>% set_names(c("Gene", "AUCR_vs_MUCR_l2fc_V3.2"))
MTCR_vs_MUCRDEGl2fc_V3.2 <- MTCR_vs_MUCRresSig_V3.2[2] %>% as.data.frame() %>%
  rownames_to_column() %>% set_names(c("Gene", "MTCR_vs_MUCR_l2fc_V3.2"))
ATCR_vs_AUCRDEGl2fc_V3.2 <- ATCR_vs_AUCRresSig_V3.2[2] %>% as.data.frame() %>%
  rownames_to_column() %>% set_names(c("Gene", "ATCR_vs_AUCR_l2fc_V3.2"))

listInputs_V3.2 <- list("AUAL_vs_MUAL_V3.2" = AUAL_vs_MUAL_DEGl2fc_V3.2$Gene,
                   "MTAL_vs_MUAL_V3.2" = MTAL_vs_MUALDEGl2fc_V3.2$Gene,
                   "ATAL_vs_AUAL_V3.2" = ATAL_vs_AUALDEGl2fc_V3.2$Gene,
                   "AUCR_vs_MUCR_V3.2" = AUCR_vs_MUCRDEGl2fc_V3.2$Gene,
                   "MTCR_vs_MUCR_V3.2" = MTCR_vs_MUCRDEGl2fc_V3.2$Gene,
                   "ATCR_vs_AUCR_V3.2" = ATCR_vs_AUCRDEGl2fc_V3.2$Gene)
upset(fromList(listInputs_V3.2), sets=c("AUAL_vs_MUAL_V3.2", "MTAL_vs_MUAL_V3.2", "ATAL_vs_AUAL_V3.2", "AUCR_vs_MUCR_V3.2", "MTCR_vs_MUCR_V3.2", "ATCR_vs_AUCR_V3.2"), order.by = "degree")
```

##### Heirachical clustering heatmap
```{r}
# let's make a dataframe with all DEGs together from all comparisons
All_DEGs_V3.2 <- full_join(AUAL_vs_MUAL_DEGl2fc_V3.2, MTAL_vs_MUALDEGl2fc_V3.2, by="Gene")
All_DEGs_V3.2 <- full_join(All_DEGs_V3.2, ATAL_vs_AUALDEGl2fc_V3.2, by="Gene")
All_DEGs_V3.2 <- full_join(All_DEGs_V3.2, AUCR_vs_MUCRDEGl2fc_V3.2, by="Gene")
All_DEGs_V3.2 <- full_join(All_DEGs_V3.2, MTCR_vs_MUCRDEGl2fc_V3.2, by="Gene")
All_DEGs_V3.2 <- full_join(All_DEGs_V3.2, ATCR_vs_AUCRDEGl2fc_V3.2, by="Gene")
rownames(All_DEGs_V3.2) <- All_DEGs_V3.2$Gene
dim(All_DEGs_V3.2) #88 DEGs, 6 comparisons and 1 column of Genes

rlog_out_V3.2 <- rlog(dds2_V3.3, blind=F)
df_V3.2 <- as.data.frame(colData(dds2_V3.3)[,c("Age","Diet","LFI")])
mat1_V3.2 <- assay(rlog_out_V3.2)[rownames(All_DEGs_V3.2),]
colnames(mat1_V3.2)
mat1.scaled_V3.2 <- t(apply(mat1_V3.2, 1, scale))
colnames(mat1.scaled_V3.2) <- colnames(mat1_V3.2)
dim(mat1.scaled_V3.2) #88 genes in 46 samples

pheatmap(mat1.scaled_V3.2, cluster_rows = T, cluster_cols = T, annotation_col = df_V3.2, show_rownames = F) #this everything with clusters

matrix.col.order_Diet_V3.2 <- c("810","816","828","830","836","837", #AL_U_M
                      "801","806","808","817","821","835", #AL_T_M
                      "863","865","874","877","878","881", #AL_U_A
                      "839","841","847","849","871", #AL_T_A
                      "805","815","823","824","829", #CR_U_M
                      "802","804","807","811","813","826", #CR_T_M
                      "872","873","875","876","879","880", #CR_U_A
                      "852","855","857","858","860","866") #CR_T_A
mat1.scaled.orderedByDiet_V3.2 <- mat1.scaled_V3.2[,matrix.col.order_Diet_V3.2]
pheatmap(mat1.scaled.orderedByDiet_V3.2, cluster_rows = T, cluster_cols = F, annotation_col = df_V3.2, show_rownames = F, show_colnames = F, cutree_rows = 5)

matrix.col.order_Age_V3.2 <- c("810","816","828","830","836","837", #AL_U_M
                      "801","806","808","817","821","835", #AL_T_M
                      "805","815","823","824","829", #CR_U_M
                      "802","804","807","811","813","826", #CR_T_M
                      "863","865","874","877","878","881", #AL_U_A
                      "839","841","847","849","871", #AL_T_A
                      "872","873","875","876","879","880", #CR_U_A
                      "852","855","857","858","860","866") #CR_T_A
mat1.scaled.orderedByAge_V3.2 <- mat1.scaled_V3.2[,matrix.col.order_Age_V3.2]
pheatmap(mat1.scaled.orderedByAge_V3.2, cluster_rows = T, cluster_cols = F, annotation_col = df_V3.2, show_rownames = F, show_colnames = F, cutree_rows = 5)

matrix.col.order_LFI_V3.2 <- c("810","816","828","830","836","837", #AL_U_M
                      "805","815","823","824","829", #CR_U_M
                      "863","865","874","877","878","881", #AL_U_A
                      "872","873","875","876","879","880", #CR_U_A
                      "801","806","808","817","821","835", #AL_T_M
                      "802","804","807","811","813","826", #CR_T_M
                      "839","841","847","849","871", #AL_T_A
                      "852","855","857","858","860","866") #CR_T_A
mat1.scaled.orderedByLFI_V3.2 <- mat1.scaled_V3.2[,matrix.col.order_LFI_V3.2]
pheatmap(mat1.scaled.orderedByLFI_V3.2, cluster_rows = T, cluster_cols = F, annotation_col = df_V3.2, show_rownames = F, show_colnames = T, cutree_rows = 9)

## All the heatmaps above did not cluster the samples but I chose where the samples fell
## Here is one that clusters the samples
pheatmap(mat1.scaled.orderedByLFI_V3.2, cluster_rows = T, cluster_cols = T, annotation_col = df_V3.2, show_rownames = F, show_colnames = T, cutree_rows = 9)

```

# ________________
## PCA stuff!
###Load in data
```{r}
# I have learned a lot about analyzing PCs so here I will put that to use with these data and make a scree plot, come correlation matrices, and plot additional PCs
## I will be working with data from "V3 w/out SAV" above
library(PCAtools)

sample_table = read.csv("/Users/school/Projects/CR/samples.txt", header = T, stringsAsFactors = T)
sample_table$P_SAV <- as.character(sample_table$P_SAV)
sample_table$Age <- factor(sample_table$Age, levels = c("M","A") )
sample_table$TRT <- factor(sample_table$TRT, levels = c("U", "T"))
names(sample_table)[names(sample_table) == "TRT"] <- "LFI"
row.names(sample_table)=sample_table$Sample_ID

# This with samples that fell outside the 99% CI for Age
# These samples is 809 Mature, CR, Untrained (M,U,CR)
# and DUD sample 840 Aged, AL, Trained (A,T,CR)

# So now I will delete these samples from my metadata file and use the new metadata fine to import my quants and so forth

metadata_V3 <- sample_table[!(sample_table$Sample_ID %in% c("809","840")),]

paste0('/Users/school/Projects/CR/quants/', 
       pull(metadata_V3, RNAseq_ID), '/quant.sf')
sample_files_V3 = paste0('/Users/school/Projects/CR/quants/',
                      pull(metadata_V3, RNAseq_ID), '/quant.sf')
names(sample_files_V3) = pull(metadata_V3, Sample_ID)
all(file.exists(sample_files_V3))
data.frame(thefiles = metadata_V3, doihave = file.exists(sample_files_V3))

txdb_aplysia <- makeTxDbFromGFF("/Users/school/Documents/UAB/UAB009/genome/AplCal3.0_GFF.gff",
                        format = "gff",
                        organism = "Aplysia californica")
k_aplysia <- keys(txdb_aplysia, keytype = "TXNAME")
tx2gene_aplysia <- AnnotationDbi::select(txdb_aplysia, k_aplysia, "GENEID", "TXNAME")
tx2gene_aplysia$TXNAME <- gsub("\\.[0-9]","",tx2gene_aplysia$TXNAME) 

count_data_V3 = tximport(files = sample_files_V3, importer=read.delim,
         type = "salmon",
         tx2gene = tx2gene_aplysia,
         ignoreTxVersion = T)
```
### Model dds
```{r}
## as a comparison I will run this will all samples (ignoring any outlieres or DUDs)
dds_V3 <- DESeqDataSetFromTximport(count_data_V3,
                                colData = metadata_V3,
                                design = ~ Age + LFI + Diet)
colData(dds_V3)
dds_V3$group <- factor(paste0(dds_V3$Age, dds_V3$LFI, dds_V3$Diet))
design(dds_V3) <- ~group
colData(dds_V3)

## counts pre-filtering
dim(dds_V3) # starting with 21,034 transcripts in 46 samples
smallestGroupSize_V3 <- 5 # there are 5 samples in the smallest group
keep_V3 <- rowSums(counts(dds_V3) >= 1) >= smallestGroupSize_V3
dds2_V3 <- dds_V3[keep_V3,]
dim(dds2_V3) # 11,852 transcripts in 46 samples
```

### PCAtools
```{r}
# first let's try to get some gene names annotated with the org.db I built for Aplysia!
myDat <- dds2_V3 # make an object with my pre-filtered count data
ens <- rownames(myDat) #make an object of a list of the genes
#install.packages("/Users/school/Documents/UAB/UAB009/Org.dbs/org.Acalifornica.eg.db", repos=NULL, type="source") # install the org.db I created
library(org.Acalifornica.eg.db) # load the org.db
keytypes(org.Acalifornica.eg.db)
symbols <- mapIds(org.Acalifornica.eg.db, keys = ens,
                  column = c('ALIAS'), keytype = 'SYMBOL') # we are going with full gene names for this because the SYMBOL just gives me back the LOC# which is meaningless and the ALIAS give me a paltry selection back. GENENAME gives me the most bang for my buck!
symbols <- symbols[!is.na(symbols)]
symbols <- symbols[match(rownames(myDat), names(symbols))]
rownames(myDat) <- symbols
keep <- !is.na(rownames(myDat))
myDat <- myDat[keep,]
# Normalize the data and transform the normalized counts to variance-stabilized expression levels
dds4PCA <- DESeqDataSet(myDat,
                        design = ~ Age + LFI + Diet)
dds4PCA <- DESeq(dds4PCA)
vsd4PCA <- assay(vst(dds4PCA))
# Conduct PCA
p <- pca(vsd4PCA, metadata = colData(myDat), removeVar = 0.1) # removing the lower 10% of variables based on variance
## the 'pca' function in PCAtools does not work. I will update it and the save it into PCAtools so the updated version will be called every time I call a function with this function nested within it.
PCAtools:::pca # this lets me see what the original function is
pcaUpdateFromPCAtools = function (mat, metadata = NULL, center = TRUE, scale = FALSE, 
    rank = NULL, removeVar = NULL, transposed = FALSE, BSPARAM = ExactParam()) 
{
    if (is.data.frame(mat)) {
        mat <- as.matrix(mat)
    }
    if (!transposed) {
        mat <- t(mat)
    }
    if (!is.null(metadata)) {
        if (!identical(rownames(mat), rownames(metadata))) {
            stop("'colnames(mat)' is not identical to 'rownames(metadata)'")
        }
    }
    .center <- if (center) 
        NULL
    else 0
    vars <- matrixStats::colVars(as.matrix(DelayedArray(mat)), useNames = TRUE, 
                                 center = .center)
    if (!is.null(removeVar)) {
        message("-- removing the lower ", removeVar * 100, "% of variables based on variance")
        varorder <- order(vars, decreasing = TRUE)
        keep <- head(varorder, max(1, ncol(mat) * (1 - removeVar)))
        mat <- mat[, keep, drop = FALSE]
        vars <- vars[keep]
    }
    if (is.null(rank)) {
        if (is(BSPARAM, "ExactParam")) {
            rank <- min(dim(mat))
        }
        else {
            stop("'rank' must be specified for approximate PCA methods")
        }
    }
    pcaobj <- runPCA(mat, center = center, scale = scale, rank = rank, 
        BSPARAM = BSPARAM)
    if (scale) {
        total.var <- length(vars)
    }
    else {
        total.var <- sum(vars)
    }
    proportionvar <- (pcaobj$sdev^2)/total.var * 100
    pcaobj <- list(rotated = data.frame(pcaobj$x), loadings = data.frame(pcaobj$rotation), 
        variance = proportionvar, sdev = pcaobj$sdev, metadata = metadata, 
        xvars = colnames(mat), yvars = rownames(mat), components = colnames(pcaobj$x))
    rownames(pcaobj$rotated) <- pcaobj$yvars
    rownames(pcaobj$loadings) <- pcaobj$xvars
    names(pcaobj$variance) <- pcaobj$components
    class(pcaobj) <- "pca"
    return(pcaobj)
} # this updated the function and named it something else
environment(pcaUpdateFromPCAtools) <- asNamespace('PCAtools') # this lets R know that this function should be in the namespace of PCAtools to allow any nested functions within this updated function to work properly
assignInNamespace('pca', pcaUpdateFromPCAtools, ns = 'PCAtools') # this now tells R to replace 'pca' with the updated function any time 'pca' is called in PCAtools
p <- pcaUpdateFromPCAtools(vsd4PCA, metadata = colData(myDat), removeVar = 0.1)
p$loadings # this give list of genes that contribute to PCs
moreThan70percentVarience <- which(cumsum(p$variance) > 70)[1] # this give how many PCs account for more than 70% variance (24)
PCAtools::screeplot(p,
          components = getComponents(p),
          vline = moreThan70percentVarience) +
  geom_label(aes(x=moreThan70percentVarience + 0.5, y = 50,
                 label = ">70%", vjust = -1, size = 8))  # this is screeplot showing 

PCAtools::eigencorplot(p, 
                       metavars = c("Age","Diet","LFI","group"))

PCAtools::eigencorplot(p,
             components = getComponents(p, 1:24),
             metavars = c("Age","Diet","LFI","group"),
             col = c('darkblue','blue2','black','red2','darkred'),
             cexCorval = 0.7,
             colCorval = 'white',
             fontCorval = 2,
             posLab = 'bottomleft',
             rotLabX = 45,
             posColKey = 'top',
             cexLabColKey = 1.5,
             scale = T,
             main = 'PC1-8 trait correlations',
             colFrame = 'white',
             plotRsquared = F)

## This shows that Age is positively correlated to PC1, Diet is positively correlated to PC3, and LFI is positively correlated to PC6 
## past PC7 there is no more significant correlations of traits to PCs. I will cut all graphs off here

eigencorplot(p, 
             components = getComponents(p, 1:7),
             metavars = c("Age","Diet","LFI","group"),
             col = c('white','cornsilk1','gold','forestgreen','darkgreen'),
             cexCorval = 1.2,
             fontCorval = 2,
             posLab = 'all',
             rotLabX = 45,
             scale = T,
             main = bquote(Principal~component~Pearson~r^2~trait~correlates),
             plotRsquared = T,
             corFUN = 'pearson',
             corUSE = 'pairwise.complete.obs',
             corMultipleTestCorrection = 'BH',
             signifSymbols = c('****', '***', '**', '*', ''),
             signifCutpoints = c(0, 0.0001, 0.001, 0.1, 0.05, 1))

## This hows that r^2 is significant for Age and group to PC1 and Diet is significant for a negative correlation to PC6 and PC7 (meaningless for me)

# plotloadings(p,
#              rangeRetain = 0.01,
#              labSize = 4,
#              title = 'Loadings plot',
#              subtitle = 'PC1, PC2, PC3, PC4, PC5, PC6',
#              caption = 'Top 1% variables',
#              shape = 24,
#              col = c('limegreen','black','red3'),
#              drawConnectors = T)




############################################################################
### All the above is cool but i notice that I am having variance consences issues between the two packages ('PCAtools' and 'DESeq2') I suspect this because PCAtools uses far more genes when calculating PC variance that DESeq2. I say this because DESeq2 uses the top 500 genes awhile PCAtools, I believe, uses all genes minus the lowest 10% that contribute to varience. I wonder if I use the top 500 genes in PCAtools if I will get the same vareence I get with DESeq2. This is what i will attempt below.
## This involves a couple steps, First subset the data going into PCAtools and the compare with DESeq2 subset.
## first I will use the fist 3 lines of code from the function 'DESeq2::plotPCA' to get a list of the top 500 genes
## Then I will attempt to incorporate this filter into 'PCAtools::pca' function and see if I get the same results
as.data.frame(vsd4PCA)
as.data.frame(t(vsd4PCA))

# DESeq2:::plotPCA.DESeqTransform
# genesDrivingPCs = function (object, ntop = 500, returnData = T) 
# {
#     rv <- rowVars(assay(object), useNames = T)
#     select <- order(rv, decreasing = TRUE)[seq_len(min(ntop, 
#         length(rv)))]
#     pca <- prcomp(t(assay(object[select, ])))
# }
# environment(genesDrivingPCs) <- asNamespace('DESeq2')
# vsd4PCA.2 <- vst(dds4PCA)
# DeseqTOP500 <- genesDrivingPCs(vsd4PCA.2)
# DeseqTOP500$rotation
# 
# pcaUpdateFromPCAtools.2 = function (mat, metadata = NULL, center = TRUE, scale = FALSE, 
#     rank = NULL, removeVar = NULL, transposed = FALSE, BSPARAM = ExactParam(), ntop = 500) 
# {
#     testing(mat)
#     
#     if (is.data.frame(mat)) {
#         mat <- as.matrix(mat)
#     }
#     if (!transposed) {
#         mat <- t(mat)
#     }
#     if (!is.null(metadata)) {
#         if (!identical(rownames(mat), rownames(metadata))) {
#             stop("'colnames(mat)' is not identical to 'rownames(metadata)'")
#         }
#     }
#     .center <- if (center) 
#         NULL
#     else 0
#     vars <- matrixStats::colVars(as.matrix(DelayedArray(mat)), useNames = TRUE, 
#                                  center = .center)
#     if (!is.null(removeVar)) {
#         message("-- removing the lower ", removeVar * 100, "% of variables based on variance")
#         varorder <- order(vars, decreasing = TRUE)
#         keep <- head(varorder, max(1, ncol(mat) * (1 - removeVar)))
#         mat <- mat[, keep, drop = FALSE]
#         vars <- vars[keep]
#     }
#     if (is.null(rank)) {
#         if (is(BSPARAM, "ExactParam")) {
#             rank <- min(dim(mat))
#         }
#         else {
#             stop("'rank' must be specified for approximate PCA methods")
#         }
#     }
#     pcaobj <- runPCA(mat, center = center, scale = scale, rank = rank, 
#         BSPARAM = BSPARAM)
#     if (scale) {
#         total.var <- length(vars)
#     }
#     else {
#         total.var <- sum(vars)
#     }
#     proportionvar <- (pcaobj$sdev^2)/total.var * 100
#     pcaobj <- list(rotated = data.frame(pcaobj$x), loadings = data.frame(pcaobj$rotation), 
#         variance = proportionvar, sdev = pcaobj$sdev, metadata = metadata, 
#         xvars = colnames(mat), yvars = rownames(mat), components = colnames(pcaobj$x))
#     rownames(pcaobj$rotated) <- pcaobj$yvars
#     rownames(pcaobj$loadings) <- pcaobj$xvars
#     names(pcaobj$variance) <- pcaobj$components
#     class(pcaobj) <- "pca"
#     return(pcaobj)
# }
# environment(pcaUpdateFromPCAtools.2) <- asNamespace('PCAtools')
# p2.2 <- pcaUpdateFromPCAtools.2(vsd4PCA, metadata = colData(myDat))
# 
# ntop = 500
# vars <- matrixStats::colVars(as.matrix(DelayedArray::DelayedArray(
#       order(vsd4PCA, decreasing = TRUE)[seq_len(min(ntop, 
#         length(vsd4PCA)))])), useNames = TRUE)
# test <- DelayedArray::DelayedArray(
#       order(vsd4PCA, decreasing = TRUE)[seq_len(min(ntop, 
#         length(vsd4PCA)))])
# 
# p2.2 <- pcaUpdateFromPCAtools(test, metadata = colData(myDat))
# colnames(DeseqTOP500$rotation)
# 
# testing = function (object, ntop = 500, returnData = T) 
# {
#     rv <- rowVars(object, useNames = T)
#     select <- order(rv, decreasing = TRUE)[seq_len(min(ntop, 
#         length(rv)))]
#     object[select,]
# }
# environment(testing) <- asNamespace('DESeq2')
# test <- testing(vsd4PCA)
# test2 <- vsd4PCA[test,]
# DeseqTOP500$rotation %>% View()
# p2.2$loadings
# p$loadings

## The above stuff is confusing. Don't try to understand it. It is me playing with code but i figured it out and that is below

#############################################################################
## OK so maybe 2 functions will have to be used
top500genes4PCA = function (object, ntop = 500, returnData = T) 
{
    rv <- rowVars(object, useNames = T)
    select <- order(rv, decreasing = TRUE)[seq_len(min(ntop, 
        length(rv)))]
    object[select,]
} # this function uses the assay(vst) data from PCAtools to filter only the top 500 genes. Then use those as input into the upadate 'pcaUpdateFromPCAtools' pca function
environment(top500genes4PCA) <- asNamespace('DESeq2') # important to state that this function is in the DESeq2 NameSpace because it has nested functions from that package within it
top500vsd4PCA <- top500genes4PCA(vsd4PCA)
p2 <- pcaUpdateFromPCAtools(top500vsd4PCA, metadata = colData(myDat))
p2$variance
moreThan70percentVarience <- which(cumsum(p2$variance) > 70)[1] # this give how many PCs account for more than 70% variance (12)
PCAtools::screeplot(p2,
          components = getComponents(p2),
          vline = moreThan70percentVarience) +
  geom_label(aes(x=moreThan70percentVarience + 0.5, y = 50,
                 label = ">70%", vjust = -1, size = 8))  # this is screeplot showing

#p2$variance %>% View() # This fixes the issues. Lets take this further and solve the rest
PCAtools::eigencorplot(p2,
             components = getComponents(p2, 1:12),
             metavars = c("Age","Diet","LFI","group"),
             col = c('darkblue','blue2','black','red2','darkred'),
             cexCorval = 0.7,
             colCorval = 'white',
             fontCorval = 2,
             posLab = 'bottomleft',
             rotLabX = 45,
             posColKey = 'top',
             cexLabColKey = 1.5,
             scale = T,
             main = 'PC1-12 trait correlations',
             colFrame = 'white',
             plotRsquared = F)

## This shows that Age is positively correlated to PC1, Diet is positively correlated to PC3, and LFI is positively correlated to PC6 

## past PC7 there is no more significant correlations of traits to PCs. I will cut all graphs off here

eigencorplot(p2, 
             components = getComponents(p2, 1:12),
             metavars = c("Age","Diet","LFI","group"),
             col = c('white','cornsilk1','gold','forestgreen','darkgreen'),
             cexCorval = 1.2,
             fontCorval = 2,
             posLab = 'all',
             rotLabX = 45,
             scale = T,
             main = bquote(Principal~component~Pearson~r^2~trait~correlates),
             plotRsquared = T,
             corFUN = 'pearson',
             corUSE = 'pairwise.complete.obs',
             corMultipleTestCorrection = 'BH',
             signifSymbols = c('****', '***', '**', '*', ''),
             signifCutpoints = c(0, 0.0001, 0.001, 0.1, 0.05, 1))
## This hows that r^2 is significant for Age and group to PC1 and Diet is significant for a negative correlation to PC6 and PC7 (meaningless for me) 
## So these graphs were the same
```

#defining new PC functions to get other PCs
```{r}
DESeq2:::plotPCA.DESeqTransform
plotPC3.4 = function (object, intgroup = "condition", ntop = 500, returnData = FALSE) 
{
    rv <- rowVars(assay(object))
    select <- order(rv, decreasing = TRUE)[seq_len(min(ntop, 
        length(rv)))]
    pca <- prcomp(t(assay(object)[select, ]))
    percentVar <- pca$sdev^2/sum(pca$sdev^2)
    if (!all(intgroup %in% names(colData(object)))) {
        stop("the argument 'intgroup' should specify columns of colData(dds)")
    }
    intgroup.df <- as.data.frame(colData(object)[, intgroup, 
        drop = FALSE])
    group <- if (length(intgroup) > 1) {
        factor(apply(intgroup.df, 1, paste, collapse = ":"))
    }
    else {
        colData(object)[[intgroup]]
    }
    d <- data.frame(PC3 = pca$x[, 3], PC4 = pca$x[, 4], group = group, 
        intgroup.df, name = colnames(object))
    if (returnData) {
        attr(d, "percentVar") <- percentVar[3:4]
        return(d)
    }
    ggplot(data = d, aes_string(x = "PC3", y = "PC4", color = "group")) + 
        geom_point(size = 3) + xlab(paste0("PC3: ", round(percentVar[3] * 
        100), "% variance")) + ylab(paste0("PC4: ", round(percentVar[4] * 
        100), "% variance")) + coord_fixed()
}
  
  

plotPC6.7 = function (object, intgroup = "condition", ntop = 500, returnData = FALSE) 
{
    rv <- rowVars(assay(object))
    select <- order(rv, decreasing = TRUE)[seq_len(min(ntop, 
        length(rv)))]
    pca <- prcomp(t(assay(object)[select, ]))
    percentVar <- pca$sdev^2/sum(pca$sdev^2)
    if (!all(intgroup %in% names(colData(object)))) {
        stop("the argument 'intgroup' should specify columns of colData(dds)")
    }
    intgroup.df <- as.data.frame(colData(object)[, intgroup, 
        drop = FALSE])
    group <- if (length(intgroup) > 1) {
        factor(apply(intgroup.df, 1, paste, collapse = ":"))
    }
    else {
        colData(object)[[intgroup]]
    }
    d <- data.frame(PC6 = pca$x[, 6], PC7 = pca$x[, 7], group = group, 
        intgroup.df, name = colnames(object))
    if (returnData) {
        attr(d, "percentVar") <- percentVar[6:7]
        return(d)
    }
    ggplot(data = d, aes_string(x = "PC6", y = "PC7", color = "group")) + 
        geom_point(size = 3) + xlab(paste0("PC6: ", round(percentVar[6] * 
        100), "% variance")) + ylab(paste0("PC7: ", round(percentVar[7] * 
        100), "% variance")) + coord_fixed()
}


plotPC8.9 = function (object, intgroup = "condition", ntop = 500, returnData = FALSE) 
{
    rv <- rowVars(assay(object))
    select <- order(rv, decreasing = TRUE)[seq_len(min(ntop, 
        length(rv)))]
    pca <- prcomp(t(assay(object)[select, ]))
    percentVar <- pca$sdev^2/sum(pca$sdev^2)
    if (!all(intgroup %in% names(colData(object)))) {
        stop("the argument 'intgroup' should specify columns of colData(dds)")
    }
    intgroup.df <- as.data.frame(colData(object)[, intgroup, 
        drop = FALSE])
    group <- if (length(intgroup) > 1) {
        factor(apply(intgroup.df, 1, paste, collapse = ":"))
    }
    else {
        colData(object)[[intgroup]]
    }
    d <- data.frame(PC8 = pca$x[, 8], PC9 = pca$x[, 9], group = group, 
        intgroup.df, name = colnames(object))
    if (returnData) {
        attr(d, "percentVar") <- percentVar[8:9]
        return(d)
    }
    ggplot(data = d, aes_string(x = "PC8", y = "PC9", color = "group")) + 
        geom_point(size = 3) + xlab(paste0("PC8: ", round(percentVar[8] * 
        100), "% variance")) + ylab(paste0("PC9: ", round(percentVar[9] * 
        100), "% variance")) + coord_fixed()
}
environment(plotPC3.4) <- asNamespace('DESeq2')
environment(plotPC6.7) <- asNamespace('DESeq2')
environment(plotPC8.9) <- asNamespace('DESeq2')
```

#### PCA DESeq2
```{r}
vsd_V3.3 <- vst(dds2_V3)
pc12_V3.3 <- plotPCA(vsd_V3.3, intgroup = c("Age","LFI","Diet","group"), returnData = T) # PC1=26%, PC2=13%
pc34_V3.3 <- plotPC3.4(vsd_V3.3, intgroup = c("Age","LFI","Diet","group"), returnData = T) # PC3=07%, PC4=05%
pc67_V3.3 <- plotPC6.7(vsd_V3.3, intgroup = c("Age","LFI","Diet","group"), returnData = T) # PC6=03%, PC7=03%
pc89_V3.3 <- plotPC8.9(vsd_V3.3, intgroup = c("Age","LFI","Diet","group"), returnData = T) # PC8=03%, PC9=02%)

# Age
ggplot(pc12_V3.3, aes(x=PC1, y=PC2, color=Age)) +
  geom_point(aes(fill=Age), 
       colour="black",pch=21, size=5)  +
  scale_fill_manual(values = c("red","blue")) +
  xlab(paste0("PC1: 26% varience")) +
  ylab(paste0("PC2: 13% variance")) +
  coord_fixed() +
  scale_color_manual(labels = c("Younger","Aged"),
                    values = c("red","blue")) +
  #ggtitle("PCA with VST data") +
  stat_ellipse(level = 0.95) +
  theme_bw() +
  guides(fill=F) +
  theme(axis.text.x=element_text(size=18), 
        axis.title.x=element_text(size=18),
        axis.text.y=element_text(size=18), 
        axis.title.y=element_text(size=18),
        legend.title = element_text(size=16),
        legend.key.size = unit(0.6, 'cm'),
        legend.position = c(0.91, 0.91)) +
  scale_x_continuous(limits = c(-25, 25)) +
  scale_y_continuous(limits = c(-25, 25))

# Diet
ggplot(pc34_V3.3, aes(x=PC3, y=PC4, color=Diet)) +
  geom_point(aes(fill=Diet), 
       colour="black",pch=21, size=5)  +
  scale_fill_manual(values = c("gold","turquoise")) +
  xlab(paste0("PC3: 7% varience")) +
  ylab(paste0("PC4: 5% variance")) +
  coord_fixed() +
  scale_color_manual(labels = c("AL","CR"),
                    values = c("gold","turquoise")) +
  #ggtitle("PCA with VST data") +
  stat_ellipse(level = 0.95) +
  theme_bw() +
  guides(fill=F) +
  theme(axis.text.x=element_text(size=18), 
        axis.title.x=element_text(size=18),
        axis.text.y=element_text(size=18), 
        axis.title.y=element_text(size=18),
        legend.title = element_text(size=16),
        legend.key.size = unit(0.6, 'cm'),
        legend.position = c(0.91, 0.91)) +
  scale_x_continuous(limits = c(-15, 15)) +
  scale_y_continuous(limits = c(-15, 15))


# Diet
ggplot(pc89_V3.3, aes(x=PC8, y=PC9, color=Diet)) +
  geom_point(aes(fill=Diet), 
       colour="black",pch=21, size=5)  +
  scale_fill_manual(values = c("gold","turquoise")) +
  xlab(paste0("PC8: 3% varience")) +
  ylab(paste0("PC9: 2% variance")) +
  coord_fixed() +
  scale_color_manual(labels = c("AL","CR"),
                    values = c("gold","turquoise")) +
  #ggtitle("PCA with VST data") +
  stat_ellipse(level = 0.95) +
  theme_bw() +
  guides(fill=F) +
  theme(axis.text.x=element_text(size=18), 
        axis.title.x=element_text(size=18),
        axis.text.y=element_text(size=18), 
        axis.title.y=element_text(size=18),
        legend.title = element_text(size=16),
        legend.key.size = unit(0.6, 'cm'),
        legend.position = c(0.91, 0.91)) +
  scale_x_continuous(limits = c(-15, 15)) +
  scale_y_continuous(limits = c(-15, 15))

# LFI
## did not correlate when using top 500 genes....
ggplot(pc67_V3.3, aes(x=PC6, y=PC7, color=LFI)) +
  geom_point(aes(fill=LFI), 
       colour="black",pch=21, size=5)  +
  scale_fill_manual(values = c("forestgreen","darkred")) +
  xlab(paste0("PC6: 3% varience")) +
  ylab(paste0("PC7: 3% variance")) +
  coord_fixed() +
  scale_color_manual(labels = c("Untrained","Trained"),
                     values = c("forestgreen","darkred")) +
  #ggtitle("PCA with VST data") +
  stat_ellipse(level = 0.95) +
  theme_bw() +
  guides(fill=F) +
  theme(axis.text.x=element_text(size=18), 
        axis.title.x=element_text(size=18),
        axis.text.y=element_text(size=18), 
        axis.title.y=element_text(size=18),
        legend.title = element_text(size=16),
        legend.key.size = unit(0.6, 'cm'),
        legend.position = c(0.91, 0.91)) +
  scale_x_continuous(limits = c(-10, 10)) +
  scale_y_continuous(limits = c(-15, 15))
```
#### Graph it all
```{r}
## Lets make an image with all this info in one
pscree <- PCAtools::screeplot(p2,
          components = getComponents(p2, 1:20),
          vline = moreThan70percentVarience) +
  geom_label(aes(x=moreThan70percentVarience + 0.5, y = 50,
                 label = ">70%", vjust = -1, size = 8)) 

peigencor <- PCAtools::eigencorplot(p2,
             components = getComponents(p2, 1:12),
             metavars = c("Age","Diet","LFI","group"),
             col = c('darkblue','blue2','black','red2','darkred'),
             cexCorval = 1.0,
             colCorval = 'white',
             fontCorval = 2,
             posLab = 'bottomleft',
             rotLabX = 45,
             posColKey = 'top',
             cexLabColKey = 1.5,
             scale = T,
             main = 'PC1-12 trait correlations',
             colFrame = 'white',
             plotRsquared = F,
             returnPlot = F)

pRsquare <- eigencorplot(p2, 
             components = getComponents(p2, 1:12),
             metavars = c("Age","Diet","LFI","group"),
             col = c('white','cornsilk1','gold','forestgreen','darkgreen'),
             cexCorval = 1.0,
             fontCorval = 2,
             posLab = 'all',
             rotLabX = 45,
             scale = T,
             main = bquote(Principal~component~Pearson~r^2~trait~correlates),
             plotRsquared = T,
             corFUN = 'pearson',
             corUSE = 'pairwise.complete.obs',
             corMultipleTestCorrection = 'BH',
             signifSymbols = c('****', '***', '**', '*', ''),
             signifCutpoints = c(0, 0.0001, 0.001, 0.1, 0.05, 1),
             returnPlot = F)

PC1.2 <- ggplot(pc12_V3.3, aes(x=PC1, y=PC2, color=Age)) +
  geom_point(aes(fill=Age), 
       colour="black",pch=21, size=5)  +
  scale_fill_manual(values = c("red","blue")) +
  xlab(paste0("PC1: 26% varience")) +
  ylab(paste0("PC2: 13% variance")) +
  coord_fixed() +
  scale_color_manual(labels = c("Younger","Aged"),
                    values = c("red","blue")) +
  #ggtitle("PCA with VST data") +
  stat_ellipse(level = 0.95) +
  theme_bw() +
  guides(fill=F) +
  theme(axis.text.x=element_text(size=18), 
        axis.title.x=element_text(size=18),
        axis.text.y=element_text(size=18), 
        axis.title.y=element_text(size=18),
        legend.title = element_text(size=16),
        legend.key.size = unit(0.6, 'cm'),
        legend.position = c(0.91, 0.91)) +
  scale_x_continuous(limits = c(-25, 25)) +
  scale_y_continuous(limits = c(-25, 25))

PC3.4 <- ggplot(pc34_V3.3, aes(x=PC3, y=PC4, color=Diet)) +
  geom_point(aes(fill=Diet), 
       colour="black",pch=21, size=5)  +
  scale_fill_manual(values = c("gold","turquoise")) +
  xlab(paste0("PC3: 7% varience")) +
  ylab(paste0("PC4: 5% variance")) +
  coord_fixed() +
  scale_color_manual(labels = c("AL","CR"),
                    values = c("gold","turquoise")) +
  #ggtitle("PCA with VST data") +
  stat_ellipse(level = 0.95) +
  theme_bw() +
  guides(fill=F) +
  theme(axis.text.x=element_text(size=18), 
        axis.title.x=element_text(size=18),
        axis.text.y=element_text(size=18), 
        axis.title.y=element_text(size=18),
        legend.title = element_text(size=16),
        legend.key.size = unit(0.6, 'cm'),
        legend.position = c(0.91, 0.91)) +
  scale_x_continuous(limits = c(-15, 15)) +
  scale_y_continuous(limits = c(-15, 15))

PC8.9 <- ggplot(pc89_V3.3, aes(x=PC8, y=PC9, color=Diet)) +
  geom_point(aes(fill=Diet), 
       colour="black",pch=21, size=5)  +
  scale_fill_manual(values = c("gold","turquoise")) +
  xlab(paste0("PC8: 3% varience")) +
  ylab(paste0("PC9: 2% variance")) +
  coord_fixed() +
  scale_color_manual(labels = c("AL","CR"),
                    values = c("gold","turquoise")) +
  #ggtitle("PCA with VST data") +
  stat_ellipse(level = 0.95) +
  theme_bw() +
  guides(fill=F) +
  theme(axis.text.x=element_text(size=18), 
        axis.title.x=element_text(size=18),
        axis.text.y=element_text(size=18), 
        axis.title.y=element_text(size=18),
        legend.title = element_text(size=16),
        legend.key.size = unit(0.6, 'cm'),
        legend.position = c(0.91, 0.91)) +
  scale_x_continuous(limits = c(-15, 15)) +
  scale_y_continuous(limits = c(-15, 15))

library(cowplot)
library(ggplotify)

top_row <- plot_grid(pscree, as.grob(peigencor), as.grob(pRsquare),
                     ncol = 3,
                     lables = c('A','B','C'),
                     label_fontfamily = 'serif',
                     label_fontface = 'bold',
                     label_size = 22,
                     align = 'h',
                     rel_widths = c(1.10, 1.10, 1.10)) +
  theme(plot.margin = unit(c(5, 0.1, 0.1, 0.1), "cm"))

bottom_row <- plot_grid(as.grob(PC1.2), as.grob(PC3.4), as.grob(PC8.9),
                        ncol = 3, 
                        labels = c('D','E','F'),
                        label_fontfamily = 'serif',
                        label_fontface = 'bold',
                        label_size = 22,
                        align = 'h',
                        rel_widths = c(0.8, 0.8, 0.8)) +
  theme(plot.margin = unit(c(0.1, 0.1, 5.0, 0.1), "cm"))

plot_grid(top_row, bottom_row, ncol = 1, align = 'v', rel_heights = c(3.6, 1.3))
```

# __________________
## DESeq2
#####DEGs
```{r}
dds2_V3.2 <- estimateSizeFactors(dds2_V3)
dds2_V3.2 <- DESeq(dds2_V3.2)

## Age differences
# AUAL vs MUAL
summary(results(dds2_V3.2, contrast = c("group", "AUAL", "MUAL"), alpha = 0.05, lfcThreshold = 0))
AUAL_vs_MUALresSig <- subset(results(dds2_V3.2, contrast = c("group", "AUAL", "MUAL"), lfcThreshold = 0), padj < 0.05)
#############################################################################
## LFI in Mature
# MTAL vs MUAL
summary(results(dds2_V3.2, contrast = c("group", "MTAL", "MUAL"), alpha = 0.05, lfcThreshold = 0))
MTAL_vs_MUALresSig <- subset(results(dds2_V3.2, contrast = c("group", "MTAL", "MUAL"), lfcThreshold = 0), padj < 0.05)
#############################################################################
## LFI in Aged
# ATAL vs AUAL
summary(results(dds2_V3.2, contrast = c("group", "ATAL", "AUAL"), alpha = 0.05, lfcThreshold = 0))
ATAL_vs_AUALresSig <- subset(results(dds2_V3.2, contrast = c("group", "ATAL", "AUAL"), lfcThreshold = 0), padj < 0.05)
#############################################################################
## Age under CR
# AUCR vs MUCR
summary(results(dds2_V3.2, contrast = c("group", "AUCR", "MUCR"), alpha = 0.05, lfcThreshold = 0))
AUCR_vs_MUCRresSig <- subset(results(dds2_V3.2, contrast = c("group", "AUCR", "MUCR"), lfcThreshold = 0), padj < 0.05)
#############################################################################
## LFI in Mature CR
# MTCR vs MUCR
summary(results(dds2_V3.2, contrast = c("group", "MTCR", "MUCR"), alpha = 0.05, lfcThreshold = 0))
MTCR_vs_MUCRresSig <- subset(results(dds2_V3.2, contrast = c("group", "MTCR", "MUCR"), lfcThreshold = 0), padj < 0.05)
#############################################################################
## LFI in Aged CR
# ATCR vs AUCR
summary(results(dds2_V3.2, contrast = c("group", "ATCR", "AUCR"), alpha = 0.05, lfcThreshold = 0))
ATCR_vs_AUCRresSig <- subset(results(dds2_V3.2, contrast = c("group", "ATCR", "AUCR"), lfcThreshold = 0), padj < 0.05)



#########################################################################
## I will plot ALL DEGs
# first make a dataframe with all the DEGS
AUAL_vs_MUAL_DEGl2fc <- AUAL_vs_MUALresSig[2] %>% as.data.frame() %>%
  rownames_to_column() %>% set_names(c("Gene", "AUAL_vs_MUAL_l2fc"))
MTAL_vs_MUALDEGl2fc <- MTAL_vs_MUALresSig[2] %>% as.data.frame() %>%
  rownames_to_column() %>% set_names(c("Gene", "MTAL_vs_MUAL_l2fc"))
ATAL_vs_AUALDEGl2fc <- ATAL_vs_AUALresSig[2] %>% as.data.frame() %>%
  rownames_to_column() %>% set_names(c("Gene", "ATAL_vs_AUAL_l2fc"))
AUCR_vs_MUCRDEGl2fc <- AUCR_vs_MUCRresSig[2] %>% as.data.frame() %>%
  rownames_to_column() %>% set_names(c("Gene", "AUCR_vs_MUCR_l2fc"))
MTCR_vs_MUCRDEGl2fc <- MTCR_vs_MUCRresSig[2] %>% as.data.frame() %>%
  rownames_to_column() %>% set_names(c("Gene", "MTCR_vs_MUCR_l2fc"))
ATCR_vs_AUCRDEGl2fc <- ATCR_vs_AUCRresSig[2] %>% as.data.frame() %>%
  rownames_to_column() %>% set_names(c("Gene", "ATCR_vs_AUCR_l2fc"))

listInputs_1 <- list("AUAL_vs_MUAL_V3.2" = AUAL_vs_MUAL_DEGl2fc$Gene,
                   "MTAL_vs_MUAL_V3.2" = MTAL_vs_MUALDEGl2fc$Gene,
                   "ATAL_vs_AUAL_V3.2" = ATAL_vs_AUALDEGl2fc$Gene,
                   "AUCR_vs_MUCR_V3.2" = AUCR_vs_MUCRDEGl2fc$Gene,
                   "MTCR_vs_MUCR_V3.2" = MTCR_vs_MUCRDEGl2fc$Gene,
                   "ATCR_vs_AUCR_V3.2" = ATCR_vs_AUCRDEGl2fc$Gene)
upset(fromList(listInputs_1), sets=c("AUAL_vs_MUAL_V3.2", "MTAL_vs_MUAL_V3.2", "ATAL_vs_AUAL_V3.2", "AUCR_vs_MUCR_V3.2", "MTCR_vs_MUCR_V3.2", "ATCR_vs_AUCR_V3.2"), order.by = "degree")

AllDEGs <- full_join(AUAL_vs_MUAL_DEGl2fc, MTAL_vs_MUALDEGl2fc, by = "Gene")
AllDEGs <- full_join(AllDEGs, ATAL_vs_AUALDEGl2fc, by = "Gene")
AllDEGs <- full_join(AllDEGs, AUCR_vs_MUCRDEGl2fc, by = "Gene")
AllDEGs <- full_join(AllDEGs, MTCR_vs_MUCRDEGl2fc, by = "Gene")
AllDEGs <- full_join(AllDEGs, ATCR_vs_AUCRDEGl2fc, by = "Gene")

#write.xlsx(AllDEGs,"/Users/school/Documents/UAB/UAB009/Results/AllDEGs.xlsx", sheetName = "All_comps", append = T, row.names = F)
```



#####GO
```{r}
library(clusterProfiler)
library(org.Acalifornica.eg.db)
## I was having a problem with reproduceability while running gene-set enrichment below. I was getting different numbers of gene ontology terms each time I re-ran the assay. So I looked it up and there may be some sort of seed issue (some randomness) that changes each time I run it: (https://support.bioconductor.org/p/99810/), (https://github.com/YuLab-SMU/clusterProfiler/issues/190).
### I will attempt to set the seed using 'set.seed()' (https://www.statology.org/set-seed-in-r/)
set.seed(1234) # setting the seed to make the below code reproducible
## Age differences
# AUAL vs MUAL
AUAL_MUAL_GO <- results(dds2_V3.2, contrast = c("group", "AUAL", "MUAL"), alpha = 0.05, lfcThreshold = 0)
AgeAL_GO <- as.data.frame(AUAL_MUAL_GO)
AgeAL_GO$Gene <- rownames(AgeAL_GO)

original_gene_list_AgeAL <- AgeAL_GO$log2FoldChange
names(original_gene_list_AgeAL) <- AgeAL_GO$Gene
gene_list_AgeAL <- na.omit(original_gene_list_AgeAL)
gene_list_AgeAL = sort(gene_list_AgeAL, decreasing = T)

gse_AgeAL <- gseGO(geneList = gene_list_AgeAL, #[1st pass: 185, 2nd pass: 189] (before seed was set), 3rd pass: 209 (after seed.set) 4th pass: 209 again! (I think this worked) 5th pass: 209 terms (after seed.set)
                   ont = "ALL",
                   keyType = "SYMBOL",
                   minGSSize = 3,
                   maxGSSize = 800,
                   pvalueCutoff = 0.05,
                   verbose = TRUE,
                   OrgDb = org.Acalifornica.eg.db,
                   pAdjustMethod = "none",
                   seed = TRUE)

require(DOSE)
dotplot(gse_AgeAL, showCategory=10, split=".sign") + facet_grid(.~.sign) # shows activated and suppressed
dotplot(gse_AgeAL, showCategory=10, split="ONTOLOGY") + facet_grid(ONTOLOGY~., scale="free") # shows the ontology (CC, BP, MF)

ggplot(gse_AgeAL, showCategory=10, split="ONTOLOGY", aes(x = NES, y = Description, fill=qvalue)) +
  geom_col() + facet_grid(ONTOLOGY~., scale="free") +
  theme_bw()# I think this the graph I want! This shows me the top whatever I choose in 'showCategory' the GO are normalized so if the bar goes to negative they are suppressed and if bar goes positive they are activated and this is split by ontology CC, MF, and BP
## This basically combines the two graphs above
## Saved as PDF 6x6 (landscape)

#############################################################################
## LFI in Mature
# MTAL vs MUAL
MTAL_MUAL_GO <- results(dds2_V3.2, contrast = c("group", "MTAL", "MUAL"), alpha = 0.05, lfcThreshold = 0)
LFIyoungAL_GO <- as.data.frame(MTAL_MUAL_GO)
LFIyoungAL_GO$Gene <- rownames(LFIyoungAL_GO) #11,852 genes

original_gene_list_LFIyoungAL <- LFIyoungAL_GO$log2FoldChange #11,852 genes
names(original_gene_list_LFIyoungAL) <- LFIyoungAL_GO$Gene #11,852 genes
gene_list_LFIyoungAL <- na.omit(original_gene_list_LFIyoungAL) #11,852 genes
gene_list_LFIyoungAL = sort(gene_list_LFIyoungAL, decreasing = T) #11,852 genes

gse_LFIyoungAL <- gseGO(geneList = gene_list_LFIyoungAL, # 142 terms after seed.set
                   ont = "ALL",
                   keyType = "SYMBOL",
                   minGSSize = 3,
                   maxGSSize = 800,
                   pvalueCutoff = 0.05,
                   verbose = TRUE,
                   OrgDb = org.Acalifornica.eg.db,
                   pAdjustMethod = "none",
                   seed = TRUE) 
## graph
ggplot(gse_LFIyoungAL, showCategory=10, split="ONTOLOGY", aes(x = NES, y = Description, fill=qvalue)) +
  geom_col() + facet_grid(ONTOLOGY~., scale="free") +
  theme_bw() ## Saved as PDF 6x7 (landscape)
#############################################################################
## LFI in Aged
# ATAL vs AUAL
ATAL_AUAL_GO <- results(dds2_V3.2, contrast = c("group", "ATAL", "AUAL"), alpha = 0.05, lfcThreshold = 0)
LFIagedAL_GO <- as.data.frame(ATAL_AUAL_GO)
LFIagedAL_GO$Gene <- rownames(LFIagedAL_GO) #11,852 genes

original_gene_list_LFIagedAL <- LFIagedAL_GO$log2FoldChange #11,852 genes
names(original_gene_list_LFIagedAL) <- LFIagedAL_GO$Gene #11,852 genes
gene_list_LFIagedAL <- na.omit(original_gene_list_LFIagedAL) #11,852 genes
gene_list_LFIagedAL = sort(gene_list_LFIagedAL, decreasing = T) #11,852 genes

gse_LFIagedAL <- gseGO(geneList = gene_list_LFIagedAL, #188 GO terms after seed.set
                   ont = "ALL",
                   keyType = "SYMBOL",
                   minGSSize = 3,
                   maxGSSize = 800,
                   pvalueCutoff = 0.05,
                   verbose = TRUE,
                   OrgDb = org.Acalifornica.eg.db,
                   pAdjustMethod = "none",
                   seed = TRUE) 
## graph
ggplot(gse_LFIagedAL, showCategory=10, split="ONTOLOGY", aes(x = NES, y = Description, fill=qvalue)) +
  geom_col() + facet_grid(ONTOLOGY~., scale="free") +
  theme_bw() ## Saved as PDF 6x6 (landscape)
#############################################################################
## Age under CR
# AUCR vs MUCR
AUCR_MUCR_GO <- results(dds2_V3.2, contrast = c("group", "AUCR", "MUCR"), alpha = 0.05, lfcThreshold = 0)
AgeCR_GO <- as.data.frame(AUCR_MUCR_GO)
AgeCR_GO$Gene <- rownames(AgeCR_GO)

original_gene_list_AgeCR <- AgeCR_GO$log2FoldChange
names(original_gene_list_AgeCR) <- AgeCR_GO$Gene
gene_list_AgeCR <- na.omit(original_gene_list_AgeCR)
gene_list_AgeCR = sort(gene_list_AgeCR, decreasing = T)

gse_AgeCR <- gseGO(geneList = gene_list_AgeCR, #298 GO terms after seed.set
                   ont = "ALL",
                   keyType = "SYMBOL",
                   minGSSize = 3,
                   maxGSSize = 800,
                   pvalueCutoff = 0.05,
                   verbose = TRUE,
                   OrgDb = org.Acalifornica.eg.db,
                   pAdjustMethod = "none",
                   seed = TRUE)
## graph
ggplot(gse_AgeCR, showCategory=10, split="ONTOLOGY", aes(x = NES, y = Description, fill=qvalue)) +
  geom_col() + facet_grid(ONTOLOGY~., scale="free") +
  theme_bw() ## Saved as PDF 6x6 (landscape)
#############################################################################
## LFI in Mature CR
# MTCR vs MUCR
MTCR_MUCR_GO <- results(dds2_V3.2, contrast = c("group", "MTCR", "MUCR"), alpha = 0.05, lfcThreshold = 0)
LFIyoungCR_GO <- as.data.frame(MTCR_MUCR_GO)
LFIyoungCR_GO$Gene <- rownames(LFIyoungCR_GO)

original_gene_list_LFIyoungCR <- LFIyoungCR_GO$log2FoldChange
names(original_gene_list_LFIyoungCR) <- LFIyoungCR_GO$Gene
gene_list_LFIyoungCR <- na.omit(original_gene_list_LFIyoungCR)
gene_list_LFIyoungCR = sort(gene_list_LFIyoungCR, decreasing = T)

gse_LFIyoungCR <- gseGO(geneList = gene_list_LFIyoungCR, # 117 terms after seed.set
                   ont = "ALL",
                   keyType = "SYMBOL",
                   minGSSize = 3,
                   maxGSSize = 800,
                   pvalueCutoff = 0.05,
                   verbose = TRUE,
                   OrgDb = org.Acalifornica.eg.db,
                   pAdjustMethod = "none",
                   seed = TRUE)
## graph
ggplot(gse_LFIyoungCR, showCategory=10, split="ONTOLOGY", aes(x = NES, y = Description, fill=qvalue)) +
  geom_col() + facet_grid(ONTOLOGY~., scale="free") +
  theme_bw() ## Saved as PDF 6x9 (landscape)
#############################################################################
## LFI in Aged CR
# ATCR vs AUCR
ATCR_AUCR_GO <- results(dds2_V3.2, contrast = c("group", "ATCR", "AUCR"), alpha = 0.05, lfcThreshold = 0)
LFIageCR_GO <- as.data.frame(ATCR_AUCR_GO)
LFIageCR_GO$Gene <- rownames(LFIageCR_GO)

original_gene_list_LFIageCR <- LFIageCR_GO$log2FoldChange
names(original_gene_list_LFIageCR) <- LFIageCR_GO$Gene
gene_list_LFIageCR <- na.omit(original_gene_list_LFIageCR)
gene_list_LFIageCR = sort(gene_list_LFIageCR, decreasing = T)

gse_LFIageCR <- gseGO(geneList = gene_list_LFIageCR, #181 terms after seed.set
                   ont = "ALL",
                   keyType = "SYMBOL",
                   minGSSize = 3,
                   maxGSSize = 800,
                   pvalueCutoff = 0.05,
                   verbose = TRUE,
                   OrgDb = org.Acalifornica.eg.db,
                   pAdjustMethod = "none",
                   seed = TRUE)
## graph
ggplot(gse_LFIageCR, showCategory=10, split="ONTOLOGY", aes(x = NES, y = Description, fill=qvalue)) +
  geom_col() + facet_grid(ONTOLOGY~., scale="free") +
  theme_bw() ## Saved as PDF 6x5 (landscape)
```





#####KEGG
# making 'protein to gene to transcript' dataframes for Aplysia
```{r}
## Make protein to gene to transcript dataframe for fish (X_maculatus)
# load in my gff
Apl_GFF = read.delim("/Users/school/Documents/UAB/UAB009/genome/AplCal3.0_GFF.gff", header = F, stringsAsFactors = F, comment.char = "#")
# create a dataframe of protein, gene, and transcript
gene2tx2protein_Apl <- inner_join(
# extract transcript to protein mapping
Apl_GFF %>%
  filter(V3=="CDS") %>%
  select(V9) %>%
  rowwise() %>%
  mutate(transcript = str_extract(V9,"Parent=rna-[aA-zZ,0-9]*") %>% str_remove(.,"Parent=rna-"),
         protein = str_extract(V9,"protein_id=[aA-zZ,0-9]*") %>% str_remove(.,"protein_id=")) %>%
  select(-V9) %>%
  distinct(),
# now extract gene to transcript mapping
Apl_GFF %>%
  filter(V3=="mRNA") %>%
  select(V9) %>%
  rowwise() %>%
  mutate(gene = str_extract(V9,"Parent=gene-[aA-zZ,0-9]*") %>% str_remove(.,"Parent=gene-"),
         transcript = str_extract(V9,"transcript_id=[aA-zZ,0-9]*") %>% str_remove(.,"transcript_id="),
         tx_version=str_extract(V9,"Name=[aA-zZ,0-9,.]*") %>% str_remove(.,"Name=")) %>%
  select(-V9) %>%
  distinct()
)
```

```{r}
AplysiaKOterms <- read.delim("/Users/school/Documents/UAB/UAB009/GhostKOALA/user_ko.txt", header = F, col.names = c("protein", "ko")) %>% as.data.frame() %>% mutate(protein = gsub("\\.[0-9]","", protein)) # 26,659 entries
gene2tx2protein2ko_Apl <- full_join(gene2tx2protein_Apl, AplysiaKOterms, by="protein")

## Age differences
# AUAL vs MUAL
gene_list_AgeAL_df <- as.data.frame(gene_list_AgeAL) %>% rownames_to_column("gene") #11,852 genes
AgeAL_KEGG <- merge(gene_list_AgeAL_df, gene2tx2protein2ko_Apl, by="gene") #17,940 genes
AgeAL_KEGG2 <- AgeAL_KEGG[!duplicated(AgeAL_KEGG$gene),] #11,422 genes
df_AgeAL = AgeAL_GO[AgeAL_GO$Gene %in% AgeAL_KEGG2$gene,] #11,422 genes
df_AgeAL$ko = AgeAL_KEGG2$ko #11,422 genes (just added ko column here)
AgeAL_keggGeneList <- df_AgeAL$log2FoldChange #11,422 genes (made list of lfc)
names(AgeAL_keggGeneList) <- df_AgeAL$ko #11,422 genes (named all lfc with ko for that gene)
AgeAL_keggGeneList <- na.omit(AgeAL_keggGeneList) #11,422 genes (no NA)
AgeAL_keggGeneList = sort(AgeAL_keggGeneList, decreasing = TRUE) #(sorted the lfc in decreasing order)

kk2_AgeAL <- gseKEGG(geneList = AgeAL_keggGeneList,
                     organism = "ko", #used ko terms from ghostKOWALA since Aplysia is not and available organism on KEGG website
                     minGSSize = 3,
                     maxGSSize = 800,
                     pvalueCutoff = 0.05,
                     pAdjustMethod = "none",
                     keyType = "kegg",
                     seed = TRUE)

dotplot(kk2_AgeAL, showCategory = 10, title = "Enriched Pathways" , split=".sign") + facet_grid(.~.sign) ## Saved as PDF 6x6 (landscape)

#############################################################################
## LFI in Mature
# MTAL vs MUAL
gene_list_LFIyoungAL_df <- as.data.frame(gene_list_LFIyoungAL) %>% rownames_to_column("gene") #11,852 genes
LFIyoungAL_KEGG <- merge(gene_list_LFIyoungAL_df, gene2tx2protein2ko_Apl, by="gene") #17,940 genes
LFIyoungAL_KEGG2 <- LFIyoungAL_KEGG[!duplicated(LFIyoungAL_KEGG$gene),] #11,422 genes
df_LFIyoungAL = LFIyoungAL_GO[LFIyoungAL_GO$Gene %in% LFIyoungAL_KEGG2$gene,] #11,422 genes
df_LFIyoungAL$ko = LFIyoungAL_KEGG2$ko #11,422 genes (just added ko column here)
LFIyoungAL_keggGeneList <- df_LFIyoungAL$log2FoldChange #11,422 genes (made list of lfc)
names(LFIyoungAL_keggGeneList) <- df_LFIyoungAL$ko #11,422 genes (named all lfc with ko for that gene)
LFIyoungAL_keggGeneList <- na.omit(LFIyoungAL_keggGeneList) #11,422 genes (no NA)
LFIyoungAL_keggGeneList = sort(LFIyoungAL_keggGeneList, decreasing = TRUE) #(sorted the lfc in decreasing order)

kk2_LFIyoungAL <- gseKEGG(geneList = LFIyoungAL_keggGeneList,
                     organism = "ko", #used ko terms from ghostKOWALA since Aplysia is not and available organism on KEGG website
                     minGSSize = 3,
                     maxGSSize = 800,
                     pvalueCutoff = 0.05,
                     pAdjustMethod = "none",
                     keyType = "kegg",
                     seed = TRUE)

dotplot(kk2_LFIyoungAL, showCategory = 10, title = "Enriched Pathways" , split=".sign") + facet_grid(.~.sign) ## Saved as PDF 6x6 (landscape)

#############################################################################
## LFI in Aged
# ATAL vs AUAL
gene_list_LFIagedAL_df <- as.data.frame(gene_list_LFIagedAL) %>% rownames_to_column("gene") #11,852 genes
LFIagedAL_KEGG <- merge(gene_list_LFIagedAL_df, gene2tx2protein2ko_Apl, by="gene") #17,940 genes
LFIagedAL_KEGG2 <- LFIagedAL_KEGG[!duplicated(LFIagedAL_KEGG$gene),] #11,422 genes
df_LFIagedAL = LFIagedAL_GO[LFIagedAL_GO$Gene %in% LFIagedAL_KEGG2$gene,] #11,422 genes
df_LFIagedAL$ko = LFIagedAL_KEGG2$ko #11,422 genes (just added ko column here)
LFIagedAL_keggGeneList <- df_LFIagedAL$log2FoldChange #11,422 genes (made list of lfc)
names(LFIagedAL_keggGeneList) <- df_LFIagedAL$ko #11,422 genes (named all lfc with ko for that gene)
LFIagedAL_keggGeneList <- na.omit(LFIagedAL_keggGeneList) #11,422 genes (no NA)
LFIagedAL_keggGeneList = sort(LFIagedAL_keggGeneList, decreasing = TRUE) #(sorted the lfc in decreasing order)

kk2_LFIagedAL <- gseKEGG(geneList = LFIagedAL_keggGeneList,
                     organism = "ko", #used ko terms from ghostKOWALA since Aplysia is not and available organism on KEGG website
                     minGSSize = 3,
                     maxGSSize = 800,
                     pvalueCutoff = 0.05,
                     pAdjustMethod = "none",
                     keyType = "kegg",
                     seed = TRUE)

dotplot(kk2_LFIagedAL, showCategory = 10, title = "Enriched Pathways" , split=".sign") + facet_grid(.~.sign) ## Saved as PDF 6x6 (landscape)

#############################################################################
## Age under CR
# AUCR vs MUCR
gene_list_AgeCR_df <- as.data.frame(gene_list_AgeCR) %>% rownames_to_column("gene") #11,852 genes
AgeCR_KEGG <- merge(gene_list_AgeCR_df, gene2tx2protein2ko_Apl, by="gene") #17,940 genes
AgeCR_KEGG2 <- AgeCR_KEGG[!duplicated(AgeCR_KEGG$gene),] #11,422 genes
df_AgeCR = AgeCR_GO[AgeCR_GO$Gene %in% AgeCR_KEGG2$gene,] #11,422 genes
df_AgeCR$ko = AgeCR_KEGG2$ko #11,422 genes (just added ko column here)
AgeCR_keggGeneList <- df_AgeCR$log2FoldChange #11,422 genes (made list of lfc)
names(AgeCR_keggGeneList) <- df_AgeCR$ko #11,422 genes (named all lfc with ko for that gene)
AgeCR_keggGeneList <- na.omit(AgeCR_keggGeneList) #11,422 genes (no NA)
AgeCR_keggGeneList = sort(AgeCR_keggGeneList, decreasing = TRUE) #(sorted the lfc in decreasing order)

kk2_AgeCR <- gseKEGG(geneList = AgeCR_keggGeneList,
                     organism = "ko", #used ko terms from ghostKOWALA since Aplysia is not and available organism on KEGG website
                     minGSSize = 3,
                     maxGSSize = 800,
                     pvalueCutoff = 0.05,
                     pAdjustMethod = "none",
                     keyType = "kegg",
                     seed = TRUE)

dotplot(kk2_AgeCR, showCategory = 10, title = "Enriched Pathways" , split=".sign") + facet_grid(.~.sign) ## Saved as PDF 6x6 (landscape)

#############################################################################
## LFI in Mature CR
# MTCR vs MUCR
gene_list_LFIyoungCR_df <- as.data.frame(gene_list_LFIyoungCR) %>% rownames_to_column("gene") #11,852 genes
LFIyoungCR_KEGG <- merge(gene_list_LFIyoungCR_df, gene2tx2protein2ko_Apl, by="gene") #17,940 genes
LFIyoungCR_KEGG2 <- LFIyoungCR_KEGG[!duplicated(LFIyoungCR_KEGG$gene),] #11,422 genes
df_LFIyoungCR = LFIyoungCR_GO[LFIyoungCR_GO$Gene %in% LFIyoungCR_KEGG2$gene,] #11,422 genes
df_LFIyoungCR$ko = LFIyoungCR_KEGG2$ko #11,422 genes (just added ko column here)
LFIyoungCR_keggGeneList <- df_LFIyoungCR$log2FoldChange #11,422 genes (made list of lfc)
names(LFIyoungCR_keggGeneList) <- df_LFIyoungCR$ko #11,422 genes (named all lfc with ko for that gene)
LFIyoungCR_keggGeneList <- na.omit(LFIyoungCR_keggGeneList) #11,422 genes (no NA)
LFIyoungCR_keggGeneList = sort(LFIyoungCR_keggGeneList, decreasing = TRUE) #(sorted the lfc in decreasing order)

kk2_LFIyoungCR <- gseKEGG(geneList = LFIyoungCR_keggGeneList,
                     organism = "ko", #used ko terms from ghostKOWALA since Aplysia is not and available organism on KEGG website
                     minGSSize = 3,
                     maxGSSize = 800,
                     pvalueCutoff = 0.05,
                     pAdjustMethod = "none",
                     keyType = "kegg",
                     seed = TRUE)

dotplot(kk2_LFIyoungCR, showCategory = 10, title = "Enriched Pathways" , split=".sign") + facet_grid(.~.sign) ## Saved as PDF 6x6 (landscape)

#############################################################################
## LFI in Aged CR
# ATCR vs AUCR
gene_list_LFIageCR_df <- as.data.frame(gene_list_LFIageCR) %>% rownames_to_column("gene") #11,852 genes
LFIageCR_KEGG <- merge(gene_list_LFIageCR_df, gene2tx2protein2ko_Apl, by="gene") #17,940 genes
LFIageCR_KEGG2 <- LFIageCR_KEGG[!duplicated(LFIageCR_KEGG$gene),] #11,422 genes
df_LFIageCR = LFIageCR_GO[LFIageCR_GO$Gene %in% LFIageCR_KEGG2$gene,] #11,422 genes
df_LFIageCR$ko = LFIageCR_KEGG2$ko #11,422 genes (just added ko column here)
LFIageCR_keggGeneList <- df_LFIageCR$log2FoldChange #11,422 genes (made list of lfc)
names(LFIageCR_keggGeneList) <- df_LFIageCR$ko #11,422 genes (named all lfc with ko for that gene)
LFIageCR_keggGeneList <- na.omit(LFIageCR_keggGeneList) #11,422 genes (no NA)
LFIageCR_keggGeneList = sort(LFIageCR_keggGeneList, decreasing = TRUE) #(sorted the lfc in decreasing order)

kk2_LFIageCR <- gseKEGG(geneList = LFIageCR_keggGeneList,
                     organism = "ko", #used ko terms from ghostKOWALA since Aplysia is not and available organism on KEGG website
                     minGSSize = 3,
                     maxGSSize = 800,
                     pvalueCutoff = 0.05,
                     pAdjustMethod = "none",
                     keyType = "kegg",
                     seed = TRUE)

dotplot(kk2_LFIageCR, showCategory = 10, title = "Enriched Pathways" , split=".sign") + facet_grid(.~.sign) ## Saved as PDF 6x6 (landscape)

```






