---
title: "DTU"
output: html_document
date: "2025-02-07"
---
####### Load libraries
```{r}
library(dplyr)
library(tximport)
library(DESeq2)
library(ggplot2)
library(sva)
library(pheatmap) # for hierarchical clustering
library(purrr) # for "set_names()"
library(tibble)
library(ggVennDiagram)
library(UpSetR)
library(GenomicFeatures)
library(devtools) # for making the Org.db
library(AnnotationForge)
library(robustbase) # needed for rrcov
library(rrcov) # for statistically testing for outliers in PCA
library(stringr) # for str_detect to filter metadata
library(xlsx)
library(colorRamp2)

library(PCAtools)
library(ComplexHeatmap)
```

```{r}
# here I am going to perform DRIMseq to properly test for differential transcript usage using the workflow in the manuscript: Swimming downstream: statistical analysis of differential transcript usage following Salmon quantification (2018) Love et al.
# manuscript -> https://pubmed.ncbi.nlm.nih.gov/30356428/
# Vingette -> https://www.bioconductor.org/packages/release/workflows/vignettes/rnaseqDTU/inst/doc/rnaseqDTU.html
```

# ___________________________
# LFI_aged_AL (ATAL_vs_AUAL)
## Importing data
```{r}
#Decided to only use DRIMSeq to identify DTU and stop at the stageR step. 
```

```{r}
## setting up our metadata table
sample_table = read.csv("/Users/school/Projects/CR/samples.txt", header = T, stringsAsFactors = T)
sample_table$P_SAV <- as.character(sample_table$P_SAV)
sample_table$Age <- factor(sample_table$Age, levels = c("M","A") )
sample_table$TRT <- factor(sample_table$TRT, levels = c("U", "T"))
names(sample_table)[names(sample_table) == "TRT"] <- "LFI"
row.names(sample_table)=sample_table$Sample_ID
# below I create a 'group' column in the metadata
sample_table$group <- factor(paste0(sample_table$Age, sample_table$LFI, sample_table$Diet))
# checking the levels of 'group' column below
levels(sample_table$group)
sample_table$group <- relevel(sample_table$group, ref = "MUCR")

# Sample 840 (Aged, Trained, CR ; ATCR) is technically a dud since its TTIM on day1 = 68 sec. This animal was retrained and dissected because we were short on aged CR animals. Still, I think it best to exclude this sample.
# Samples 809 and 805 were determined to be outliers after performing a Robust PCA. Please see the code chunk above this one for more context. Both samples were (Mature, Untrained, CR ; MUCR)
# I will delete this sample from the metadata file which will be used to import the sample data, so this sample will not be imported

metadata <- sample_table[!(sample_table$Sample_ID %in% c("840","809","805")),]
#############################
## making our sample paths to quant files

paste0('/Users/school/Projects/CR/quants/', 
       pull(metadata, RNAseq_ID), '/quant.sf')
sample_files = paste0('/Users/school/Projects/CR/quants/',
                      pull(metadata, RNAseq_ID), '/quant.sf')
names(sample_files) = pull(metadata, Sample_ID)
all(file.exists(sample_files))
data.frame(thefiles = metadata, doihave = file.exists(sample_files))


#################################
## we now import transcript level counts usinf tximport
txi <- tximport(sample_files,
                type = "salmon",
                txOut = T,
                countsFromAbundance = "scaledTPM")
cts_LFIagedAL <- txi$counts
dim(cts_LFIagedAL) # 29,261 transcripts in 45 samples
cts_LFIagedAL <- cts_LFIagedAL[rowSums(cts_LFIagedAL) > 0,]
dim(cts_LFIagedAL) # 19,742 transcripts in 45 samples
#View(cts)

############################################
## Transcript-to-gene mapping
gff <- "/Users/school/Documents/Documents/UAB/UAB009/genome/AplCal3.0_GFF.gff"
txdb.filename <- "/Users/school/Documents/Documents/UAB/UAB009/genome/AplCal3.0_GFF.sqlite"
txdb <- makeTxDbFromGFF(gff)
saveDb(txdb, txdb.filename)

txdb <- loadDb(txdb.filename)
txdf <- AnnotationDbi::select(txdb, keys(txdb, "GENEID"), 
               "TXNAME", "GENEID")
tab <- table(txdf$GENEID)
txdf$ntx <- tab[match(txdf$GENEID, names(tab))]

```

## DRIMSeq
```{r}
cts[1:3,1:3]
range(colSums(cts)/1e6)
head(txdf)
all(rownames(cts) %in% txdf$TXNAME) # this is FALSE
dim(cts) # this has less transcripts (I already filtered) that is why it is FALSE above. 19,742 transcripts
dim(txdf) # 29,250 transcripts
txdf <- txdf[match(rownames(cts), txdf$TXNAME),]
dim(txdf) # now there are 19,742 transcripts
all(rownames(cts) == txdf$TXNAME) # this is still FALSE/NA
# let's test what is making this FALSE
which(!rownames(cts) %in% txdf$TXNAME)
cts[1264,,drop=F]
txdf[1264,]
# OK I found them. I will make them into a vector and then eliminate them from the cts object and match the two dataframes again to see if that works
getRid <- which(!rownames(cts) %in% txdf$TXNAME)
#rownames_to_column(cts)
typeof(cts)# "double"
class(cts) # "matrix" "array"
test <- cts[getRid,,drop=F]
test <- rownames(test)
test2 <- as.data.frame(cts)
test2 <- rownames_to_column(test2)
test3 <- test2[!(test2$rowname %in% test),]
test3 <- as.data.frame(test3)
rownames(test3) <- test3$rowname
cts <- test3[,-1] 
cts <- as.matrix(cts)
typeof(cts) # "double"
class(cts) # "matrix" "array"
txdf <- txdf[match(rownames(cts), txdf$TXNAME),]
dim(txdf) # now we are at 19,690
all(rownames(cts) == txdf$TXNAME)


#####
## build a dataframe with geneID, transcript (feature) ID, and column for each sample
counts <- data.frame(gene_id = txdf$GENEID,
                     feature_id = txdf$TXNAME,
                     cts)
colnames(counts) <- sub("X","",colnames(counts))
```

```{r}
# Now we will load in DRIMSeq and create a dmDSdata object with our counts and samples
#BiocManager::install("DRIMSeq")
library(DRIMSeq)
metadata
colnames(metadata)[1] <- "sample_id"
metadata$sample_id <- as.character(metadata$sample_id)

d <- dmDSdata(counts=counts, samples = metadata)
d # 15,026 genes 45 samples
#View(d)
methods(class = class(d))
counts(d[1,])[,1:4]

```

```{r}
#We use all three of the possible filters: for a transcript to be retained in the dataset, we require that (1) it has a count of at least 10 in at least n.small samples, (2) it has a relative abundance proportion of at least 0.1 in at least n.small samples, and (3) the total count of the corresponding gene is at least 10 in all n samples.
dim(counts)
n <- 45 # total n of samples
n.small <- 4 # smallest group size
d <- dmFilter(d,
              min_samps_feature_expr=n.small, min_feature_expr=1,
              min_samps_feature_prop=n.small, min_feature_prop=0.01,
              min_samps_gene_expr=n, min_gene_expr=1)
# I only get 296 genes back!

# test how many genes have N isoforms 
table(table(counts(d)$gene_id))

# Lets make a model design
design_full <- model.matrix(~ 0 + group, data=DRIMSeq::samples(d))
colnames(design_full)
design_full %>% View()

set.seed(1)

d_LFIagedAL <- dmPrecision(d, design=design_full)
d_LFIagedAL <- dmFit(d_LFIagedAL, design=design_full)
d_LFIagedAL <- dmTest(d_LFIagedAL, contrast=c(0,1,0,-1,0,0,0,0)) # LFI aged AL contrast

# we can either get a p-value per gene, which tests whether there is any differnetial transcript usage in that gene
res_LFIagedAL <- DRIMSeq::results(d_LFIagedAL)
head(res_LFIagedAL)
View(res_LFIagedAL)
# Or we can get a p-value per transcript, to test whether the proportions for that transcript chnages with the gene
res.txp_LFIagedAL <- DRIMSeq::results(d_LFIagedAL, level="feature")
head(res.txp_LFIagedAL)
View(res.txp_LFIagedAL)

# turn any "NA"s in the p-value column into 1's so not to mess up downstream functions
no.na <- function(x) ifelse(is.na(x), 1, x)
res_LFIagedAL$pvalue <- no.na(res_LFIagedAL$pvalue)
res.txp_LFIagedAL$pvalue <- no.na(res.txp_LFIagedAL$pvalue)

# now let's plot the estimated proportions for one of the sigificant genes, where we can see evidence of switching
idx_LFIagedCR <- which(res_LFIagedCR$adj_pvalue < 0.05)[1] # there is only one gene that has a p.adj<0.05 = LOC101862197 = septin-11 [ Aplysia californica (California sea hare) ] > Filament-forming cytoskeletal GTPase. May play a role in cytokinesis (Potential). May play a role in the cytoarchitecture of neurons, including dendritic arborization and dendritic spines, and in GABAergic synaptic connectivity (By similarity). < from Uniprot septin-11 in humans 
res[idx,]


plotProportions(d, gene_id=res$gene_id[idx], group_variable="LFI", plot_type="barplot") # I keep getting back stacked barplots when I want barplots with position=dodge so they are side by side like in the vingette but I cannot figure out how to get that so I am moving on for now

# plot gene-level p-values
plotPValues(d_LFIagedAL)

# to get the log2 fold changes use the below code
DRIMSeq::proportions(d_LFIagedAL)


## If I use boxplots it will seperate the leves in the group factor in plotProportion()
plotProportions(d_LFIagedAL, gene_id = "LOC101862197", group_variable = "LFI", plot_type = "boxplot1")

## Let's try something ### everything commented out was my attempt. the bottom part works though!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
tttest <- subset(counts,counts$gene_id == "LOC101862197")
tttest2 <- tttest[,c("gene_id","feature_id","852","855","857","858","860","866","872","873","875","876","879","880")]
View(tttest2)

## below is it!!!
data_longer <- tidyr::pivot_longer(tttest2[2:14], -feature_id)
data_longer$LFI <- ifelse(data_longer$name=="872" | data_longer$name=="873" | data_longer$name=="875" | data_longer$name=="876" | data_longer$name=="879" | data_longer$name=="880", "TT2UCR","TT2TCR")
ggplot(data_longer, aes(x=feature_id, y=value, fill=LFI)) +
  geom_boxplot(position = position_dodge()) +
  ggtitle("LOC101862197 â€“ septin-11")## wow! that was a deal but I got it!!!!!!!!!!!
```

### stageR following DRIMseq
```{r}
dim(res)
dim(res.txp)

# below we are keeping only the first 15 characters of the transcript NCBI IDs
pScreen <- res$pvalue
strp <- function(x) substr(x,1,15)
names(pScreen) <- strp(res$gene_id)
# we make one coulmn matric of confirmation p-values
pConfirmation <- matrix(res.txp$pvalue, ncol=1)
rownames(pConfirmation) <- strp(res.txp$feature_id)
# We arrange a 2 coumn data frame with transcript and gene_IDs
tx2gene <- res.txp[,c("feature_id","gene_id")]
for(i in 1:2) tx2gene[,i] <- strp(tx2gene[,i])
# the following performns the 'stageR' analysis
library(stageR)
stageRObj <- stageRTx(pScreen=pScreen,
                      pConfirmation=pConfirmation,
                      pScreenAdjusted=F,
                      tx2gene=tx2gene)
stageRObj <- stageWiseAdjustment(stageRObj,
                                 method="dtu",
                                 alpha=0.05)
suppressWarnings({
  drim.padj <- getAdjustedPValues(stageRObj, 
                                  order=F, onlySignificantGenes=T)
})
head(drim.padj) # LOC101862197 same septin-11 as above in DRIMSeq. the gene is significant but the transcripts are not
View(drim.padj)
# The final table above with adjusted p-values summarizes the information from the two-stage analysis. Only genes that passed the filter are included in the table, so the table already represents screened genes
# The transcripts with values in the column, transcript, less than 0.05 pass the confirmation stage on a target 5% overall false discovery rate, or OFDR. This means that, in expectation, no more than 5% of the genes that pass screening will either (1) not contain any DTU, so be falsely screened genes, or (2) contain a falsely confirmed transcript.
```


# ___________________________
# Age_AL (AUAL_vs_MUAL)
## DRIMSeq
```{r}
d
set.seed(1)

d_AgeAL <- dmPrecision(d, design=design_full)
d_AgeAL <- dmFit(d_AgeAL, design=design_full)
d_AgeAL <- dmTest(d_AgeAL, contrast=c(0,0,0,1,0,0,0,-1)) # Age AL contrast

# we can either get a p-value per gene, which tests whether there is any differential transcript usage in that gene
res_AgeAL <- DRIMSeq::results(d_AgeAL)
head(res_AgeAL)
View(res_AgeAL)
# Or we can get a p-value per transcript, to test whether the proportions for that transcript chnages with the gene
res.txp_AgeAL <- DRIMSeq::results(d_AgeAL, level="feature")
head(res.txp_AgeAL)
View(res.txp_AgeAL)

# turn any "NA"s in the p-value column into 1's so not to mess up downstream functions
no.na <- function(x) ifelse(is.na(x), 1, x)
res_AgeAL$pvalue <- no.na(res_AgeAL$pvalue)
res.txp_AgeAL$pvalue <- no.na(res.txp_AgeAL$pvalue)

# now let's plot the estimated proportions for one of the sigificant genes, where we can see evidence of switching
idx_AgeAL <- which(res_AgeAL$adj_pvalue < 0.05) #
res_AgeAL[idx_AgeAL,]
# to check which are significant (below to get the first one)
idx_AgeAL <- which(res_AgeAL$adj_pvalue < 0.05)[1] #
res_AgeAL[idx_AgeAL,]

plotProportions(d_AgeAL, gene_id=res_AgeAL$gene_id[idx_AgeAL], group_variable="LFI", plot_type="barplot") # I keep getting back stacked barplots when I want barplots with position=dodge so they are side by side like in the vingette but I cannot figure out how to get that so I am moving on for now


# plot gene-level p-values
plotPValues(d_AgeAL)

# to get the log2 fold changes use the below code
DRIMSeq::proportions(d_AgeAL)


## If I use boxplots it will seperate the leves in the group factor in plotProportion()
plotProportions(d_AgeAL, gene_id = "LOC101850213", group_variable = "LFI", plot_type = "boxplot1")

## Let's try something ### everything commented out was my attempt. the bottom part works though!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
tttest <- subset(counts,counts$gene_id == "LOC101845050")
tttest2 <- tttest[,c("gene_id","feature_id","863","865","874","877","878","881","810","816","828","830","836","837")]
View(tttest2)


## below is it!!!
data_longer <- tidyr::pivot_longer(tttest2[2:14], -feature_id)
data_longer$Age <- ifelse(data_longer$name=="810" | data_longer$name=="816" | data_longer$name=="828" | data_longer$name=="830" | data_longer$name=="836" | data_longer$name=="837", "TT1UAL","TT2UAL")
ggplot(data_longer, aes(x=feature_id, y=value, fill=Age)) +
  geom_boxplot(position = position_dodge()) +
  ggtitle("LOC101845050 - tropomyosin-2") +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=0))## wow! that was a deal but I got it!!!!!!!!!!!
```

### stageR following DRIMseq
```{r}
dim(res_AgeAL)
dim(res.txp_AgeAL)

# below we are keeping only the first 15 characters of the transcript NCBI IDs
pScreen_AgeAL <- res_AgeAL$pvalue
strp <- function(x) substr(x,1,15)
names(pScreen_AgeAL) <- strp(res_AgeAL$gene_id)
# we make one coulmn matric of confirmation p-values
pConfirmation_AgeAL <- matrix(res.txp_AgeAL$pvalue, ncol=1)
rownames(pConfirmation_AgeAL) <- strp(res.txp_AgeAL$feature_id)
# We arrange a 2 coumn data frame with transcript and gene_IDs
tx2gene_AgeAL <- res.txp_AgeAL[,c("feature_id","gene_id")]
for(i in 1:2) tx2gene_AgeAL[,i] <- strp(tx2gene_AgeAL[,i])
# the following performns the 'stageR' analysis
library(stageR)
stageRObj_AgeAL <- stageRTx(pScreen=pScreen_AgeAL,
                      pConfirmation=pConfirmation_AgeAL,
                      pScreenAdjusted=F,
                      tx2gene=tx2gene)
stageRObj_AgeAL <- stageWiseAdjustment(stageRObj_AgeAL,
                                 method="dtu",
                                 alpha=0.1)
suppressWarnings({
  drim.padj_AgeAL <- getAdjustedPValues(stageRObj_AgeAL, 
                                  order=F, onlySignificantGenes=T)
})
head(drim.padj_AgeAL)
View(drim.padj_AgeAL)# LOC101862197 same septin-11 as above in DRIMSeq. the gene is significant but the transcripts are not

# The final table above with adjusted p-values summarizes the information from the two-stage analysis. Only genes that passed the filter are included in the table, so the table already represents screened genes
# The transcripts with values in the column, transcript, less than 0.05 pass the confirmation stage on a target 5% overall false discovery rate, or OFDR. This means that, in expectation, no more than 5% of the genes that pass screening will either (1) not contain any DTU, so be falsely screened genes, or (2) contain a falsely confirmed transcript.
```

# ___________________________
# LFI_young_AL (MTAL_vs_MUAL)
## DRIMSeq
```{r}
d
set.seed(1)

d_LFIyoungAL <- dmPrecision(d, design=design_full)
d_LFIyoungAL <- dmFit(d_LFIyoungAL, design=design_full)
d_LFIyoungAL <- dmTest(d_LFIyoungAL, contrast=c(0,0,0,0,0,1,0,-1)) # Age AL contrast

# we can either get a p-value per gene, which tests whether there is any differnetial transcript usage in that gene
res_LFIyoungAL <- DRIMSeq::results(d_LFIyoungAL)
head(res_LFIyoungAL)
View(res_LFIyoungAL)
# Or we can get a p-value per transcript, to test whether the proportions for that transcript chnages with the gene
res.txp_LFIyoungAL <- DRIMSeq::results(d_LFIyoungAL, level="feature")
head(res.txp_LFIyoungAL)
View(res.txp_LFIyoungAL)

# turn any "NA"s in the p-value column into 1's so not to mess up downstream functions
no.na <- function(x) ifelse(is.na(x), 1, x)
res_LFIyoungAL$pvalue <- no.na(res_LFIyoungAL$pvalue)
res.txp_LFIyoungAL$pvalue <- no.na(res.txp_LFIyoungAL$pvalue)

# now let's plot the estimated proportions for one of the sigificant genes, where we can see evidence of switching
idx_LFIyoungAL <- which(res_LFIyoungAL$adj_pvalue < 0.05)[1] # there is only one gene that has a p.adj<0.05 = LOC101862197 = septin-11 [ Aplysia californica (California sea hare) ] > Filament-forming cytoskeletal GTPase. May play a role in cytokinesis (Potential). May play a role in the cytoarchitecture of neurons, including dendritic arborization and dendritic spines, and in GABAergic synaptic connectivity (By similarity). < from Uniprot septin-11 in humans 
res[idx,]


plotProportions(d, gene_id=res$gene_id[idx], group_variable="LFI", plot_type="barplot") # I keep getting back stacked barplots when I want barplots with position=dodge so they are side by side like in the vingette but I cannot figure out how to get that so I am moving on for now


# plot gene-level p-values
plotPValues(d)

# to get the log2 fold changes use the below code
DRIMSeq::proportions(d)


## If I use boxplots it will seperate the leves in the group factor in plotProportion()
plotProportions(d, gene_id = "LOC101862197", group_variable = "LFI", plot_type = "boxplot1")

## Let's try something ### everything commented out was my attempt. the bottom part works though!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
tttest <- subset(counts,counts$gene_id == "LOC101862197")
tttest2 <- tttest[,c("gene_id","feature_id","839","841","847","849","871","863","865","874","877","878","881")]
View(tttest2)

## below is it!!!
data_longer <- tidyr::pivot_longer(tttest2[2:13], -feature_id)
data_longer$LFI <- ifelse(data_longer$name=="863" | data_longer$name=="865" | data_longer$name=="874" | data_longer$name=="877" | data_longer$name=="878" | data_longer$name=="881", "AUAL","ATAL")
ggplot(data_longer, aes(x=feature_id, y=value, fill=LFI)) +
  geom_boxplot(position = position_dodge()) ## wow! that was a deal but I got it!!!!!!!!!!!
```


### stageR following DRIMseq
```{r}
dim(res_LFIyoungAL)
dim(res.txp_LFIyoungAL)

# below we are keeping only the first 15 characters of the transcript NCBI IDs
pScreen_LFIyoungAL <- res_LFIyoungAL$pvalue
strp <- function(x) substr(x,1,15)
names(pScreen_LFIyoungAL) <- strp(res_LFIyoungAL$gene_id)
# we make one coulmn matric of confirmation p-values
pConfirmation_LFIyoungAL <- matrix(res.txp_LFIyoungAL$pvalue, ncol=1)
rownames(pConfirmation_LFIyoungAL) <- strp(res.txp_LFIyoungAL$feature_id)
# We arrange a 2 coumn data frame with transcript and gene_IDs
tx2gene_LFIyoungAL <- res.txp_LFIyoungAL[,c("feature_id","gene_id")]
for(i in 1:2) tx2gene_LFIyoungAL[,i] <- strp(tx2gene_LFIyoungAL[,i])
# the following performns the 'stageR' analysis
library(stageR)
stageRObj_LFIyoungAL <- stageRTx(pScreen=pScreen_LFIyoungAL,
                      pConfirmation=pConfirmation_LFIyoungAL,
                      pScreenAdjusted=F,
                      tx2gene=tx2gene)
stageRObj_LFIyoungAL <- stageWiseAdjustment(stageRObj_LFIyoungAL,
                                 method="dtu",
                                 alpha=0.05)
suppressWarnings({
  drim.padj_LFIyoungAL <- getAdjustedPValues(stageRObj_LFIyoungAL, 
                                  order=F, onlySignificantGenes=T)
})
head(drim.padj_LFIyoungAL) # LOC101862197 same septin-11 as above in DRIMSeq. the gene is significant but the transcripts are not

# The final table above with adjusted p-values summarizes the information from the two-stage analysis. Only genes that passed the filter are included in the table, so the table already represents screened genes
# The transcripts with values in the column, transcript, less than 0.05 pass the confirmation stage on a target 5% overall false discovery rate, or OFDR. This means that, in expectation, no more than 5% of the genes that pass screening will either (1) not contain any DTU, so be falsely screened genes, or (2) contain a falsely confirmed transcript.
```



# ___________________________
# Age_CR (AUCR_vs_MUCR)
## DRIMSeq
```{r}
d
set.seed(1)

d_AgeCR <- dmPrecision(d, design=design_full)
d_AgeCR <- dmFit(d_AgeCR, design=design_full)
d_AgeCR <- dmTest(d_AgeCR, contrast=c(-1,0,0,0,1,0,0,0)) # Age CR contrast

# we can either get a p-value per gene, which tests whether there is any differnetial transcript usage in that gene
res_AgeCR <- DRIMSeq::results(d_AgeCR)
head(res_AgeCR)
View(res_AgeCR)
# Or we can get a p-value per transcript, to test whether the proportions for that transcript chnages with the gene
res.txp_AgeCR <- DRIMSeq::results(d_AgeCR, level="feature")
head(res.txp_AgeCR)
View(res.txp_AgeCR)

# turn any "NA"s in the p-value column into 1's so not to mess up downstream functions
no.na <- function(x) ifelse(is.na(x), 1, x)
res_AgeCR$pvalue <- no.na(res_AgeCR$pvalue)
res.txp_AgeCR$pvalue <- no.na(res.txp_AgeCR$pvalue)

# now let's plot the estimated proportions for one of the sigificant genes, where we can see evidence of switching
idx_AgeCR <- which(res_AgeCR$adj_pvalue < 0.05) # 
res[idx,]


plotProportions(d, gene_id=res$gene_id[idx], group_variable="LFI", plot_type="barplot") # I keep getting back stacked barplots when I want barplots with position=dodge so they are side by side like in the vingette but I cannot figure out how to get that so I am moving on for now


# plot gene-level p-values
plotPValues(d)

# to get the log2 fold changes use the below code
DRIMSeq::proportions(d)


## If I use boxplots it will seperate the leves in the group factor in plotProportion()
plotProportions(d, gene_id = "LOC101862197", group_variable = "LFI", plot_type = "boxplot1")

## Let's try something ### everything commented out was my attempt. the bottom part works though!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
tttest <- subset(counts,counts$gene_id == "LOC101845050")
tttest2 <- tttest[,c("gene_id","feature_id","815","823","824","829","872","873","875","876","879","880")]
View(tttest2)


## below is it!!!
data_longer <- tidyr::pivot_longer(tttest2[2:12], -feature_id)
data_longer$Age <- ifelse(data_longer$name=="872" | data_longer$name=="873" | data_longer$name=="875" | data_longer$name=="876" | data_longer$name=="879" | data_longer$name=="880", "TT2UCR","TT1UCR")
ggplot(data_longer, aes(x=feature_id, y=value, fill=Age)) +
  geom_boxplot(position = position_dodge()) +
  ggtitle("LOC101845050 - tropomyosin-2") +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=0))## wow! that was a deal but I got it!!!!!!!!!!!
```


### stageR following DRIMseq
```{r}
dim(res_AgeCR)
dim(res.txp_AgeCR)

# below we are keeping only the first 15 characters of the transcript NCBI IDs
pScreen_AgeCR <- res_AgeCR$pvalue
strp <- function(x) substr(x,1,15)
names(pScreen_AgeCR) <- strp(res_AgeCR$gene_id)
# we make one coulmn matric of confirmation p-values
pConfirmation_AgeCR <- matrix(res.txp_AgeCR$pvalue, ncol=1)
rownames(pConfirmation_AgeCR) <- strp(res.txp_AgeCR$feature_id)
# We arrange a 2 coumn data frame with transcript and gene_IDs
tx2gene_AgeCR <- res.txp_AgeCR[,c("feature_id","gene_id")]
for(i in 1:2) tx2gene_AgeCR[,i] <- strp(tx2gene_AgeCR[,i])
# the following performns the 'stageR' analysis
library(stageR)
stageRObj_AgeCR <- stageRTx(pScreen=pScreen_AgeCR,
                      pConfirmation=pConfirmation_AgeCR,
                      pScreenAdjusted=F,
                      tx2gene=tx2gene)
stageRObj_AgeCR <- stageWiseAdjustment(stageRObj_AgeCR,
                                 method="dtu",
                                 alpha=0.05)
suppressWarnings({
  drim.padj_AgeCR <- getAdjustedPValues(stageRObj_AgeCR, 
                                  order=F, onlySignificantGenes=T)
})
head(drim.padj_AgeCR) # LOC101862197 same septin-11 as above in DRIMSeq. the gene is significant but the transcripts are not

# The final table above with adjusted p-values summarizes the information from the two-stage analysis. Only genes that passed the filter are included in the table, so the table already represents screened genes
# The transcripts with values in the column, transcript, less than 0.05 pass the confirmation stage on a target 5% overall false discovery rate, or OFDR. This means that, in expectation, no more than 5% of the genes that pass screening will either (1) not contain any DTU, so be falsely screened genes, or (2) contain a falsely confirmed transcript.
```





# ___________________________
# LFI_young_CR (MTCR_vs_MUCR)
## DRIMSeq
```{r}
d
set.seed(1)

d_LFIyoungCR <- dmPrecision(d, design=design_full)
d_LFIyoungCR <- dmFit(d_LFIyoungCR, design=design_full)
d_LFIyoungCR <- dmTest(d_LFIyoungCR, contrast=c(-1,0,0,0,0,0,1,0)) # LFI young CR contrast

# we can either get a p-value per gene, which tests whether there is any differnetial transcript usage in that gene
res_LFIyoungCR <- DRIMSeq::results(d_LFIyoungCR)
head(res_LFIyoungCR)
View(res_LFIyoungCR)
# Or we can get a p-value per transcript, to test whether the proportions for that transcript chnages with the gene
res.txp_LFIyoungCR <- DRIMSeq::results(d_LFIyoungCR, level="feature")
head(res.txp_LFIyoungCR)
View(res.txp_LFIyoungCR)

# turn any "NA"s in the p-value column into 1's so not to mess up downstream functions
no.na <- function(x) ifelse(is.na(x), 1, x)
res_LFIyoungCR$pvalue <- no.na(res_LFIyoungCR$pvalue)
res.txp_LFIyoungCR$pvalue <- no.na(res.txp_LFIyoungCR$pvalue)

# now let's plot the estimated proportions for one of the sigificant genes, where we can see evidence of switching
idx_LFIyoungCR <- which(res_LFIyoungCR$adj_pvalue < 0.05)[1] # there is only one gene that has a p.adj<0.05 = LOC101862197 = septin-11 [ Aplysia californica (California sea hare) ] > Filament-forming cytoskeletal GTPase. May play a role in cytokinesis (Potential). May play a role in the cytoarchitecture of neurons, including dendritic arborization and dendritic spines, and in GABAergic synaptic connectivity (By similarity). < from Uniprot septin-11 in humans 
res[idx,]


plotProportions(d, gene_id=res$gene_id[idx], group_variable="LFI", plot_type="barplot") # I keep getting back stacked barplots when I want barplots with position=dodge so they are side by side like in the vingette but I cannot figure out how to get that so I am moving on for now


# plot gene-level p-values
plotPValues(d)

# to get the log2 fold changes use the below code
DRIMSeq::proportions(d)


## If I use boxplots it will seperate the leves in the group factor in plotProportion()
plotProportions(d, gene_id = "LOC101862197", group_variable = "LFI", plot_type = "boxplot1")

## Let's try something ### everything commented out was my attempt. the bottom part works though!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
tttest <- subset(counts,counts$gene_id == "LOC101862197")
tttest2 <- tttest[,c("gene_id","feature_id","839","841","847","849","871","863","865","874","877","878","881")]
View(tttest2)

## below is it!!!
data_longer <- tidyr::pivot_longer(tttest2[2:13], -feature_id)
data_longer$LFI <- ifelse(data_longer$name=="863" | data_longer$name=="865" | data_longer$name=="874" | data_longer$name=="877" | data_longer$name=="878" | data_longer$name=="881", "AUAL","ATAL")
ggplot(data_longer, aes(x=feature_id, y=value, fill=LFI)) +
  geom_boxplot(position = position_dodge()) ## wow! that was a deal but I got it!!!!!!!!!!!
```


### stageR following DRIMseq
```{r}
dim(res_LFIyoungCR)
dim(res.txp_LFIyoungCR)

# below we are keeping only the first 15 characters of the transcript NCBI IDs
pScreen_LFIyoungCR <- res_LFIyoungCR$pvalue
strp <- function(x) substr(x,1,15)
names(pScreen_LFIyoungCR) <- strp(res_LFIyoungCR$gene_id)
# we make one coulmn matric of confirmation p-values
pConfirmation_LFIyoungCR <- matrix(res.txp_LFIyoungCR$pvalue, ncol=1)
rownames(pConfirmation_LFIyoungCR) <- strp(res.txp_LFIyoungCR$feature_id)
# We arrange a 2 coumn data frame with transcript and gene_IDs
tx2gene_LFIyoungCR <- res.txp_LFIyoungCR[,c("feature_id","gene_id")]
for(i in 1:2) tx2gene_LFIyoungCR[,i] <- strp(tx2gene_LFIyoungCR[,i])
# the following performns the 'stageR' analysis
library(stageR)
stageRObj_LFIyoungCR <- stageRTx(pScreen=pScreen_LFIyoungCR,
                      pConfirmation=pConfirmation_LFIyoungCR,
                      pScreenAdjusted=F,
                      tx2gene=tx2gene)
stageRObj_LFIyoungCR <- stageWiseAdjustment(stageRObj_LFIyoungCR,
                                 method="dtu",
                                 alpha=0.05)
suppressWarnings({
  drim.padj_LFIyoungCR <- getAdjustedPValues(stageRObj_LFIyoungCR, 
                                  order=F, onlySignificantGenes=T)
})
head(drim.padj_LFIyoungCR) # LOC101862197 same septin-11 as above in DRIMSeq. the gene is significant but the transcripts are not

# The final table above with adjusted p-values summarizes the information from the two-stage analysis. Only genes that passed the filter are included in the table, so the table already represents screened genes
# The transcripts with values in the column, transcript, less than 0.05 pass the confirmation stage on a target 5% overall false discovery rate, or OFDR. This means that, in expectation, no more than 5% of the genes that pass screening will either (1) not contain any DTU, so be falsely screened genes, or (2) contain a falsely confirmed transcript.
```




# ___________________________
# LFI_aged_CR (ATCR_vs_AUCR)
## DRIMSeq
```{r}
d
set.seed(1)

d_LFIagedCR <- dmPrecision(d, design=design_full)
d_LFIagedCR <- dmFit(d_LFIagedCR, design=design_full)
d_LFIagedCR <- dmTest(d_LFIagedCR, contrast=c(0,0,1,0,-1,0,0,0)) # LFI young CR contrast

# we can either get a p-value per gene, which tests whether there is any differnetial transcript usage in that gene
res_LFIagedCR <- DRIMSeq::results(d_LFIagedCR)
head(res_LFIagedCR)
View(res_LFIagedCR)
# Or we can get a p-value per transcript, to test whether the proportions for that transcript chnages with the gene
res.txp_LFIagedCR <- DRIMSeq::results(d_LFIagedCR, level="feature")
head(res.txp_LFIagedCR)
View(res.txp_LFIagedCR)

# turn any "NA"s in the p-value column into 1's so not to mess up downstream functions
no.na <- function(x) ifelse(is.na(x), 1, x)
res_LFIagedCR$pvalue <- no.na(res_LFIagedCR$pvalue)
res.txp_LFIagedCR$pvalue <- no.na(res.txp_LFIagedCR$pvalue)

# now let's plot the estimated proportions for one of the sigificant genes, where we can see evidence of switching
idx_LFIagedCR <- which(res_LFIagedCR$adj_pvalue < 0.05)[1] # there is only one gene that has a p.adj<0.05 = LOC101862197 = septin-11 [ Aplysia californica (California sea hare) ] > Filament-forming cytoskeletal GTPase. May play a role in cytokinesis (Potential). May play a role in the cytoarchitecture of neurons, including dendritic arborization and dendritic spines, and in GABAergic synaptic connectivity (By similarity). < from Uniprot septin-11 in humans 
res[idx,]


plotProportions(d, gene_id=res$gene_id[idx], group_variable="LFI", plot_type="barplot") # I keep getting back stacked barplots when I want barplots with position=dodge so they are side by side like in the vingette but I cannot figure out how to get that so I am moving on for now


# plot gene-level p-values
plotPValues(d)

# to get the log2 fold changes use the below code
DRIMSeq::proportions(d)


## If I use boxplots it will seperate the leves in the group factor in plotProportion()
plotProportions(d, gene_id = "LOC101862197", group_variable = "LFI", plot_type = "boxplot1")

## Let's try something ### everything commented out was my attempt. the bottom part works though!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
tttest <- subset(counts,counts$gene_id == "LOC101862197")
tttest2 <- tttest[,c("gene_id","feature_id","839","841","847","849","871","863","865","874","877","878","881")]
View(tttest2)

## below is it!!!
data_longer <- tidyr::pivot_longer(tttest2[2:13], -feature_id)
data_longer$LFI <- ifelse(data_longer$name=="863" | data_longer$name=="865" | data_longer$name=="874" | data_longer$name=="877" | data_longer$name=="878" | data_longer$name=="881", "AUAL","ATAL")
ggplot(data_longer, aes(x=feature_id, y=value, fill=LFI)) +
  geom_boxplot(position = position_dodge()) ## wow! that was a deal but I got it!!!!!!!!!!!
```


### stageR following DRIMseq
```{r}
dim(res_LFIagedCR)
dim(res.txp_LFIagedCR)

# below we are keeping only the first 15 characters of the transcript NCBI IDs
pScreen_LFIagedCR <- res_LFIagedCR$pvalue
strp <- function(x) substr(x,1,15)
names(pScreen_LFIagedCR) <- strp(res_LFIagedCR$gene_id)
# we make one coulmn matric of confirmation p-values
pConfirmation_LFIagedCR <- matrix(res.txp_LFIagedCR$pvalue, ncol=1)
rownames(pConfirmation_LFIagedCR) <- strp(res.txp_LFIagedCR$feature_id)
# We arrange a 2 coumn data frame with transcript and gene_IDs
tx2gene_LFIagedCR <- res.txp_LFIagedCR[,c("feature_id","gene_id")]
for(i in 1:2) tx2gene_LFIagedCR[,i] <- strp(tx2gene_LFIagedCR[,i])
# the following performns the 'stageR' analysis
library(stageR)
stageRObj_LFIagedCR <- stageRTx(pScreen=pScreen_LFIagedCR,
                      pConfirmation=pConfirmation_LFIagedCR,
                      pScreenAdjusted=F,
                      tx2gene=tx2gene)
stageRObj_LFIagedCR <- stageWiseAdjustment(stageRObj_LFIagedCR,
                                 method="dtu",
                                 alpha=0.05)
suppressWarnings({
  drim.padj_LFIagedCR <- getAdjustedPValues(stageRObj_LFIagedCR, 
                                  order=F, onlySignificantGenes=T)
})
head(drim.padj_LFIagedCR) # LOC101862197 same septin-11 as above in DRIMSeq. the gene is significant but the transcripts are not

# The final table above with adjusted p-values summarizes the information from the two-stage analysis. Only genes that passed the filter are included in the table, so the table already represents screened genes
# The transcripts with values in the column, transcript, less than 0.05 pass the confirmation stage on a target 5% overall false discovery rate, or OFDR. This means that, in expectation, no more than 5% of the genes that pass screening will either (1) not contain any DTU, so be falsely screened genes, or (2) contain a falsely confirmed transcript.
```






